From bf1939c3f989cb4076abb7fb83b0b16fbe8daeef Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 26 Apr 2014 14:50:22 -0700
Subject: [PATCH] Add run_command to qpt

---
 util/qpt.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/util/qpt.py b/util/qpt.py
index 69d63d1..f58c71b 100644
--- a/util/qpt.py
+++ b/util/qpt.py
@@ -50,6 +50,11 @@ def _send(_queue, line):
         _queue.put(line.strip())
 
 
+def _run_command(conn, _queue, text):
+    _send(_queue, ":TestUser!~user@example.com PRIVMSG #channel :"
+                  + conn.config.get("command_prefix") + text)
+
+
 def run_test(obj_with_shutdown, func, bot, conn, input_queue, output_queue):
     """
     :type bot: core.bot.CloudBot
@@ -81,6 +86,8 @@ def run_test(obj_with_shutdown, func, bot, conn, input_queue, output_queue):
                 params.append(lambda: output_queue.get(timeout=5))
             elif arg == "send":
                 params.append(lambda line: _send(input_queue, line))
+            elif arg == "run_command":
+                params.append(lambda text: _run_command(conn, input_queue, text))
             elif arg == "test_number":
                 params.append(test_number)
             else:
-- 
1.9.2

