From f4c79c3541066c927035d90f1edb74ca43c112a9 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 18 Jun 2014 20:12:16 -0500
Subject: [PATCH] Not at all a tracking system - TODO: improve


diff --git a/cloudbot/client.py b/cloudbot/client.py
index 7909fd6..34ebb2e 100644
--- a/cloudbot/client.py
+++ b/cloudbot/client.py
@@ -1,4 +1,5 @@
 import asyncio
+from collections import deque
 import logging
 
 from cloudbot.permissions import PermissionManager
@@ -121,3 +122,49 @@ class Client:
     @property
     def connected(self):
         raise NotImplementedError
+
+
+# TODO: Tracking of user 'mode' in channels
+class User:
+    """
+    :param name: The nickname of this User
+    :param ident: The IRC ident of this User, if applicable
+    :param host: The hostname of this User, if applicable
+    :param mask: The IRC mask (nick!ident@host), if applicable
+    :type name: str
+    :type ident: str
+    :type host: str
+    :type mask: str
+    """
+
+    def __init__(self, name, ident, host, mask):
+        self.name = name
+        self.ident = ident
+        self.host = host
+        self.mask = mask
+
+
+class Channel:
+    """
+    name: the name of this channel
+    users: A dict from nickname to User in this channel
+    user_modes: A dict from User to an str containing all of the user's modes in this channel
+    history: A list of (User, timestamp, message content)
+    :type name: str
+    :type users: dict[str, User]
+    :type user_modes: dict[User, str]
+    :type history: deque[(User, datetime, str)]
+
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self.users = {}
+        self.user_modes = {}
+        self.history = deque(maxlen=100)
+
+    def track_message(self):
+        """
+        Adds a message to this channel's history, adding user info from the message as well
+        """
+        pass
diff --git a/cloudbot/plugin.py b/cloudbot/plugin.py
index e6719b8..97cdf49 100644
--- a/cloudbot/plugin.py
+++ b/cloudbot/plugin.py
@@ -294,7 +294,14 @@ class PluginManager:
             logger.info("Loaded {}".format(hook))
             logger.debug("Loaded {}".format(repr(hook)))
 
-    def _prepare_parameters(self, hook, event):
+    def add_hook(self, hook_type, function, *args, **kwargs):
+        # this is a little hacky, but I think it's a good design choice. hook.py does the actual processing,
+        # so this works.
+        
+        pass
+
+    @staticmethod
+    def _prepare_parameters(hook, event):
         """
         Prepares arguments for the given hook
 
-- 
2.0.0

