From d40a291ecdb12427c91f1d4733c2275f73b04d68 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 21 Jun 2014 23:32:09 -0500
Subject: [PATCH] Fix raw file logging, and add config option to default config


diff --git a/config.default b/config.default
index 29a3b86..1635da9 100644
--- a/config.default
+++ b/config.default
@@ -72,6 +72,7 @@
     "logging": {
         "show_plugin_loading": true,
         "show_motd": true,
-        "show_server_info": true
+        "show_server_info": true,
+        "raw_file_log": false
     }
 }
diff --git a/plugins/log.py b/plugins/log.py
index b3f960c..0bd101d 100644
--- a/plugins/log.py
+++ b/plugins/log.py
@@ -174,7 +174,7 @@ def get_log_stream(server, chan):
 def get_raw_log_filename(server):
     current_time = time.gmtime()
     folder_name = time.strftime(folder_format, current_time)
-    file_name = time.strftime(file_format.format(server=server), current_time).lower()
+    file_name = time.strftime(raw_file_format.format(server=server), current_time).lower()
     return os.path.join(cloudbot.log_dir, "raw", folder_name, file_name)
 
 
-- 
2.0.0

