From 02f1fd2d9e5120b25551b9008fe1bd380010735e Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 21 Jun 2014 21:00:48 -0500
Subject: [PATCH] Working on a tracking system


diff --git a/cloudbot/client.py b/cloudbot/client.py
index 7909fd6..34ebb2e 100644
--- a/cloudbot/client.py
+++ b/cloudbot/client.py
@@ -1,4 +1,5 @@
 import asyncio
+from collections import deque
 import logging
 
 from cloudbot.permissions import PermissionManager
@@ -121,3 +122,49 @@ class Client:
     @property
     def connected(self):
         raise NotImplementedError
+
+
+# TODO: Tracking of user 'mode' in channels
+class User:
+    """
+    :param name: The nickname of this User
+    :param ident: The IRC ident of this User, if applicable
+    :param host: The hostname of this User, if applicable
+    :param mask: The IRC mask (nick!ident@host), if applicable
+    :type name: str
+    :type ident: str
+    :type host: str
+    :type mask: str
+    """
+
+    def __init__(self, name, ident, host, mask):
+        self.name = name
+        self.ident = ident
+        self.host = host
+        self.mask = mask
+
+
+class Channel:
+    """
+    name: the name of this channel
+    users: A dict from nickname to User in this channel
+    user_modes: A dict from User to an str containing all of the user's modes in this channel
+    history: A list of (User, timestamp, message content)
+    :type name: str
+    :type users: dict[str, User]
+    :type user_modes: dict[User, str]
+    :type history: deque[(User, datetime, str)]
+
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self.users = {}
+        self.user_modes = {}
+        self.history = deque(maxlen=100)
+
+    def track_message(self):
+        """
+        Adds a message to this channel's history, adding user info from the message as well
+        """
+        pass
-- 
2.0.0

