From 9a8b82bd95e38f743a78f53d762e79fc604b5fae Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 19:09:15 -0800
Subject: [PATCH] Don't run regex matching if command is already matched


diff --git a/core/main.py b/core/main.py
index a856b35..eed61a3 100644
--- a/core/main.py
+++ b/core/main.py
@@ -184,6 +184,7 @@ def main(conn, out):
 
                 func, args = bot.commands[command]
                 dispatch(input, "command", func, args, autohelp=True)
+                return
 
         # REGEXES
         for func, args in bot.plugs['regex']:
-- 
1.9.1

