From 0e4dbf86c0b53e879a67ab589ab401c13f884b6c Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 19:15:02 -0800
Subject: [PATCH] Add .message command to raw message someone


diff --git a/plugins/admin.py b/plugins/admin.py
index f434a1a..c5d1334 100644
--- a/plugins/admin.py
+++ b/plugins/admin.py
@@ -209,6 +209,16 @@ def say(inp, conn=None, chan=None):
     conn.send(out)
 
 
+@hook.command(permissions=["botcontrol"])
+def message(inp, conn=None, chan=None):
+    """message <name> <message> -- Makes the bot say <message> to <name>.
+    If <name> is a channel, the bot will say the message in that channel."""
+    inp = inp.split(" ")
+    message = " ".join(inp[1:])
+    out = "PRIVMSG {} :{}".format(inp[0], message)
+    conn.send(out)
+
+
 @hook.command("act", permissions=["botcontrol"])
 @hook.command(permissions=["botcontrol"])
 def me(inp, conn=None, chan=None):
-- 
1.9.2

