From 2f66456a92c2735d824f6df9b0afc0149765794c Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 27 Nov 2013 03:11:37 -0800
Subject: [PATCH] Add sayto command to say to user/channel


diff --git a/plugins/admin.py b/plugins/admin.py
index 987e1b4..0aec0a2 100644
--- a/plugins/admin.py
+++ b/plugins/admin.py
@@ -207,6 +207,17 @@ def say(inp, conn=None, chan=None):
     conn.send(out)
 
 
+@hook.command(permissions=["botcontrol"])
+def sayto(inp, conn=None, chan=None):
+    """sayto channel <message> -- Makes the bot say <message> in [channel].
+    If [channel] is blank the bot will say the <message> in the channel
+    the command was used in."""
+    inp = inp.split(" ")
+    message = " ".join(inp[1:])
+    out = "PRIVMSG {} :{}".format(inp[0], message)
+    conn.send(out)
+
+
 @hook.command("act", permissions=["botcontrol"])
 @hook.command(permissions=["botcontrol"])
 def me(inp, conn=None, chan=None):
-- 
1.8.3.2

