From d2302c943fbc62decbc7513edbf39dda84e6a2bf Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 19:10:43 -0800
Subject: [PATCH] Add support for multiple channels to .join and .leave


diff --git a/plugins/admin.py b/plugins/admin.py
index 39bbce1..ede2e3f 100644
--- a/plugins/admin.py
+++ b/plugins/admin.py
@@ -139,8 +139,11 @@ def clearlogs(inp, input=None):
 @hook.command(permissions=["botcontrol"])
 def join(inp, conn=None, notice=None):
     """join <channel> -- Joins <channel>."""
-    notice("Attempting to join {}...".format(inp))
-    conn.join(inp)
+    for target in inp.split(" "):
+        if not target.startswith("#"):
+            target = "#{}".format(target)
+        notice("Attempting to join {}...".format(target))
+        conn.join(target)
 
 
 @hook.command(autohelp=False, permissions=["botcontrol"])
@@ -149,11 +152,14 @@ def part(inp, conn=None, chan=None, notice=None):
     If [channel] is blank the bot will leave the
     channel the command was used in."""
     if inp:
-        target = inp
+        targets = inp
     else:
-        target = chan
-    notice("Attempting to leave {}...".format(target))
-    conn.part(target)
+        targets = chan
+    for target in targets.split(" "):
+        if not target.startswith("#"):
+            target = "#{}".format(target)
+        notice("Attempting to leave {}...".format(target))
+        conn.part(target)
 
 
 @hook.command(autohelp=False, permissions=["botcontrol"])
-- 
1.8.3.2

