From b6b152b37b733d4542d14b84c45e92123cb5271f Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:05:12 -0800
Subject: [PATCH] Update minecraft_status to use unicode checks, and with a new
 format.


diff --git a/plugins/minecraft_status.py b/plugins/minecraft_status.py
index 4ca67d3..d9e416d 100644
--- a/plugins/minecraft_status.py
+++ b/plugins/minecraft_status.py
@@ -2,8 +2,16 @@ import json
 
 from util import hook, http
 
+green_prefix = u"\x02\x0f"
+green_suffix = u": \x033\x02\u2714"
+yellow_prefix = u"\x02\x0f"
+yellow_suffix = u": \x037\x02\u26A0"
+red_prefix = u"\x02\x0f"
+red_suffix = u": \x034\x02\u2716"
+
 
 @hook.command(autohelp=False)
+@hook.command("mcs", autohelp=False)
 def mcstatus(inp):
     """mcstatus -- Checks the status of various Mojang (the creators of Minecraft) servers."""
 
@@ -15,30 +23,20 @@ def mcstatus(inp):
     # lets just reformat this data to get in a nice format
     data = json.loads(request.replace("}", "").replace("{", "").replace("]", "}").replace("[", "{"))
 
-    out = []
-
     # use a loop so we don't have to update it if they add more servers
-    green = []
-    yellow = []
-    red = []
+    servers = []
     for server, status in data.items():
+        if server == "minecraft.net":
+            server = "MC|Website"
+        elif server.endswith(".mojang.com"):
+            server = "MJ|{}".format(server[:-11].capitalize())
+        elif server.endswith(".minecraft.net"):
+            server = "MC|{}".format(server[:-14].capitalize())
+
         if status == "green":
-            green.append(server)
+            servers.append(u"{}{}{}".format(green_prefix, server, green_suffix))
         elif status == "yellow":
-            yellow.append(server)
+            servers.append(u"{}{}{}".format(yellow_prefix, server, yellow_suffix))
         else:
-            red.append(server)
-
-    if green:
-        out = "\x033\x02Online\x02\x0f: " + ", ".join(green)
-        if yellow:
-            out += " "
-    if yellow:
-        out += "\x02Issues\x02: " + ", ".join(yellow)
-        if red:
-            out += " "
-    if red:
-        out += "\x034\x02Offline\x02\x0f: " + ", ".join(red)
-
-    return "\x0f" + out.replace(".mojang.com", ".mj") \
-                       .replace(".minecraft.net", ".mc")
+            servers.append(u"{}{}{}".format(red_prefix, server, red_suffix))
+    return "  ".join(servers)
-- 
1.9.2

