From b80d9dd162649efc8abeca293dd72de535ba86dd Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 27 Nov 2013 03:14:03 -0800
Subject: [PATCH] Change ctcp replies for Eliralin


diff --git a/plugins/core_ctcp.py b/plugins/core_ctcp.py
index 4f399f6..5d54254 100644
--- a/plugins/core_ctcp.py
+++ b/plugins/core_ctcp.py
@@ -5,7 +5,7 @@ from util import hook
 # CTCP responses
 @hook.regex(r'^\x01VERSION\x01$')
 def ctcp_version(inp, notice=None):
-    notice('\x01VERSION: CloudBot - http://git.io/cloudbotirc')
+    notice('\x01VERSION Eliralin 1.0')
 
 
 @hook.regex(r'^\x01PING\x01$')
@@ -14,5 +14,5 @@ def ctcp_ping(inp, notice=None):
 
 
 @hook.regex(r'^\x01TIME\x01$')
-def ctcp_time(inp, notice=None):
-    notice('\x01TIME: The time is: {}'.format(time.strftime("%r", time.localtime())))
+def ctcp_time(inp, nick=None, notice=None):
+    notice('\x01TIME: Hey {}, the time is {}'.format(nick, time.strftime("%r", time.localtime())))
-- 
1.8.3.2

