From 58f1d2043578796b453abbe52c49d8e3b442f63c Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 22:55:24 -0800
Subject: [PATCH] Use permissions instead of adminonly


diff --git a/plugins/plpaste.py b/plugins/plpaste.py
index 238037d..bf039fc 100644
--- a/plugins/plpaste.py
+++ b/plugins/plpaste.py
@@ -1,7 +1,7 @@
 from util import hook, web
 
 
-@hook.command(adminonly=True)
+@hook.command(permissions=["adminonly"])
 def plpaste(inp):
     if "/" in inp and inp.split("/")[0] != "util":
         return "Invalid input"
-- 
1.7.10.4

