From 630a1e6c41d169eaeecce78cf536bfb28e415780 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:11:02 -0800
Subject: [PATCH] Improve self-attack matching.


diff --git a/plugins/attacks.py b/plugins/attacks.py
index feb00b8..1fe7fc2 100644
--- a/plugins/attacks.py
+++ b/plugins/attacks.py
@@ -1,4 +1,5 @@
 import random
+import re
 
 from util import hook
 
@@ -26,7 +27,7 @@ def lart(inp, action=None, nick=None, conn=None, notice=None):
         return
 
     # if the user is trying to make the bot slap itself, slap them
-    if target.lower() == conn.nick.lower() or target.lower() == "itself":
+    if re.search("(itself|herself|himself|{}).*".format(conn.nick.lower()), target.lower()):
         target = nick
 
     values = {"user": target}
@@ -45,7 +46,7 @@ def insult(inp, nick=None, action=None, conn=None, notice=None):
         notice("Invalid username!")
         return
 
-    if target == conn.nick.lower() or target == "itself":
+    if re.search("(itself|herself|himself|{}).*".format(conn.nick.lower()), target.lower()):
         target = nick
     else:
         target = inp
@@ -55,7 +56,7 @@ def insult(inp, nick=None, action=None, conn=None, notice=None):
 
 
 @hook.command
-def flirt(inp, action=None, conn=None, notice=None):
+def flirt(inp, nick=None, message=None, conn=None, notice=None):
     """flirt <user> -- Make the bot flirt with <user>."""
     target = inp.strip()
 
@@ -63,10 +64,9 @@ def flirt(inp, action=None, conn=None, notice=None):
         notice("Invalid username!")
         return
 
-    if target == conn.nick.lower() or target == "itself":
-        target = 'itself'
+    if re.search("(itself|herself|himself|{}).*".format(conn.nick.lower()), target.lower()):
+        target = nick
     else:
         target = inp
 
-    out = 'flirts with {}... "{}"'.format(target, random.choice(flirts))
-    action(out)
+    message('{}, {}'.format(target, random.choice(flirts)))
-- 
1.8.5.3

