From c023e711d875128eca013c36a20a68dbc924ff78 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:13:03 -0800
Subject: [PATCH] Change ctcp reply to return 'Eliralin'


diff --git a/plugins/core_ctcp.py b/plugins/core_ctcp.py
index 3e7a200..73af9bd 100644
--- a/plugins/core_ctcp.py
+++ b/plugins/core_ctcp.py
@@ -6,7 +6,7 @@ from util import hook
 # CTCP responses
 @hook.regex(r'^\x01VERSION\x01$')
 def ctcp_version(inp, notice=None):
-    notice('\x01VERSION: CloudBot - http://git.io/cloudbotirc')
+    notice('\x01VERSION Eliralin 1.0')
 
 
 @hook.regex(r'^\x01PING\x01$')
-- 
1.9.1

