From d28310423d939282f4165c5a644653d031dd217d Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:11:02 -0800
Subject: [PATCH] Improve self-attack matching.


diff --git a/plugins/attacks.py b/plugins/attacks.py
index 643a184..6746f11 100644
--- a/plugins/attacks.py
+++ b/plugins/attacks.py
@@ -1,5 +1,6 @@
 from util import hook
 import random
+import re
 
 with open("plugins/data/larts.txt") as f:
     larts = [line.strip() for line in f.readlines()
@@ -24,7 +25,7 @@ def lart(inp, action=None, nick=None, conn=None, notice=None):
         return
 
     # if the user is trying to make the bot slap itself, slap them
-    if target.lower() == conn.nick.lower() or target.lower() == "itself":
+    if re.search("(itself|herself|himself|{}).*".format(conn.nick.lower()), target.lower()):
         target = nick
 
     values = {"user": target}
@@ -43,7 +44,7 @@ def insult(inp, nick=None, action=None, conn=None, notice=None):
         notice("Invalid username!")
         return
 
-    if target == conn.nick.lower() or target == "itself":
+    if re.search("(itself|herself|himself|{}).*".format(conn.nick.lower()), target.lower()):
         target = nick
     else:
         target = inp
@@ -53,7 +54,7 @@ def insult(inp, nick=None, action=None, conn=None, notice=None):
 
 
 @hook.command
-def flirt(inp, action=None, conn=None, notice=None):
+def flirt(inp, nick=None, message=None, conn=None, notice=None):
     """flirt <user> -- Make the bot flirt with <user>."""
     target = inp.strip()
 
@@ -61,10 +62,9 @@ def flirt(inp, action=None, conn=None, notice=None):
         notice("Invalid username!")
         return
 
-    if target == conn.nick.lower() or target == "itself":
-        target = 'itself'
+    if re.search("(itself|herself|himself|{}).*".format(conn.nick.lower()), target.lower()):
+        target = nick
     else:
         target = inp
 
-    out = 'flirts with {}... "{}"'.format(target, random.choice(flirts))
-    action(out)
+    message('{}, {}'.format(target, random.choice(flirts)))
-- 
1.8.5.3

