From dbe817eb620d9b926f227a3c739b915949293f7a Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:24:30 -0800
Subject: [PATCH] Add 'listfactoids' command to list all factoids


diff --git a/plugins/factoids.py b/plugins/factoids.py
index 457a4b7..9864f87 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -158,3 +158,17 @@ def factoid(inp, message=None, idb=None, bot=None, action=None, conn=None, input
                 message("\x02[{}]:\x02 {}".format(factoid_id, result))
             else:
                 message(result)
+
+@hook.command(autoHelp=False, permissions=["adminonly"])
+def listfactoids(inp, idb=None, reply=None):
+    db_init(idb)
+    text = False
+    for word in idb.execute("select word from mem").fetchall():
+        if not text:
+            text = word[0]
+        else:
+            text += ", {}".format(word[0])
+        if len(text) > 400:
+            reply(text.rsplit(', ', 1)[0])
+            text = word[0]
+    return text
-- 
1.9.0

