From 4c3acc09eba3257a5b233dbc2764e9f062348664 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:13:03 -0800
Subject: [PATCH] Change ctcp reply to return 'Eliralin'


diff --git a/plugins/core_ctcp.py b/plugins/core_ctcp.py
index 4f399f6..e7fdc60 100644
--- a/plugins/core_ctcp.py
+++ b/plugins/core_ctcp.py
@@ -5,7 +5,7 @@ from util import hook
 # CTCP responses
 @hook.regex(r'^\x01VERSION\x01$')
 def ctcp_version(inp, notice=None):
-    notice('\x01VERSION: CloudBot - http://git.io/cloudbotirc')
+    notice('\x01VERSION Eliralin 1.0')
 
 
 @hook.regex(r'^\x01PING\x01$')
-- 
1.8.3.2

