From 925fa5db6aa6def077596ea6fccd809a630f5e87 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:18:52 -0800
Subject: [PATCH] Add 'steal' command.


diff --git a/plugins/steal.py b/plugins/steal.py
new file mode 100644
index 0000000..78ccddb
--- /dev/null
+++ b/plugins/steal.py
@@ -0,0 +1,62 @@
+from util import hook
+from random import randint
+
+
+def db_init(db):
+    db.execute("CREATE TABLE IF NOT EXISTS stolen(word, PRIMARY KEY (word))")
+    db.commit()
+
+
+def get_random(db):
+    count = db.execute("SELECT COUNT(word) FROM stolen").fetchone()[0]
+    row = db.execute("SELECT word FROM stolen LIMIT 1 OFFSET ?", [randint(0, int(count-1))]).fetchone()
+    if row:
+        return row[0]
+    else:
+        return None
+
+
+def list_steals(db):
+    row = db.execute("SELECT word FROM stolen").fetchall()
+    return row
+
+
+def add_word(db, stolen):
+    db.execute("REPLACE INTO stolen (word) VALUES(?)", [stolen])
+    db.commit()
+
+
+@hook.command(autoHelp=False)
+@hook.command("steal", autoHelp=False)
+def stealit(inp, idb=None, nick=None, action=None):
+    """steal [[object] username]- Steals an object from a user or randomly chooses a previously stolen object if none is given"""
+    db_init(idb)
+    args = inp.strip().split()
+    if not args:
+        steal_from = nick
+        to_steal = get_random(idb)
+        action("steals {}'s {}".format(steal_from, to_steal))
+    elif len(args) < 2:
+        steal_from = args[0]
+        to_steal = get_random(idb)
+        action("steals {}'s {}".format(steal_from, to_steal))
+    else:
+        steal_from = args[len(args) - 1]
+        to_steal = " ".join(args[0:-1])
+        action("steals {}'s {}".format(steal_from, to_steal))
+        add_word(idb, to_steal)
+
+
+@hook.command(autoHelp=False, permissions=["adminonly"])
+def liststolen(inp, idb=None, reply=None):
+    db_init(idb)
+    text = False
+    for word in list_steals(idb):
+        if not text:
+            text = word[0]
+        else:
+            text += ", {}".format(word[0])
+        if len(text) > 400:
+            reply(text.rsplit(', ', 1)[0])
+            text = word[0]
+    return text
-- 
1.8.3.2

