From e9c6d86ed3028f3ed44cc6386ed11236674858f9 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 27 Nov 2013 03:26:35 -0800
Subject: [PATCH] Fix twitch - credit to blha303


diff --git a/plugins/twitch.py b/plugins/twitch.py
index 842057d..357431f 100644
--- a/plugins/twitch.py
+++ b/plugins/twitch.py
@@ -92,17 +92,20 @@ def twitch_lookup(location):
             views = views + "s" if not views[0:2] == "1 " else views
             return h.unescape(fmt.format(title, channel, playing, views))
     else:
-        data = http.get_json("http://api.justin.tv/api/stream/list.json?channel=" + channel)[0]
-        if data:
-            title = data['title']
-            playing = data['meta_game']
-            viewers = "\x033\x02Online now!\x02\x0f " + str(data["channel_count"]) + " viewer"
+        data = http.get_json("http://api.justin.tv/api/stream/list.json?channel=" + channel)
+        if data and len(data) >= 1:
+            title = data[0]['title']
+            playing = data[0]['meta_game']
+            viewers = "\x033\x02Online now!\x02\x0f " + str(data[0]["channel_count"]) + " viewer"
             print viewers
             viewers = viewers + "s" if not " 1 view" in viewers else viewers
             print viewers
             return h.unescape(fmt.format(title, channel, playing, viewers))
         else:
-            data = http.get_json("https://api.twitch.tv/kraken/channels/" + channel)
+            try:
+                data = http.get_json("https://api.twitch.tv/kraken/channels/" + channel)
+            except:
+                pass
             title = data['status']
             playing = data['game']
             viewers = "\x034\x02Offline\x02\x0f"
-- 
1.8.3.2

