From 67acedbbacb542625c225a886ea66d8e4ffa9a72 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:24:30 -0800
Subject: [PATCH] Add 'listfactoids' command to list all factoids


diff --git a/plugins/factoids.py b/plugins/factoids.py
index 9fb41fe..74590a1 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -151,3 +151,17 @@ def factoid(inp, message=None, idb=None, bot=None, action=None, conn=None, input
                 message("\x02[{}]:\x02 {}".format(factoid_id, result))
             else:
                 message(result)
+
+@hook.command(autoHelp=False, permissions=["adminonly"])
+def listfactoids(inp, idb=None, reply=None):
+    db_init(idb)
+    text = False
+    for word in idb.execute("select word from mem").fetchall():
+        if not text:
+            text = word[0]
+        else:
+            text += ", {}".format(word[0])
+        if len(text) > 400:
+            reply(text.rsplit(', ', 1)[0])
+            text = word[0]
+    return text
-- 
1.8.3.2

