From 0c27d5431c068a5b92a2f1cc1188de9da0826765 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:25:28 -0800
Subject: [PATCH] Replace ? with ^ for factoids


diff --git a/plugins/factoids.py b/plugins/factoids.py
index 0859d80..4385b8f 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -111,7 +111,7 @@ def info(inp, notice=None, idb=None):
         notice("Unknown Factoid.")
 
 
-@hook.regex(r'^\? ?(.+)')
+@hook.regex(r'^\^ ?(.+)')
 def factoid(inp, message=None, idb=None, bot=None, action=None, conn=None, input=None):
     """?<word> -- Shows what data is associated with <word>."""
     try:
-- 
1.9.1

