From 2ed123bb411add5113ad6c53c6518f4ad6c8a785 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 27 Nov 2013 03:29:25 -0800
Subject: [PATCH] Create josephus plugin


diff --git a/plugins/josephus.py b/plugins/josephus.py
new file mode 100644
index 0000000..c5ab8ee
--- /dev/null
+++ b/plugins/josephus.py
@@ -0,0 +1,26 @@
+from util import hook
+
+def josephus_calc(size, every_x, current):
+    alive, till_kill = [True] * size, 0
+    while True:
+        if alive[current]:
+            if sum(alive) == 1:
+                break
+            elif till_kill == 0:
+                alive[current] = False
+                till_kill = every_x - 1
+            else:
+                till_kill -= 1
+        current += 1 if current < size - 1 else 1 - size
+    return "Josephus should be at position {} to survive.".format(current)
+
+
+@hook.command()
+@hook.command("jose")
+def josephus(inp):
+    """jose [size] [every x] [starting person] - Calculates who dies last """
+    split = inp.split()
+    if len(split) != 3:
+        return "Not enough / too many arguments. {}".format(len(split))
+    size, every_x, current = [int(x) for x in split]
+    return josephus_calc(size, every_x, current)
-- 
1.8.3.2

