From f1c15e5ae6bdd78c6329afb4644041d4449ffd9f Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:25:28 -0800
Subject: [PATCH] Replace ? with ^ for factoids


diff --git a/plugins/factoids.py b/plugins/factoids.py
index 9864f87..aac0e6a 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -71,10 +71,10 @@ def remember(inp, nick='', idb=None, notice=None):
         if append:
             notice("Appending \x02{}\x02 to \x02{}\x02".format(new_data, old_data))
         else:
-            notice('Remembering \x02{}\x02 for \x02{}\x02. Type ?{} to see it.'.format(data, word, word))
+            notice('Remembering \x02{}\x02 for \x02{}\x02. Type ^{} to see it.'.format(data, word, word))
             notice('Previous data was \x02{}\x02'.format(old_data))
     else:
-        notice('Remembering \x02{}\x02 for \x02{}\x02. Type ?{} to see it.'.format(data, word, word))
+        notice('Remembering \x02{}\x02 for \x02{}\x02. Type ^{} to see it.'.format(data, word, word))
 
 
 @hook.command("f", permissions=["delfactoid"])
@@ -109,7 +109,7 @@ def info(inp, notice=None, idb=None):
         notice("Unknown Factoid.")
 
 
-@hook.regex(r'^\? ?(.+)')
+@hook.regex(r'^\^ ?(.+)')
 def factoid(inp, message=None, idb=None, bot=None, action=None, conn=None, input=None):
     """?<word> -- Shows what data is associated with <word>."""
     try:
-- 
1.8.3.2

