From d61117b66c24d1d86aa416bc3351b528863581b3 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 30 Nov 2013 01:38:11 +0100
Subject: [PATCH] Update .gitignore


diff --git a/.gitignore b/.gitignore
index 20e8101..8f9baef 100644
--- a/.gitignore
+++ b/.gitignore
@@ -12,3 +12,4 @@ gitflow
 *.sublime-workspace
 .idea/
 plugins/data/GeoLiteCity.dat
+plugins/data/bukgetplugins
-- 
1.8.3.2

