From 0bfec67c7449a9fc8d76a45fecbef8d12498331d Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 27 Jan 2014 03:59:51 +0100
Subject: [PATCH] Add per-channel command prefixes


diff --git a/core/db.py b/core/db.py
index 6bdf8fa..59b317c 100644
--- a/core/db.py
+++ b/core/db.py
@@ -1,6 +1,7 @@
 import os
 import sqlite3
 import thread
+import re
 
 threaddbs = {}
 
@@ -23,4 +24,16 @@ def get_db_connection(conn, name=''):
         threaddbs[name] = {threadid: db}
     return db
 
+
+def get_command_prefixes(inp):
+    db = get_db_connection(inp.conn)
+    db.execute("CREATE TABLE IF NOT EXISTS prefixes(channel, prefix, UNIQUE(channel, prefix) ON CONFLICT REPLACE)")
+    db.commit()
+    row = db.execute("SELECT prefix FROM prefixes WHERE channel = ?", [inp.chan]).fetchall()
+    prefixes = ""
+    for prefix in row:
+        prefixes += re.escape(prefix[0])
+    return prefixes
+
+
 bot.get_db_connection = get_db_connection
diff --git a/core/main.py b/core/main.py
index 3c065e4..6d373ad 100644
--- a/core/main.py
+++ b/core/main.py
@@ -161,11 +161,11 @@ def main(conn, out):
     if inp.command == 'PRIVMSG':
         # COMMANDS
         if inp.chan == inp.nick:  # private message, no command prefix
-            prefix = '^(?:[{}]?|'.format(command_prefix)
+            command_re = '^(?:[{}{}]?|'.format(command_prefix, get_command_prefixes(inp))
         else:
-            prefix = '^(?:[{}]|'.format(command_prefix)
+            command_re = '^(?:[{}{}]|'.format(command_prefix, get_command_prefixes(inp))
 
-        command_re = prefix + inp.conn.nick
+        command_re += inp.conn.nick
         command_re += r'[,;:]+\s+)(\w+)(?:$|\s+)(.*)'
 
         m = re.match(command_re, inp.lastparam)
diff --git a/plugins/eliralin_utility.py b/plugins/eliralin_utility.py
index 92c5c56..c37a455 100644
--- a/plugins/eliralin_utility.py
+++ b/plugins/eliralin_utility.py
@@ -146,4 +146,12 @@ def unicodecommand(inp, reply=None):
         return u"'{}'".format(unichr(int(inp)))
     except Exception:
         reply("Failed")
-        raise
\ No newline at end of file
+        raise
+
+
+@hook.command(permissions=["botcontrol"])
+def addprefix(inp, chan=None, db=None):
+    """addprefix [prefix] - Adds a one-letter prefix to the bot in this channel"""
+    db.execute("CREATE TABLE IF NOT EXISTS prefixes(channel, prefix, UNIQUE(channel, prefix) ON CONFLICT REPLACE)")
+    db.execute("INSERT INTO prefixes(channel, prefix) VALUES(?, ?)", [chan, inp])
+    db.commit()
-- 
1.9.1

