From c2022d472f4b593507239267771d476bb3a8c78d Mon Sep 17 00:00:00 2001
From: Steven Smith <stevensmith.ome@gmail.com>
Date: Mon, 24 Mar 2014 11:01:46 +0800
Subject: [PATCH] Add support for plugins without commands

regex plugins, etc

diff --git a/plugins/plpaste.py b/plugins/plpaste.py
index bf039fc..d46e672 100644
--- a/plugins/plpaste.py
+++ b/plugins/plpaste.py
@@ -1,12 +1,14 @@
 from util import hook, web
+from os import listdir
 
 
 @hook.command(permissions=["adminonly"])
-def plpaste(inp):
-    if "/" in inp and inp.split("/")[0] != "util":
-        return "Invalid input"
-    try:
-        with open("plugins/%s.py" % inp) as f:
+def plpaste(inp, bot=None):
+    if inp in bot.commands:
+        with open(bot.commands[inp][0].func_code.co_filename.strip()) as f:
             return web.haste(f.read(), ext='py')
-    except IOError:
-        return "Plugin not found (must be in plugins folder)"
+    elif inp + ".py" in listdir('plugins/'):
+        with open('plugins/{}.py'.format(inp)) as f:
+            return web.haste(f.read(), ext='py')
+    else:
+        return "Could not find specified plugin."
-- 
1.9.2

