From 8aecba1fd42f1402e29ab61dcb9cd6ecc7ff028f Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 18 Feb 2014 17:23:21 -0800
Subject: [PATCH] Fix minecraft_ping for non-dict descriptions


diff --git a/plugins/minecraft_ping.py b/plugins/minecraft_ping.py
index ba806c8..a2c2944 100644
--- a/plugins/minecraft_ping.py
+++ b/plugins/minecraft_ping.py
@@ -2,6 +2,8 @@
 import socket
 import struct
 import json
+from pprint import pprint
+import traceback
 
 from util import hook
 
@@ -13,7 +15,6 @@ try:
 except ImportError:
     pydns_installed = False
 
-
 mccolors = [u"\x0300,\xa7f", u"\x0301,\xa70", u"\x0302,\xa71", u"\x0303,\xa72", u"\x0304,\xa7c", u"\x0305,\xa74",
             u"\x0306,\xa75", u"\x0307,\xa76", u"\x0308,\xa7e", u"\x0309,\xa7a", u"\x0310,\xa73", u"\x0311,\xa7b",
             u"\x0312,\xa71", u"\x0313,\xa7d", u"\x0314,\xa78", u"\x0315,\xa77", u"\x02,\xa7l", u"\x0310,\xa79",
@@ -76,15 +77,15 @@ def mcping_modern(host, port):
     data = json.loads(d.decode('utf8'))
     try:
         version = data["version"]["name"]
-        if data["description"].get("text", None):
+        if isinstance(data["description"], dict) and data["description"].get("text", None):
             desc = u" ".join(data["description"]["text"].split())
         else:
             desc = u" ".join(data["description"].split())
         max_players = data["players"]["max"]
         online = data["players"]["online"]
     except Exception as e:
-        from pprint import pprint
         pprint(data)
+        traceback.print_exc(e)
         return "Invalid data - check console ({})".format(e)
     return mc_color_format(u"{}\x0f - {}\x0f - {}/{} players.".format(desc, version, online,
                                                                       max_players)).replace("\n", u"\x0f - ")
@@ -150,7 +151,7 @@ def mcping6(inp):
     #try:
     host, port = parse_input(inp)
     #except Exception as ex:
-     #   return ex.args[0]
+    #    return ex.args[0]
     try:
         return mcping_legacy(host, port)
     except:
@@ -175,11 +176,10 @@ def mcping7(inp):
 @hook.command("mcp")
 def mcping(inp):
     """mcping <server>[:port] - Ping a Minecraft server to check status."""
-  #  try:
+    #try:
     host, port = parse_input(inp)
     #except Exception as e:
-     #   return e.args[0]
-#
+    #    return e.args[0]
 
     try:
         return mcping_modern(host, port)
-- 
1.8.3.2

