From afdd0c81528ea16ef4aa549960a66fb17f70e173 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 1 Dec 2013 22:16:54 +0100
Subject: [PATCH] Eliralin utility


diff --git a/plugins/eliralin_utility.py b/plugins/eliralin_utility.py
new file mode 100644
index 0000000..c751cac
--- /dev/null
+++ b/plugins/eliralin_utility.py
@@ -0,0 +1,6 @@
+from util import hook
+
+
+@hook.command("print")
+def print_thing(inp, nick=None):
+    print "Print issued by {}: '{}'".format(nick, inp)
-- 
1.8.3.2

