From c0004928d83699af15efb15ee9d7c138bf8cc765 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 1 Apr 2014 21:03:51 -0700
Subject: [PATCH] Add more phrases


diff --git a/plugins/recipe.py b/plugins/recipe.py
index 0e04572..ca1d83b 100644
--- a/plugins/recipe.py
+++ b/plugins/recipe.py
@@ -17,6 +17,11 @@ phrases = [
     u"WHY DON'T YOU EAT SOME FUCKING \x02{}?\x02",
     u"MAKE SOME FUCKING \x02{}\x02",
     u"INDUCE FOOD COMA WITH SOME FUCKING \x02{}\x02"
+    u"CLASSILY PARTAKE IN SOME FUCKING \x02{}\x02",
+    u"COOK UP SOME FUCKING \x02{}\x02",
+    u"CURE YOUR MOUTH'S POST TRAUMATIC STRESS DISORDER WITH SOME FUCKING \x02{}\x02",
+    u"PROCURE SOME CHILD LABOR TO COOK UP SOME FUCKING \x02{}\x02",
+    u"YOUR INDECISION IS FAR LESS APPETIZING THAN SOME FUCKING \x02{}\x02"
 ]
 
 clean_key = lambda i: i.split("#")[1]
-- 
1.9.2

