From 16400e0dd0764d2b9d9df6da3ab1d2d7fc0bcc4d Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 24 Mar 2014 20:46:47 -0700
Subject: [PATCH] Fix unicode support for wikipedia


diff --git a/plugins/wikipedia.py b/plugins/wikipedia.py
index 6b3827a..a60e1d9 100644
--- a/plugins/wikipedia.py
+++ b/plugins/wikipedia.py
@@ -29,7 +29,7 @@ def wiki(inp):
             return 'No results found.'
 
     def extract(item):
-        return [item.find(ns + x).text for x in
+        return [item.find(ns + x2).text for x2 in
                 ('Text', 'Description', 'Url')]
 
     title, desc, url = extract(items[0])
@@ -42,8 +42,8 @@ def wiki(inp):
     if title.lower() not in desc.lower():
         desc = title + desc
 
-    desc = re.sub('\s+', ' ', desc).strip()  # remove excess spaces
+    desc = u' '.join(desc.split())  # remove excess spaces
 
     desc = text.truncate_str(desc, 200)
 
-    return '{} :: {}'.format(desc, http.quote(url, ':/'))
+    return u'{} :: {}'.format(desc, http.quote(url, ':/'))
-- 
1.9.1

