From d0502d989224e9b6124232f771cf4f2ad5c266ea Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 1 Dec 2013 22:17:15 +0100
Subject: [PATCH] Colors utility


diff --git a/plugins/colors.py b/plugins/colors.py
new file mode 100644
index 0000000..f7ed937
--- /dev/null
+++ b/plugins/colors.py
@@ -0,0 +1,16 @@
+from util import hook
+
+
+@hook.command
+def colors(inp):
+    if inp:
+        intinp = int(inp)
+        if intinp > 70:
+            return "Please use a number smaller than or equal to 70"
+        forrange = range(intinp)
+    else:
+        forrange = range(30)
+    result = ""
+    for i in forrange if inp else range(30):
+        result += u"\x03{0:02d} {0}".format(i)
+    return result
-- 
1.8.3.2

