From 9fcffc9adc7bb34449204152d5252ceec9f88aaf Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Thu, 27 Mar 2014 09:23:25 -0700
Subject: [PATCH] Fix unicode support for dictionary.py


diff --git a/plugins/dictionary.py b/plugins/dictionary.py
index b13db4b..faa421a 100644
--- a/plugins/dictionary.py
+++ b/plugins/dictionary.py
@@ -19,10 +19,10 @@ def define(inp):
                          '//div[@class="example"]')
 
     if not definition:
-        return 'No results for ' + inp + ' :('
+        return u'No results for {} :('.format(inp)
 
     def format_output(show_examples):
-        result = '{}: '.format(h.xpath('//dt[@class="title-word"]/a/text()')[0])
+        result = u'{}: '.format(h.xpath('//dt[@class="title-word"]/a/text()')[0])
 
         correction = h.xpath('//span[@class="correct-word"]/text()')
         if correction:
@@ -70,18 +70,18 @@ def define(inp):
 def etymology(inp):
     """etymology <word> -- Retrieves the etymology of <word>."""
 
-    url = 'http://www.etymonline.com/index.php'
+    url = u'http://www.etymonline.com/index.php'
 
     h = http.get_html(url, term=inp)
 
     etym = h.xpath('//dl')
 
     if not etym:
-        return 'No etymology found for {} :('.format(inp)
+        return u'No etymology found for {} :('.format(inp)
 
     etym = etym[0].text_content()
 
-    etym = ' '.join(etym.split())
+    etym = u' '.join(etym.split())
 
     if len(etym) > 400:
         etym = etym[:etym.rfind(' ', 0, 400)] + ' ...'
-- 
1.9.1

