From 37446208bad64117323c6d5db092f073de0c129d Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 2 Dec 2013 10:08:10 +0100
Subject: [PATCH] Create regex_chans.py for limiting regex


diff --git a/plugins/regex_chans.py b/plugins/regex_chans.py
new file mode 100644
index 0000000..cd2e2ba
--- /dev/null
+++ b/plugins/regex_chans.py
@@ -0,0 +1,20 @@
+from util import hook
+
+regex_chans = ["#aem", "#simplicitemc", "#ltttalk", "#shadowboxnetwork",
+                "#newlanders", "#daboross", "#skywars", "#eliralin",
+                "#ten.java", "#skybirdsoar", "#jamiesinn", "#olive"]
+
+
+print "Regex allowed chans: {}".format(regex_chans)
+
+
+@hook.sieve
+def sieve_regex(bot, input, func, kind, args):
+    if kind is 'regex':
+        if input.chan in regex_chans or not input.chan.startswith("#"):
+            print "Allowing input.raw={}, kind={}, args={} from {}".format(input.raw, kind, args, input.chan)
+        else:
+            print "Denying input.raw={}, kind={}, args={} from {}".format(input.raw, kind, args, input.chan)
+            return None
+
+    return input
-- 
1.8.3.2

