From b42460b271116d60b1eaee211be09997d9086063 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 3 Dec 2013 10:40:32 +0100
Subject: [PATCH] Make more sense possibly?


diff --git a/core/main.py b/core/main.py
index b1886d0..7e32995 100644
--- a/core/main.py
+++ b/core/main.py
@@ -12,6 +12,7 @@ class Input(dict):
         chan = paraml[0].lower()
         if chan == conn.nick.lower():  # is a PM
             chan = nick
+        mnick = text.munge(nick)
 
         def message(message, target=chan):
             """sends a message to a specific or current channel/user"""
@@ -22,7 +23,7 @@ class Input(dict):
             if target == nick:
                 conn.msg(target, message)
             else:
-                conn.msg(target, u"{}, {}".format(text.munge(nick), message))
+                conn.msg(target, u"{}, {}".format(mnick, message))
 
         def action(message, target=chan):
             """sends an action to the current channel/user or a specific channel/user"""
-- 
1.8.3.2

