From 6dbcf735bac563301cd1efb1c7fe466102d4aec4 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 10 Dec 2013 19:54:10 +0100
Subject: [PATCH] stolen not STOLEN


diff --git a/plugins/steal.py b/plugins/steal.py
index 7cd2a2a..60bc9f6 100644
--- a/plugins/steal.py
+++ b/plugins/steal.py
@@ -17,7 +17,7 @@ def get_random(db):
 
 
 def list_steals(db):
-    row = db.execute("SELECT word FROM STOLEN").fetchall()
+    row = db.execute("SELECT word FROM stolen").fetchall()
     return row    
 
 
-- 
1.8.3.2

