From 42fd2074dfe14dde41cd9288c4b8a4a4357c9789 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 14 Dec 2013 10:13:49 +0100
Subject: [PATCH] Update minecraft_status for new server


diff --git a/plugins/minecraft_status.py b/plugins/minecraft_status.py
index 6d1ab3a..0fca419 100644
--- a/plugins/minecraft_status.py
+++ b/plugins/minecraft_status.py
@@ -37,6 +37,8 @@ def mcstatus(inp):
             server = "MC|Session"
         elif server == "minecraft.net":
             server = "MC|Website"
+        elif server == "sessionserver.mojang.com":
+            server = "MJ|SessionServer"
 
         if status == "green":
             servers.append(u"{}{}{}".format(yes_prefix, server, yes_suffix))
-- 
1.8.3.2

