From 401c4cdfb891090e74124e7a12e238621f37203f Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 16 Dec 2013 22:23:25 +0100
Subject: [PATCH] Add chanserv flag plugin


diff --git a/plugins/chanserv_flags.py b/plugins/chanserv_flags.py
new file mode 100644
index 0000000..2c519c2
--- /dev/null
+++ b/plugins/chanserv_flags.py
@@ -0,0 +1,6 @@
+from util import hook
+
+
+@hook.command(permissions=["adminonly"])
+def flags(inp, message=None, chan=None):
+    message("flags {}".format(inp if inp.startswith("#") else "{} {}".format(chan, inp)), target="ChanServ")
-- 
1.8.3.2

