From a1c3549edb3c772ce0919f15682899caf1839a28 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 16 Dec 2013 22:38:27 +0100
Subject: [PATCH] Add 'idb' independent database, to be used instead of the
 server-dependent 'db' database in some cases.


diff --git a/core/main.py b/core/main.py
index 7e32995..97e33ba 100644
--- a/core/main.py
+++ b/core/main.py
@@ -60,6 +60,8 @@ def run(func, input):
     if args:
         if 'db' in args and 'db' not in input:
             input.db = get_db_connection(input.conn)
+        if 'idb' in args and 'idb' not in input:
+            input.idb = get_db_connection(None, name='idb.db')
         if 'input' in args:
             input.input = input
         if 0 in args:
-- 
1.8.3.2

