From e6668503ce6fae6fda4723e751d5a3b8f879e3ad Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 29 Dec 2013 11:05:41 +0100
Subject: [PATCH] Add threaddump


diff --git a/plugins/threaddump.py b/plugins/threaddump.py
new file mode 100644
index 0000000..e8825be
--- /dev/null
+++ b/plugins/threaddump.py
@@ -0,0 +1,15 @@
+import sys
+import traceback
+from util import hook, web
+
+@hook.command
+def threaddump(inp):
+    print "Running thread dump"
+    code = []
+    for threadId, stack in sys._current_frames().items():
+        code.append("\n# ThreadID: %s" % threadId)
+        for filename, lineno, name, line in traceback.extract_stack(stack):
+            code.append('File: "%s", line %d, in %s' % (filename, lineno, name))
+            if line:
+                code.append("  %s" % (line.strip()))
+    return "Thread dump located at {}".format(web.haste("\n".join(code), ext='py'))
-- 
1.8.3.2

