From 886e9c5fbde633560669c903a8f427e65c03915f Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 1 Jan 2014 03:58:34 +0100
Subject: [PATCH] Don't do the 'flirst with ... "'


diff --git a/plugins/attacks.py b/plugins/attacks.py
index 189cfaf..4999d46 100644
--- a/plugins/attacks.py
+++ b/plugins/attacks.py
@@ -54,7 +54,7 @@ def insult(inp, nick=None, action=None, conn=None, notice=None):
 
 
 @hook.command
-def flirt(inp, nick=None, action=None, conn=None, notice=None):
+def flirt(inp, nick=None, message=None, conn=None, notice=None):
     """flirt <user> -- Make the bot flirt with <user>."""
     target = inp.strip()
 
@@ -67,5 +67,4 @@ def flirt(inp, nick=None, action=None, conn=None, notice=None):
     else:
         target = inp
 
-    out = 'flirts with {}... "{}"'.format(target, random.choice(flirts))
-    action(out)
+    message('{}, {}'.format(random.choice((target, nick)), random.choice(flirts)))
-- 
1.8.3.2

