From b9571927b63c1937120ccbb710ade2721b6b8bd6 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Fri, 20 Jun 2014 20:36:36 -0500
Subject: [PATCH] Fix spelling error in pyexec


diff --git a/cloudbot/util/pyexec.py b/cloudbot/util/pyexec.py
index 3ccee54..50bec7c 100644
--- a/cloudbot/util/pyexec.py
+++ b/cloudbot/util/pyexec.py
@@ -22,7 +22,7 @@ def eval_py(code, paste_multiline=True):
     if "Traceback (most recent call last):" in output:
         status = "Python error: "
     else:
-        status = "Code executed sucessfully: "
+        status = "Code executed successfully: "
 
     if "\n" in output and paste_multiline:
         return status + web.paste(output, 'py')
-- 
2.0.0

