From 52c72692a6611ad5832728b15b4a280af02b4749 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 19:09:15 -0800
Subject: [PATCH] Don't run regex matching if command is already matched


diff --git a/core/main.py b/core/main.py
index 640a3a9..1ceecd0 100644
--- a/core/main.py
+++ b/core/main.py
@@ -179,6 +179,7 @@ def main(bot, conn, out):
 
                 func, args = bot.commands[command]
                 dispatch(bot, input, "command", func, args, autohelp=True)
+                return
 
         # REGEXES
         for func, args in bot.plugins['regex']:
-- 
1.9.1

