From b529fd9cd6d0fd4bf099bfeff63762e83af525b9 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 14:57:15 -0700
Subject: [PATCH] Documentation type updates in core/irc.py


diff --git a/core/irc.py b/core/irc.py
index c6048cc..d5fb8a8 100644
--- a/core/irc.py
+++ b/core/irc.py
@@ -213,14 +213,14 @@ class BotConnection(object):
     :type bot: core.bot.CloudBot
     :type name: str
     :type channels: list[str]
-    :type config: dict[str, ?]
+    :type config: dict[str, unknown]
     :type ssl: bool
     :type server: str
     :type port: int
     :type logger: logging.Logger
     :type nick: str
     :type vars: dict
-    :type history: dict
+    :type history: dict[str, list[tuple]]
     :type parsed_queue: queue.Queue
     :type input_queue: queue.Queue
     :type output_queue: queue.Queue
@@ -239,7 +239,7 @@ class BotConnection(object):
         :type ssl: bool
         :type logger: logging.Logger
         :type channels: list[str]
-        :type config: dict[str, ?]
+        :type config: dict[str, unknown]
         """
         self.bot = bot
         self.name = name
-- 
1.9.2

