From 282b20eb3694256cced47525c337c6288e2d991c Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 20 Apr 2014 03:59:28 -0700
Subject: [PATCH] Fix unicode encoding error in youtube


diff --git a/plugins/youtube.py b/plugins/youtube.py
index a29f31a..6fbdaa9 100644
--- a/plugins/youtube.py
+++ b/plugins/youtube.py
@@ -134,4 +134,4 @@ def ytplaylist_url(match):
     author = soup.find('img', {'class': 'channel-header-profile-image'})['title']
     num_videos = soup.find('ul', {'class': 'header-stats'}).findAll('li')[0].text.split(' ')[0]
     views = soup.find('ul', {'class': 'header-stats'}).findAll('li')[1].text.split(' ')[0]
-    return "\x02{}\x02 - \x02{}\x02 views - \x02{}\x02 videos - \x0{}\x02".format(title, views, num_videos, author)
+    return "\x02{}\x02 - \x02{}\x02 views - \x02{}\x02 videos - \x02{}\x02".format(title, views, num_videos, author)
-- 
1.9.2

