From 9cbcb7605ae3a77b232b5a935e1e9677d5bd69df Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 20 Apr 2014 03:59:08 -0700
Subject: [PATCH] onload does not exist!!!


diff --git a/modules/horoscope.py b/modules/horoscope.py
index 5654169..c436169 100644
--- a/modules/horoscope.py
+++ b/modules/horoscope.py
@@ -13,13 +13,6 @@ def db_init(db):
         db.commit()
         db_ready = True
 
-
-# TODO: this isn't a thing. We should make this a thing
-@hook.onload
-def init(db):
-    db_init(db)
-
-
 @hook.command(autohelp=False)
 def horoscope(text, db, notice, nick):
     """horoscope <sign> -- Get your horoscope."""
-- 
1.9.2

