From 5478cd91eb3cff47ea6a42c980dffe4adeda3bb4 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 6 Apr 2014 23:56:03 -0700
Subject: [PATCH] Don't run config reload event if bot is no longer running.
 This adds better support for restarting to config


diff --git a/core/config.py b/core/config.py
index b395459..e8b8ae1 100644
--- a/core/config.py
+++ b/core/config.py
@@ -58,17 +58,23 @@ class Config(dict):
 
         pattern = "*{}".format(self.filename)
 
-        self.event_handler = ConfigEventHandler(self, patterns=[pattern])
+        self.event_handler = ConfigEventHandler(self.bot, self, patterns=[pattern])
         self.observer.schedule(self.event_handler, path='.', recursive=False)
         self.observer.start()
 
 
 class ConfigEventHandler(Trick):
-    def __init__(self, config, *args, **kwargs):
+    def __init__(self, bot, config, *args, **kwargs):
+        """
+        :type bot: core.bot.CloudBot
+        :type config: Config
+        """
+        self.bot = bot
         self.config = config
         self.logger = config.logger
         Trick.__init__(self, *args, **kwargs)
 
     def on_any_event(self, event):
-        self.logger.info("Config changed, triggering reload.")
-        self.config.load_config()
+        if self.bot.running:
+            self.logger.info("Config changed, triggering reload.")
+            self.config.load_config()
-- 
1.9.1

