From 227b766e43b20bbc094e652d48fe98077be69d80 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 23 Apr 2014 16:42:08 -0700
Subject: [PATCH] Add a has_run attribute to Plugin for use in determining test
 completeness


diff --git a/core/main.py b/core/main.py
index 8dbbe4d..0878ec9 100644
--- a/core/main.py
+++ b/core/main.py
@@ -296,6 +296,8 @@ def dispatch(bot, input, plugin):
             input.notice(input.conn.config["command_prefix"] + plugin.name + " requires additional arguments.")
         return
 
+    plugin.has_run = True
+
     if plugin.type == "event" and plugin.args.get("run_sync"):
         run(bot, plugin, input)  # make sure the event runs in sync, *before* commands and regexes are run
         # TODO: Possibly make this run in the command's/regex's own thread, or make all of 'main' run async?
diff --git a/core/pluginmanager.py b/core/pluginmanager.py
index 9fa73e5..9315e55 100644
--- a/core/pluginmanager.py
+++ b/core/pluginmanager.py
@@ -333,6 +333,7 @@ class Plugin:
         self.function = func_hook.function
         self.function_name = self.function.__name__
         self.args = func_hook.kwargs
+        self.has_run = False
 
     def __repr__(self):
         return "type: {}, module: {}, args: {}".format(self.type, self.module.file_name, self.args)
-- 
1.9.2

