From d6e0b0218b74cc200e7c6039d84335f0b272429c Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 22 Apr 2014 10:29:56 -0700
Subject: [PATCH] Add first test, tests rpass.


diff --git a/qpt_tests/generic_tests.py b/qpt_tests/generic_tests.py
new file mode 100644
index 0000000..3aa7567
--- /dev/null
+++ b/qpt_tests/generic_tests.py
@@ -0,0 +1,16 @@
+import re
+
+from util import qpt
+
+rpass_re = re.compile("NOTICE Dabo :Your password is '\w+ \w+ \w+ \w+'. "
+                      "Feel free to remove the spaces when using it.")
+
+
+def match_re(reg, _str):
+    assert reg.match(_str), reg.pattern + " didn't match " + _str
+
+
+@qpt.test
+def test_rpass(bot, send, get):
+    send(":Dabo!~dabo@dabo.us PRIVMSG #dabo :>rpass")
+    match_re(rpass_re, get())
-- 
1.9.2

