From 63880122f69b29c1b93f35cb61e850a78996eef8 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 5 Apr 2014 23:11:52 -0700
Subject: [PATCH] Change ctcp to Eliralin


diff --git a/plugins/core_ctcp.py b/plugins/core_ctcp.py
index 3e7a200..73af9bd 100644
--- a/plugins/core_ctcp.py
+++ b/plugins/core_ctcp.py
@@ -6,7 +6,7 @@ from util import hook
 # CTCP responses
 @hook.regex(r'^\x01VERSION\x01$')
 def ctcp_version(inp, notice=None):
-    notice('\x01VERSION: CloudBot - http://git.io/cloudbotirc')
+    notice('\x01VERSION Eliralin 1.0')
 
 
 @hook.regex(r'^\x01PING\x01$')
-- 
1.9.1

