From 9a8a582f488a9c2af34dafd1505a039ab61054f7 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 8 Apr 2014 00:16:51 -0700
Subject: [PATCH] blha303 updates to plpaste


diff --git a/plugins/plpaste.py b/plugins/plpaste.py
index 238037d..d46e672 100644
--- a/plugins/plpaste.py
+++ b/plugins/plpaste.py
@@ -1,12 +1,14 @@
 from util import hook, web
+from os import listdir
 
 
-@hook.command(adminonly=True)
-def plpaste(inp):
-    if "/" in inp and inp.split("/")[0] != "util":
-        return "Invalid input"
-    try:
-        with open("plugins/%s.py" % inp) as f:
+@hook.command(permissions=["adminonly"])
+def plpaste(inp, bot=None):
+    if inp in bot.commands:
+        with open(bot.commands[inp][0].func_code.co_filename.strip()) as f:
             return web.haste(f.read(), ext='py')
-    except IOError:
-        return "Plugin not found (must be in plugins folder)"
+    elif inp + ".py" in listdir('plugins/'):
+        with open('plugins/{}.py'.format(inp)) as f:
+            return web.haste(f.read(), ext='py')
+    else:
+        return "Could not find specified plugin."
-- 
1.9.1

