From d6df98130dbedd10774cf081585a33cf7e2297c0 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 16 Apr 2014 17:40:34 -0700
Subject: [PATCH] Make user_groups actually return a user's groups, not
 permissions.


diff --git a/plugins/admin.py b/plugins/admin.py
index e331268..746ed85 100644
--- a/plugins/admin.py
+++ b/plugins/admin.py
@@ -91,11 +91,11 @@ def get_user_groups(inp, conn=None, has_permission=None, notice=None, mask=None)
 
     permission_manager = conn.permissions
 
-    user_permissions = permission_manager.get_user_permissions(user)
-    if user_permissions:
-        return "User {} has permissions: {}".format(user, user_permissions)
+    user_groups = permission_manager.get_user_groups(user)
+    if user_groups:
+        return "User {} is in groups: {}".format(user, user_groups)
     else:
-        return "User {} has no elevated permissions".format(user)
+        return "User {} is in no permission groups".format(user)
 
 
 @hook.command("deluser", permissions=["permissions_users"])
-- 
1.9.2

