From 746ecaeb112f3d117de10bc51983f7450317869b Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 6 Apr 2014 21:57:12 -0700
Subject: [PATCH] Initialize PermissionManager in irc.py instead of bot.py


diff --git a/core/bot.py b/core/bot.py
index 673821d..26a9ee9 100644
--- a/core/bot.py
+++ b/core/bot.py
@@ -100,9 +100,6 @@ class CloudBot(threading.Thread):
         # start bot instances
         self.create_connections()
 
-        for connection in self.connections:
-            connection.permissions = PermissionManager(self, connection)
-
         # run plugin loader
         self.plugins = collections.defaultdict(list)
 
@@ -152,7 +149,7 @@ class CloudBot(threading.Thread):
 
             self.logger.debug("Creating BotInstance for {}.".format(name))
 
-            self.connections.append(irc.BotConnection(name, server, nick, config=conf,
+            self.connections.append(irc.BotConnection(self, name, server, nick, config=conf,
                                                       port=port, logger=self.logger, channels=conf['channels'],
                                                       ssl=conf['connection'].get('ssl', False)))
             self.logger.debug("({}) Created connection.".format(name))
diff --git a/core/irc.py b/core/irc.py
index 44e7283..18b7303 100644
--- a/core/irc.py
+++ b/core/irc.py
@@ -7,6 +7,7 @@ from ssl import wrap_socket, CERT_NONE, CERT_REQUIRED, SSLError
 
 from core.permissions import PermissionManager
 
+
 irc_prefix_rem = re.compile(r'(.*?) (.*?) (.*)').match
 irc_noprefix_rem = re.compile(r'()(.*?) (.*)').match
 irc_netmask_rem = re.compile(r':?([^!@]*)!?([^@]*)@?(.*)').match
@@ -210,7 +211,6 @@ class BotConnection(object):
 
     def __init__(self, bot, name, server, nick, port=6667, ssl=False, logger=None, channels=None, config=None):
         """
-
         :type bot: core.bot.CloudBot
         :type name: str
         :type server: str
@@ -344,4 +344,4 @@ class BotConnection(object):
         except:
             # if this doesn't work, no big deal
             pass
-        self.output_queue.put(string)
\ No newline at end of file
+        self.output_queue.put(string)
-- 
1.9.1

