From a4ed0420b6f9f7c894b47a50a23422ff829d1f6b Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 14:57:40 -0700
Subject: [PATCH] Assign None to input.trigger in __init__


diff --git a/core/main.py b/core/main.py
index 408aedd..2562813 100644
--- a/core/main.py
+++ b/core/main.py
@@ -64,6 +64,8 @@ class Input:
         if self.chan == conn.nick.lower():  # is a PM
             self.chan = nick
 
+        self.trigger = None  # assigned later?
+
     def message(self, message, target=None):
         """sends a message to a specific or current channel/user
         :type message: str
-- 
1.9.2

