From d0ba628c3e58be4aa2f8b3d494f4ed47b504ef54 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 14:58:24 -0700
Subject: [PATCH] Use notice for error messages in coin.py, and change coins =
 0 message


diff --git a/plugins/coin.py b/plugins/coin.py
index 7cc2a2a..f084052 100644
--- a/plugins/coin.py
+++ b/plugins/coin.py
@@ -4,21 +4,24 @@ from util import hook
 
 
 @hook.command(autohelp=False)
-def coin(inp, action=None):
-    """coin [amount] -- Flips [amount] of coins."""
+def coin(inp, notice=None, action=None):
+    """coin [amount] -- Flips [amount] of coins.
+    :type inp: str
+    """
 
     if inp:
         try:
             amount = int(inp)
         except (ValueError, TypeError):
-            return "Invalid input!"
+            notice("Invalid input '{}': not a number".format(inp))
+            return
     else:
         amount = 1
 
     if amount == 1:
         action("flips a coin and gets {}.".format(random.choice(["heads", "tails"])))
     elif amount == 0:
-        action("makes a coin flipping motion with its hands.")
+        action("makes a coin flipping motion")
     else:
         heads = int(random.normalvariate(.5 * amount, (.75 * amount) ** .5))
         tails = amount - heads
-- 
1.9.2

