From db483019ceb2e412fa05e34f53ff685f14481e33 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 23 Apr 2014 10:06:48 -0700
Subject: [PATCH] Add 'run_sync' option to event plugins, to make sure they run
 before any commands or regexes


diff --git a/core/main.py b/core/main.py
index 4698ac4..54beeb5 100644
--- a/core/main.py
+++ b/core/main.py
@@ -255,7 +255,10 @@ def dispatch(bot, input, plugin):
             input.notice(input.conn.config["command_prefix"] + plugin.name + " requires additional arguments.")
         return
 
-    if plugin.args.get("singlethread", False):
+    if plugin.type == "event" and plugin.args.get("run_sync"):
+        run(bot, plugin, input)  # make sure the event runs in sync, *before* commands and regexes are run
+        # TODO: Possibly make this run in the command's/regex's own thread, or make all of 'main' run async?
+    elif plugin.args.get("singlethread", False):
         if plugin in bot.threads:
             bot.threads[plugin].put(input)
         else:
diff --git a/core/pluginmanager.py b/core/pluginmanager.py
index de1a158..ad11014 100644
--- a/core/pluginmanager.py
+++ b/core/pluginmanager.py
@@ -287,14 +287,13 @@ class CommandPlugin(Plugin):
         :type module: Module
         :type cmd_hook: hook._CommandHook
         """
+        Plugin.__init__(self, "command", module, cmd_hook)
 
         # make sure that autohelp and permissions are set
-        if not "autohelp" in cmd_hook.kwargs:
-            cmd_hook.kwargs["autohelp"] = True
-        if not "permissions" in cmd_hook.kwargs:
-            cmd_hook.kwargs["permissions"] = []
-
-        Plugin.__init__(self, "command", module, cmd_hook)
+        if not "autohelp" in self.args:
+            self.args["autohelp"] = True
+        if not "permissions" in self.args:
+            self.args["permissions"] = []
 
         self.name = cmd_hook.main_alias
         self.aliases = list(cmd_hook.aliases)  # turn the set into a list
@@ -340,6 +339,10 @@ class EventPlugin(Plugin):
         :type event_hook: hook._EventHook
         """
         Plugin.__init__(self, "event", module, event_hook)
+
+        if not "run_sync" in self.args:
+            self.args["run_sync"] = False
+
         self.events = event_hook.events
 
     def is_catch_all(self):
-- 
1.9.2

