From 03309bef8b6b14c7d803b57e91960f9a2d65fd1e Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:05:12 -0800
Subject: [PATCH] Update minecraft_status to use unicode checks, and with a new
 format.


diff --git a/modules/minecraft_status.py b/modules/minecraft_status.py
index de60e94..c26706e 100644
--- a/modules/minecraft_status.py
+++ b/modules/minecraft_status.py
@@ -3,42 +3,40 @@ import json
 from util import hook, http
 
 
-@hook.command(autohelp=False)
+green_prefix = "\x02\x0f"
+green_suffix = ": \x033\x02\u2714"
+yellow_prefix = "\x02\x0f"
+yellow_suffix = ": \x037\x02\u26A0"
+red_prefix = "\x02\x0f"
+red_suffix = ": \x034\x02\u2716"
+
+
+@hook.command(["mcs", "mcstatus", "mojang"], autohelp=False)
 def mcstatus():
     """mcstatus -- Checks the status of various Mojang (the creators of Minecraft) servers."""
 
     try:
-        request = http.get("http://status.mojang.com/check")
+        result = http.get("http://status.mojang.com/check")
     except (http.URLError, http.HTTPError) as e:
         return "Unable to get Minecraft server status: {}".format(e)
 
-    # lets just reformat this data to get in a nice format
-    data = json.loads(request.replace("}", "").replace("{", "").replace("]", "}").replace("[", "{"))
-
-    out = []
+    data = json.loads(result)
 
     # use a loop so we don't have to update it if they add more servers
-    green = []
-    yellow = []
-    red = []
-    for server, status in list(data.items()):
-        if status == "green":
-            green.append(server)
-        elif status == "yellow":
-            yellow.append(server)
-        else:
-            red.append(server)
-
-    if green:
-        out = "\x033\x02Online\x02\x0f: " + ", ".join(green)
-        if yellow:
-            out += " "
-    if yellow:
-        out += "\x02Issues\x02: " + ", ".join(yellow)
-        if red:
-            out += " "
-    if red:
-        out += "\x034\x02Offline\x02\x0f: " + ", ".join(red)
-
-    return "\x0f" + out.replace(".mojang.com", ".mj") \
-        .replace(".minecraft.net", ".mc")
+    servers = []
+    for server_dict in data:
+        for server, status in server_dict.items():
+            if server == "minecraft.net":
+                server = "MC|Website"
+            elif server.endswith(".mojang.com"):
+                server = "MJ|{}".format(server[:-11].capitalize())
+            elif server.endswith(".minecraft.net"):
+                server = "MC|{}".format(server[:-14].capitalize())
+
+            if status == "green":
+                servers.append("{}{}{}".format(green_prefix, server, green_suffix))
+            elif status == "yellow":
+                servers.append("{}{}{}".format(yellow_prefix, server, yellow_suffix))
+            else:
+                servers.append("{}{}{}".format(red_prefix, server, red_suffix))
+    return "  ".join(servers)
-- 
1.9.2

