From f7336d9dd26d2b789dacff5145e34cd3bf2580d8 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 14:58:54 -0700
Subject: [PATCH] Add type documentation to core_misc plugin


diff --git a/plugins/core_misc.py b/plugins/core_misc.py
index fa47447..679caf9 100644
--- a/plugins/core_misc.py
+++ b/plugins/core_misc.py
@@ -8,15 +8,24 @@ socket.setdefaulttimeout(10)
 
 # Auto-join on Invite (Configurable, defaults to True)
 @hook.event('INVITE')
-def invite(paraml, conn=None):
+def invite(paramlist, conn=None):
+    """
+    :type paramlist: list[str]
+    :type conn: core.irc.BotConnection
+    """
     invite_join = conn.config.get('invite_join', True)
     if invite_join:
-        conn.join(paraml[-1])
+        conn.join(paramlist[-1])
 
 
 # Identify to NickServ (or other service)
 @hook.event('004')
-def onjoin(paraml, conn=None, bot=None):
+def onjoin(paramlist, conn=None, bot=None):
+    """
+    :type paramlist: list[str]
+    :type conn: core.irc.BotConnection
+    :type bot: core.bot.CloudBot
+    """
     bot.logger.info("ONJOIN hook triggered.")
     nickserv = conn.config.get('nickserv')
     if nickserv:
@@ -52,7 +61,11 @@ def onjoin(paraml, conn=None, bot=None):
 
 @hook.singlethread
 @hook.event('004')
-def keep_alive(paraml, conn=None):
+def keep_alive(paramlist, conn=None):
+    """
+    :type paramlist: list[str]
+    :type conn: core.irc.BotConnection
+    """
     keepalive = conn.config.get('keep_alive', False)
     if keepalive:
         while True:
-- 
1.9.2

