From a4575f8d3aa6f1f13b5a79b815b153d8354addee Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 23 Apr 2014 10:08:04 -0700
Subject: [PATCH] Don't need autohelp on password commands


diff --git a/modules/password.py b/modules/password.py
index 7575b9d..0ceb42b 100644
--- a/modules/password.py
+++ b/modules/password.py
@@ -12,7 +12,7 @@ with open("data/password_words.txt") as f:
     common_words = [line.strip() for line in f.readlines()]
 
 
-@hook.command
+@hook.command(autoHelp=False)
 def password(text, notice):
     """password <length> [types] -- Generates a password of <length> (default 10).
     [types] can include 'alpha', 'no caps', 'numeric', 'symbols' or any combination of the inp, eg. 'numbers symbols'"""
@@ -56,7 +56,7 @@ def password(text, notice):
     notice(pw)
 
 
-@hook.command(["rpass", "rpassword", "readablepassword"])
+@hook.command(["rpass", "rpassword", "readablepassword"], autohelp=False)
 def readable_password(text, notice):
     """rpass [length]  -- Generates an easy to remember password with [length] (default 4) commonly used words."""
     if text:
-- 
1.9.2

