From 6b897fe03181a67a083c5462f70ba68c5c106dee Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 6 Apr 2014 21:59:57 -0700
Subject: [PATCH] Start on plugin cleanups - to be completed later


diff --git a/plugins/core_sieve.py b/plugins/core_sieve.py
index 1f6836f..09107db 100644
--- a/plugins/core_sieve.py
+++ b/plugins/core_sieve.py
@@ -58,9 +58,11 @@ def sieve_suite(bot, input, func, kind, args):
         allowed_permissions = args.get('permissions', [])
         for perm in allowed_permissions:
             if conn.permissions.has_perm_mask(mask, perm):
+                print("Allowed mask {} access to {}".format(mask, perm))
                 return input
-
+        print("Disallowed mask {} from access to {}".format(mask, args.get('permissions')))
         input.notice("Sorry, you are not allowed to use this command.")
         return None
-
+    else:
+        print("Allowed mask {} access to command {} without permissions".format(input.mask.lower(), func.__name__))
     return input
-- 
1.9.1

