From 60602038cf2034dcd80075c41fec5dc3ed994529 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Thu, 17 Apr 2014 19:47:33 -0700
Subject: [PATCH] Document updates for factoid, all on one line - to show.


diff --git a/plugins/factoids.py b/plugins/factoids.py
index a702b70..96515f5 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -40,8 +40,7 @@ def get_memory(db, word):
 @hook.command("r", permissions=["addfactoid"])
 @hook.command(permissions=["addfactoid"])
 def remember(inp, nick='', db=None, notice=None):
-    """remember <word> [+]<data> -- Remembers <data> with <word>. Add +
-    to <data> to append."""
+    """remember <word> [+]<data> -- Remembers <data> with <word>. Add + to <data> to append."""
     db_init(db)
 
     append = False
-- 
1.9.2

