From 30e50bcb46641ff91fb477a801b678025ff8dc26 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 17:39:33 -0700
Subject: [PATCH] Make eightball use the old format - as it wasn't using either
 the old or the new command format


diff --git a/plugins/eightball.py b/plugins/eightball.py
index 4730bc4..1254ef4 100644
--- a/plugins/eightball.py
+++ b/plugins/eightball.py
@@ -15,9 +15,12 @@ with open("./data/8ball_responses.txt") as f:
 
 
 @hook.command()
-def eightball(input, conn):
-    """8ball <question> -- The all knowing magic eight ball,
-    in electronic form. Ask and it shall be answered!"""
+@hook.command("8")
+@hook.command("8ball")
+def eightball(inp, action=None):
+    """8ball <question> -- The all knowing magic eight ball, in electronic form. Ask and it shall be answered!
+    :type inp: str
+    """
 
     magic = text.multiword_replace(random.choice(responses), color_codes)
-    input.action("shakes the magic 8 ball... {}".format(magic))
+    action("shakes the magic 8 ball... {}".format(magic))
-- 
1.9.2

