From 31272d3d892f95639b8c004a3af82c9171071bc1 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 22 Apr 2014 22:14:26 -0700
Subject: [PATCH] Make log plugin log events to console separately from logging
 to file - to prevent console lag


diff --git a/modules/log.py b/modules/log.py
index 14d317c..390893e 100644
--- a/modules/log.py
+++ b/modules/log.py
@@ -123,4 +123,8 @@ def log(input, bot):
             channel_log = get_log_stream(bot.data_dir, input.server, channel)
             channel_log.write(human_readable + '\n')
 
-        bot.logger.info(human_readable)
+# Log console separately to prevent lag
+@hook.event("*")
+def console_log(input, bot):
+    human_readable = beautify(input)
+    bot.logger.info(human_readable)
-- 
1.9.2

