From d21c00d9d7c66a494bbad629cada1a4060c79cde Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 22 Apr 2014 22:25:33 -0700
Subject: [PATCH] Fix minecraft_ping for python 3


diff --git a/modules/minecraft_ping.py b/modules/minecraft_ping.py
index cc31679..f590f03 100644
--- a/modules/minecraft_ping.py
+++ b/modules/minecraft_ping.py
@@ -74,8 +74,8 @@ def mcping_modern(host, port):
             raise PingError("Request timed out")
 
         # send handshake + status request
-        s.send(pack_data("\x00\x00" + pack_data(host.encode('utf8')) + pack_port(port) + "\x01"))
-        s.send(pack_data("\x00"))
+        s.send(pack_data(b"\x00\x00" + pack_data(host.encode('utf8')) + pack_port(port) + b"\x01"))
+        s.send(pack_data(b"\x00"))
 
         # read response
         unpack_varint(s)  # Packet length
@@ -85,7 +85,7 @@ def mcping_modern(host, port):
         if not l > 1:
             raise PingError("Invalid response")
 
-        d = ""
+        d = b""
         while len(d) < l:
             d += s.recv(1024)
 
-- 
1.9.2

