From edb90a4be40c1ba44c4ea33180b77d7e0f182dad Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 6 Apr 2014 23:11:08 -0700
Subject: [PATCH] Remove conflict


diff --git a/cloudbot.py b/cloudbot.py
index e8a905f..83f2ad1 100755
--- a/cloudbot.py
+++ b/cloudbot.py
@@ -7,11 +7,6 @@ import signal
 from core import bot
 
 
-
-<<<<<<< HEAD
-=======
-
->>>>>>> Sleep for 1 second before restarting
 # check python version 
 if sys.version_info < (3, 2, 0):
     print("CloudBot3 requires Python 3.2 or newer.")
-- 
1.9.1

