From 9418639041080c2e9c0f67ac12c5f2fc8a6b7381 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 26 Apr 2014 19:43:21 -0700
Subject: [PATCH] Replace symbol for factoids.

Changes ?fact to ^fact

diff --git a/modules/factoids.py b/modules/factoids.py
index edc8a5f..5689b0c 100644
--- a/modules/factoids.py
+++ b/modules/factoids.py
@@ -97,7 +97,7 @@ def info(text, notice, db):
         notice("Unknown Factoid.")
 
 
-@hook.regex(r'^\? ?(.+)')
+@hook.regex(r'^\^ ?(.+)')
 def factoid(inp, input, db, message, action):
     """?<word> -- Shows what data is associated with <word>."""
 
-- 
1.9.2

