From bbcc0ef111a1f4756d8a2e1d140043deee85fc9f Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 15:01:49 -0700
Subject: [PATCH] Make 'get' function decode message into string, to add python
 3 support to get_json and other get_ methods


diff --git a/util/http.py b/util/http.py
index d07ca6a..746571c 100644
--- a/util/http.py
+++ b/util/http.py
@@ -31,7 +31,7 @@ jar = http.cookiejar.CookieJar()
 
 
 def get(*args, **kwargs):
-    return open(*args, **kwargs).read()
+    return open(*args, **kwargs).read().decode()
 
 
 def get_url(*args, **kwargs):
-- 
1.9.2

