From 83ab77e8b01dd39687269685ae595b3f26b67717 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 22 Apr 2014 22:26:42 -0700
Subject: [PATCH] Fix ping for python 3


diff --git a/modules/ping.py b/modules/ping.py
index 66286df..1ccfce1 100644
--- a/modules/ping.py
+++ b/modules/ping.py
@@ -5,7 +5,6 @@ import os
 
 from util import hook
 
-
 ping_regex = re.compile(r"(\d+.\d+)/(\d+.\d+)/(\d+.\d+)/(\d+.\d+)")
 
 
@@ -36,7 +35,7 @@ def ping(inp, reply=None):
 
     reply("Attempting to ping {} {} times...".format(host, count))
 
-    pingcmd = subprocess.check_output(["ping", "-c", count, host])
+    pingcmd = subprocess.check_output(["ping", "-c", count, host]).decode("utf-8")
     if "request timed out" in pingcmd or "unknown host" in pingcmd:
         return "error: could not ping host"
     else:
-- 
1.9.2

