From 0d1f867084b97bdbf3c1748bc442b7b4b16f4ade Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:18:39 -0800
Subject: [PATCH] Add per-channel regex filtering.


diff --git a/modules/regex_chans.py b/modules/regex_chans.py
new file mode 100644
index 0000000..ac230d2
--- /dev/null
+++ b/modules/regex_chans.py
@@ -0,0 +1,129 @@
+from util import hook
+
+
+# Default value.
+# If True, all channels without a setting will have regex enabled
+# If False, all channels without a setting will have regex disabled
+default_enabled = True
+
+db_already_initiated = False
+
+
+def db_init(db):
+    global db_already_initiated
+    if not db_already_initiated:
+        db_already_initiated = True
+        db.execute("CREATE TABLE IF NOT EXISTS regexchans(channel PRIMARY KEY, status)")
+        db.commit()
+
+
+def get_status(db, channel):
+    row = db.execute("SELECT status FROM regexchans WHERE channel = ?", [channel]).fetchone()
+    if row:
+        return row[0]
+    else:
+        return None
+
+
+def set_status(db, channel, status):
+    row = db.execute("REPLACE INTO regexchans (channel, status) VALUES(?, ?)", [channel, status])
+    db.commit()
+
+
+def delete_status(db, channel):
+    row = db.execute("DELETE FROM regexchans WHERE channel = ?", [channel])
+    db.commit()
+
+
+def list_status(db):
+    row = db.execute("SELECT * FROM regexchans").fetchall()
+    result = None
+    for values in row:
+        if result:
+            result += ", {}: {}".format(values[0], values[1])
+        else:
+            result = "{}: {}".format(values[0], values[1])
+    return result
+
+
+@hook.sieve
+def sieve_regex(bot, input, plugin):
+    db = bot.get_db_connection(input.conn)
+    db_init(db)
+    if plugin.type == "regex" and input.chan.startswith("#") and plugin.module.title != "factoids":
+        chanstatus = get_status(db, input.chan)
+        if chanstatus != "ENABLED" and (chanstatus == "DISABLED" or not default_enabled):
+            bot.logger.info("[{}] Denying {} from {}"
+                            .format(input.conn.readable_name, plugin.function_name, input.chan))
+            return None
+        bot.logger.info("[{}] Allowing {} to {}".format(input.conn.readable_name, plugin.function_name, input.chan))
+
+    return input
+
+
+@hook.command(permissions=["botcontrol"])
+def enableregex(inp, db, message, notice, chan, nick):
+    db_init(db)
+    inp = inp.strip().lower()
+    if not inp:
+        channel = chan
+    elif inp.startswith("#"):
+        channel = inp
+    else:
+        channel = "#{}".format(inp)
+
+    message("Enabling regex matching (youtube, etc) (issued by {})".format(nick), target=channel)
+    notice("Enabling regex matching (youtube, etc) in channel {}".format(channel))
+    set_status(db, channel, "ENABLED")
+
+
+@hook.command(permissions=["botcontrol"])
+def disableregex(inp, db, message, notice, chan, nick):
+    db_init(db)
+    inp = inp.strip().lower()
+    if not inp:
+        channel = chan
+    elif inp.startswith("#"):
+        channel = inp
+    else:
+        channel = "#{}".format(inp)
+
+    message("Disabling regex matching (youtube, etc) (issued by {})".format(nick), target=channel)
+    notice("Disabling regex matching (youtube, etc) in channel {}".format(channel))
+    set_status(db, channel, "DISABLED")
+
+
+@hook.command(permissions=["botcontrol"])
+def resetregex(inp, db, message, notice, chan, nick):
+    db_init(db)
+    inp = inp.strip().lower()
+    if not inp:
+        channel = chan
+    elif inp.startswith("#"):
+        channel = inp
+    else:
+        channel = "#{}".format(inp)
+
+    message("Resetting regex matching setting (youtube, etc) (issued by {})".format(nick), target=channel)
+    notice("Resetting regex matching setting (youtube, etc) in channel {}".format(channel))
+    delete_status(db, channel)
+
+
+@hook.command(permissions=["botcontrol"])
+def regexstatus(inp, db, chan):
+    db_init(db)
+    inp = inp.strip().lower()
+    if not inp:
+        channel = chan
+    elif inp.startswith("#"):
+        channel = inp
+    else:
+        channel = "#{}".format(inp)
+
+    return "Regex status for {}: {}".format(channel, get_status(db, channel))
+
+
+@hook.command(permissions=["botcontrol"])
+def listregex(db):
+    db_init(db)
+    return list_status(db)
-- 
1.9.2

