From 48010c7047c842c1632aa86b6f28ebb067e50ee9 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 29 Apr 2014 16:58:41 -0700
Subject: [PATCH] Update per-channel-prefixes


diff --git a/modules/per_channel_prefix.py b/modules/per_channel_prefix.py
index 9117d53..d5e024e 100644
--- a/modules/per_channel_prefix.py
+++ b/modules/per_channel_prefix.py
@@ -80,6 +80,5 @@ def run_extra_prefix(input, bot, conn, chan, lastparam):
                 # this input is a one-use input, so we can just modify it and send it to the plugin
                 input.trigger = command
                 input.text = match.group(3).strip()
+                input.inp = input.text  # re-set inp as well, for compatibilty
                 main.dispatch(bot, input, plugin)
-            else:
-                print("Command not found {}".format(command))
-- 
1.9.2

