From 775453782279c8be46fee54fe19470615676c874 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 19 Apr 2014 14:22:10 -0700
Subject: [PATCH] Respect nickserv "enabled" status


diff --git a/plugins/core_misc.py b/plugins/core_misc.py
index 679caf9..0195b27 100644
--- a/plugins/core_misc.py
+++ b/plugins/core_misc.py
@@ -29,6 +29,8 @@ def onjoin(paramlist, conn=None, bot=None):
     bot.logger.info("ONJOIN hook triggered.")
     nickserv = conn.config.get('nickserv')
     if nickserv:
+        if "enabled" in nickserv and not nickserv["enabled"]:
+            return
         nickserv_password = nickserv.get('nickserv_password', '')
         nickserv_name = nickserv.get('nickserv_name', 'nickserv')
         nickserv_account_name = nickserv.get('nickserv_user', '')
-- 
1.9.2

