From 7a86090cbe1c9175198b3e4204cc4f71da2ae1e7 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 22 Apr 2014 22:28:32 -0700
Subject: [PATCH] Fix hash for python 3


diff --git a/modules/utility.py b/modules/utility.py
index 86eaf37..2d1cc49 100644
--- a/modules/utility.py
+++ b/modules/utility.py
@@ -5,6 +5,7 @@ import re
 from util import hook, formatting
 
 
+
 # variables
 
 colors = collections.OrderedDict([
@@ -141,9 +142,9 @@ def reverse(inp):
 
 
 @hook.command("hash")
-def hash_command(inp):
+def hash_command(text):
     """hash <string> -- Returns hashes of <string>."""
-    return ', '.join(x + ": " + getattr(hashlib, x)(inp).hexdigest()
+    return ', '.join(x + ": " + getattr(hashlib, x)(text.encode("utf-8")).hexdigest()
                      for x in ['md5', 'sha1', 'sha256'])
 
 
-- 
1.9.2

