From 4d7c4b88472657abe379d4c21a53a63b2b486562 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 17:04:36 -0700
Subject: [PATCH] Document and format dictionary plugin


diff --git a/plugins/dictionary.py b/plugins/dictionary.py
index e80a9b1..a763815 100644
--- a/plugins/dictionary.py
+++ b/plugins/dictionary.py
@@ -6,9 +6,11 @@ from util import http
 
 
 @hook.command('dictionary')
-@hook.command
+@hook.command()
 def define(inp):
-    """define <word> -- Fetches definition of <word>."""
+    """define <word> -- Fetches definition of <word>.
+    :type inp: str
+    """
 
     url = 'http://ninjawords.com/'
 
@@ -42,7 +44,7 @@ def define(inp):
             result += article[0]
             if len(article) > 2:
                 result += ' '.join('{}. {}'.format(n + 1, section)
-                                    for n, section in enumerate(article[1:]))
+                                   for n, section in enumerate(article[1:]))
             else:
                 result += article[1] + ' '
 
@@ -66,9 +68,11 @@ def define(inp):
 
 
 @hook.command('e')
-@hook.command
+@hook.command()
 def etymology(inp):
-    """etymology <word> -- Retrieves the etymology of <word>."""
+    """etymology <word> -- Retrieves the etymology of <word>.
+    :type inp: str
+    """
 
     url = 'http://www.etymonline.com/index.php'
 
-- 
1.9.2

