From c1d596fc57fc7c9c44efbbc7f12dd1fc2fbcbc6e Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 17:04:52 -0700
Subject: [PATCH] Fix drama plugin for python 3


diff --git a/plugins/drama.py b/plugins/drama.py
index d348cba..fe7fde1 100644
--- a/plugins/drama.py
+++ b/plugins/drama.py
@@ -1,4 +1,5 @@
 import re
+from urllib import parse
 
 from util import hook, http, text
 
@@ -12,13 +13,13 @@ def drama(inp):
     """drama <phrase> -- Gets the first paragraph of
     the Encyclopedia Dramatica article on <phrase>."""
 
-    j = http.get_json(api_url, search=inp)
+    data = http.get_json(api_url, search=inp)
 
-    if not j[1]:
+    if not data[1]:
         return "No results found."
-    article_name = j[1][0].replace(' ', '_').encode('utf8')
+    article_name = data[1][0].replace(' ', '_')
 
-    url = ed_url + http.quote(article_name, '')
+    url = ed_url + parse.quote(article_name, '')
     page = http.get_html(url)
 
     for p in page.xpath('//div[@id="bodyContent"]/p'):
@@ -26,6 +27,6 @@ def drama(inp):
             summary = " ".join(p.text_content().splitlines())
             summary = re.sub("\[\d+\]", "", summary)
             summary = text.truncate_str(summary, 220)
-            return "{} :: {}".format(summary, url)
+            return "{} - {}".format(summary, url)
 
     return "Unknown Error."
-- 
1.9.2

