From 34eae1f2b7ccd037e8a60ab59044a179a1d72dcb Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 18:03:20 -0700
Subject: [PATCH] Fix plpaste refresh support


diff --git a/plugins/plpaste.py b/plugins/plpaste.py
index d46e672..edcadb9 100644
--- a/plugins/plpaste.py
+++ b/plugins/plpaste.py
@@ -1,3 +1,4 @@
+import os
 from util import hook, web
 from os import listdir
 
@@ -5,7 +6,7 @@ from os import listdir
 @hook.command(permissions=["adminonly"])
 def plpaste(inp, bot=None):
     if inp in bot.commands:
-        with open(bot.commands[inp][0].func_code.co_filename.strip()) as f:
+        with open(os.path.join("plugins", bot.commands[inp][0].__code__.co_filename.strip())) as f:
             return web.haste(f.read(), ext='py')
     elif inp + ".py" in listdir('plugins/'):
         with open('plugins/{}.py'.format(inp)) as f:
-- 
1.9.2

