From 17dd10cf5a1d209e184053224f7be51dbcf248ac Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 8 Apr 2014 01:19:13 -0700
Subject: [PATCH] Set config indent to 4.


diff --git a/cloudbot/core/config.py b/cloudbot/core/config.py
index 1191fb9..b15f4aa 100644
--- a/cloudbot/core/config.py
+++ b/cloudbot/core/config.py
@@ -70,7 +70,7 @@ class Config(dict):
 
     def save_config(self):
         """saves the contents of the config dict to the config file"""
-        json.dump(self, open(self.path, 'w'), sort_keys=True, indent=2)
+        json.dump(self, open(self.path, 'w'), sort_keys=True, indent=4)
         self.logger.info("Config saved to file.")
 
 
-- 
1.9.3

