From 88d58b0283b2c6af50fc37213a3e8f8cd71cdbb2 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 23 Apr 2014 16:42:08 -0700
Subject: [PATCH] Add a has_run attribute to Plugin for use in determining test
 completeness


diff --git a/core/main.py b/core/main.py
index 06e7f60..3c474ba 100644
--- a/core/main.py
+++ b/core/main.py
@@ -342,6 +342,8 @@ def dispatch(bot, input, plugin):
             input.notice(input.conn.config["command_prefix"] + plugin.name + " requires additional arguments.")
         return
 
+    plugin.has_run = True
+
     if plugin.run_sync:
         yield from run_sync(bot, plugin, input)
     elif plugin.single_thread:
diff --git a/core/pluginmanager.py b/core/pluginmanager.py
index 61506b3..6f472c0 100644
--- a/core/pluginmanager.py
+++ b/core/pluginmanager.py
@@ -399,6 +399,7 @@ class Plugin:
             logging.getLogger("cloudbot").warning("Ignoring extra args {} from {}:{}".format(
                 func_hook.kwargs, self.module.title, self.function_name
             ))
+        self.has_run = False
 
     def __repr__(self):
         return "type: {}, module: {}, ignore_bots: {}, permissions: {}, single_thread: {}, run_sync: {}".format(
-- 
1.9.2

