From f505efc4bc9f8235e408ca4c97671c78a839276c Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 23 Apr 2014 16:42:08 -0700
Subject: [PATCH] Add a has_run attribute to Plugin for use in determining test
 completeness


diff --git a/core/main.py b/core/main.py
index 2c3580b..f1155f4 100644
--- a/core/main.py
+++ b/core/main.py
@@ -349,6 +349,7 @@ def dispatch(bot, input, plugin):
             input.notice(input.conn.config["command_prefix"] + plugin.name + " requires additional arguments.")
         return False
 
+    plugin.has_run = True
     if plugin.threaded:
         if plugin.single_thread:
             if plugin.module.title in bot.threads:
diff --git a/core/pluginmanager.py b/core/pluginmanager.py
index 768c85d..14dec9e 100644
--- a/core/pluginmanager.py
+++ b/core/pluginmanager.py
@@ -406,6 +406,7 @@ class Plugin:
             logging.getLogger("cloudbot").warning("Ignoring extra args {} from {}:{}".format(
                 func_hook.kwargs, self.module.title, self.function_name
             ))
+        self.has_run = False
 
     def __repr__(self):
         return "type: {}, module: {}, ignore_bots: {}, permissions: {}, single_thread: {}, threaded: {}".format(
-- 
1.9.2

