From e1d9845025d9633fa5ddd6ee8abc62d146ace4f4 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 8 Jun 2014 17:22:56 -0700
Subject: [PATCH] Add logger property to event


diff --git a/cloudbot/core/events.py b/cloudbot/core/events.py
index c9a1a20..e98521e 100644
--- a/cloudbot/core/events.py
+++ b/cloudbot/core/events.py
@@ -189,6 +189,10 @@ class BaseEvent:
         """
         return self.bot.loop
 
+    @property
+    def logger(self):
+        return self.bot.logger
+
     def message(self, message, target=None):
         """sends a message to a specific or current channel/user
         :type message: str
-- 
2.0.0

