From 6ec2d596a5c7f026b659f1e8072d8e262588579e Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Thu, 29 May 2014 23:20:24 -0700
Subject: [PATCH] Documentation updates! Tons of them!

(also merges kill.py into attacks.py)

diff --git a/cloudbot/core/events.py b/cloudbot/core/events.py
index 53af157..24beac0 100644
--- a/cloudbot/core/events.py
+++ b/cloudbot/core/events.py
@@ -212,10 +212,12 @@ class CommandEvent(BaseEvent):
         """sends a notice containing this command's docstring to the current channel/user or a specific channel/user
         :type target: str
         """
+        if self.triggered_command is None:
+            raise ValueError("Triggered command not set on this event")
         if self.hook.doc is None:
-            message = self.conn.config["command_prefix"] + self.hook.name + " requires additional arguments."
+            message = self.conn.config["command_prefix"] + self.triggered_command + " requires additional arguments."
         else:
-            message = self.conn.config["command_prefix"] + self.hook.name + self.hook.doc
+            message = self.conn.config["command_prefix"] + self.triggered_command + self.hook.doc
         self.notice(message, target=target)
 
 
@@ -247,4 +249,4 @@ class RegexEvent(BaseEvent):
         super().__init__(bot=bot, conn=conn, hook=hook, base_event=base_event, irc_raw=irc_raw, irc_prefix=irc_prefix,
                          irc_command=irc_command, irc_paramlist=irc_paramlist, irc_message=irc_message, nick=nick,
                          user=user, host=host, mask=mask)
-        self.match = match
\ No newline at end of file
+        self.match = match
diff --git a/plugins/admin.py b/plugins/admin.py
index 8562c3a..726726d 100644
--- a/plugins/admin.py
+++ b/plugins/admin.py
@@ -7,8 +7,8 @@ from cloudbot import hook
 @asyncio.coroutine
 @hook.command(["groups", "listgroups", "permgroups"], permissions=["permissions_users"], autohelp=False)
 def get_permission_groups(conn):
-    """groups -- lists all valid groups
-    :type conn: core.irc.BotConnection
+    """- lists all valid groups
+    :type conn: cloudbot.core.connection.BotConnection
     """
     return "Valid groups: {}".format(conn.permissions.get_groups())
 
@@ -16,9 +16,9 @@ def get_permission_groups(conn):
 @asyncio.coroutine
 @hook.command("gperms", permissions=["permissions_users"])
 def get_group_permissions(text, conn, notice):
-    """gperms <group> -- lists permissions of a group
+    """<group> - lists permissions given to <group>
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     """
     group = text.strip().lower()
     permission_manager = conn.permissions
@@ -35,9 +35,9 @@ def get_group_permissions(text, conn, notice):
 @asyncio.coroutine
 @hook.command("gusers", permissions=["permissions_users"])
 def get_group_users(text, conn, notice):
-    """gusers <group> -- lists users in a group
+    """<group> - lists users in <group>
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     """
     group = text.strip().lower()
     permission_manager = conn.permissions
@@ -54,9 +54,9 @@ def get_group_users(text, conn, notice):
 @asyncio.coroutine
 @hook.command("uperms", autohelp=False)
 def get_user_permissions(text, conn, mask, has_permission, notice):
-    """uperms [user] -- lists all permissions given to a user, or the current user if none is given
+    """[user] - lists all permissions given to [user], or the caller if no user is specified
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     :type mask: str
     """
     if text:
@@ -79,9 +79,9 @@ def get_user_permissions(text, conn, mask, has_permission, notice):
 @asyncio.coroutine
 @hook.command("ugroups", autohelp=False)
 def get_user_groups(text, conn, mask, has_permission, notice):
-    """uperms [user] -- lists all permissions given to a user, or the current user if none is given
+    """[user] - lists all permissions given to [user], or the caller if no user is specified
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     :type mask: str
     """
     if text:
@@ -104,10 +104,10 @@ def get_user_groups(text, conn, mask, has_permission, notice):
 @asyncio.coroutine
 @hook.command("deluser", permissions=["permissions_users"])
 def remove_permission_user(text, bot, conn, notice, reply):
-    """deluser <user> [group] -- Removes a user from a permission group, or all permission groups if none is specified
+    """<user> [group] - removes <user> from [group], or from all groups if no group is specified
     :type text: str
-    :type bot: core.bot.CloudBot
-    :type conn: core.irc.BotConnection
+    :type bot: cloudbot.core.bot.CloudBot
+    :type conn: cloudbot.core.connection.BotConnection
     """
     split = text.split()
     if len(split) > 2:
@@ -160,10 +160,10 @@ def remove_permission_user(text, bot, conn, notice, reply):
 @asyncio.coroutine
 @hook.command("adduser", permissions=["permissions_users"])
 def add_permissions_user(text, conn, bot, notice, reply):
-    """adduser <user> <group> -- Adds a user to a permission group
+    """<user> <group> - adds <user> to <group>
     :type text: str
-    :type conn: core.irc.BotConnection
-    :type bot: core.bot.CloudBot
+    :type conn: cloudbot.core.connection.BotConnection
+    :type bot: cloudbot.core.bot.CloudBot
     """
     split = text.split()
     if len(split) > 2:
@@ -201,7 +201,7 @@ def add_permissions_user(text, conn, bot, notice, reply):
 
 @hook.command(["stop", "quit"], permissions=["botcontrol"], autohelp=False)
 def stop(text, bot):
-    """stop [reason] -- Stops the bot with [reason] as its quit message.
+    """[reason] - stops me with [reason] as its quit message.
     :type text: str
     :type bot: core.bot.CloudBot
     """
@@ -213,7 +213,7 @@ def stop(text, bot):
 
 @hook.command(permissions=["botcontrol"], autohelp=False)
 def restart(text, bot):
-    """restart [reason] -- Restarts the bot with [reason] as its quit message.
+    """[reason] - restarts me with [reason] as its quit message.
     :type text: str
     :type bot: core.bot.CloudBot
     """
@@ -226,9 +226,9 @@ def restart(text, bot):
 @asyncio.coroutine
 @hook.command(permissions=["botcontrol"])
 def join(text, conn, notice):
-    """join <channel> -- Joins a given channel
+    """<channel> - joins <channel>
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     """
     for target in text.split():
         if not target.startswith("#"):
@@ -240,9 +240,9 @@ def join(text, conn, notice):
 @asyncio.coroutine
 @hook.command(permissions=["botcontrol"], autohelp=False)
 def part(text, conn, chan, notice):
-    """part [channel] -- Leaves a given channel, or the current one if no channel is specified
+    """[#channel] - parts [#channel], or the caller's channel if no channel is specified
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     :type chan: str
     """
     if text:
@@ -259,9 +259,9 @@ def part(text, conn, chan, notice):
 @asyncio.coroutine
 @hook.command(autohelp=False, permissions=["botcontrol"])
 def cycle(text, conn, chan, notice):
-    """cycle <channel> -- Cycles a given channel, or the current one if no channel is specified
+    """[#channel] - cycles [#channel], or the caller's channel if no channel is specified
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     :type chan: str
     """
     if text:
@@ -279,9 +279,9 @@ def cycle(text, conn, chan, notice):
 @asyncio.coroutine
 @hook.command(permissions=["botcontrol"])
 def nick(text, conn, notice):
-    """nick <nick> -- Changes the bot's nickname
+    """<nick> - changes my nickname to <nick>
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     """
     if not re.match("^[a-z0-9_|.-\]\[]*$", text.lower()):
         notice("Invalid username '{}'".format(text))
@@ -293,9 +293,9 @@ def nick(text, conn, notice):
 @asyncio.coroutine
 @hook.command(permissions=["botcontrol"])
 def raw(text, conn, notice):
-    """raw <command> -- Sends a irc_raw IRC command
+    """<command> - sends <command> as a raw IRC command
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     """
     notice("Raw command sent.")
     conn.send(text)
@@ -304,9 +304,9 @@ def raw(text, conn, notice):
 @asyncio.coroutine
 @hook.command(permissions=["botcontrol"])
 def say(text, conn, chan):
-    """say [channel] <message> -- Makes the bot say <message> in [channel], or the current channel if none is specified
+    """[#channel] <message> - says <message> to [#channel], or to the caller's channel if no channel is specified
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     :type chan: str
     """
     text = text.strip()
@@ -321,11 +321,11 @@ def say(text, conn, chan):
 
 
 @asyncio.coroutine
-@hook.command(permissions=["botcontrol"])
+@hook.command(["message", "sayto"], permissions=["botcontrol"])
 def message(text, conn):
-    """message <name> <message> -- Makes the bot say <message> to <name>, <name> may be a #channel or a nickname
+    """<name> <message> - says <message> to <name>
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     """
     split = text.split(None, 1)
     channel = split[0]
@@ -336,9 +336,9 @@ def message(text, conn):
 @asyncio.coroutine
 @hook.command(["me", "act"], permissions=["botcontrol"])
 def me(text, conn, chan):
-    """me [channel] <action> -- Makes the bot act out <action> in a [channel], or the current channel if none is given
+    """[#channel] <action> - acts out <action> in a [#channel], or in the current channel of none is specified
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     :type chan: str
     """
     text = text.strip()
diff --git a/plugins/attacks.py b/plugins/attacks.py
index 00edafe..72d6973 100644
--- a/plugins/attacks.py
+++ b/plugins/attacks.py
@@ -1,25 +1,37 @@
+import codecs
+import json
+import os
 import random
 import asyncio
 import re
 
 from cloudbot import hook
+from cloudbot.util import textgen
 
-with open("data/larts.txt") as f:
-    larts = [line.strip() for line in f.readlines()
-             if not line.startswith("//")]
 
-with open("data/insults.txt") as f:
-    insults = [line.strip() for line in f.readlines()
-               if not line.startswith("//")]
+@hook.onload()
+def load_attacks(bot):
+    """
+    :type bot: cloudbot.core.bot.CloudBot
+    """
+    global larts, insults, flirts, kills
+
+    with codecs.open(os.path.join(bot.data_dir, "larts.txt"), encoding="utf-8") as f:
+        larts = [line.strip() for line in f.readlines() if not line.startswith("//")]
+
+    with codecs.open(os.path.join(bot.data_dir, "insults.txt"), encoding="utf-8") as f:
+        insults = [line.strip() for line in f.readlines() if not line.startswith("//")]
 
-with open("data/flirts.txt") as f:
-    flirts = [line.strip() for line in f.readlines()
-              if not line.startswith("//")]
+    with codecs.open(os.path.join(bot.data_dir, "flirts.txt"), encoding="utf-8") as f:
+        flirts = [line.strip() for line in f.readlines() if not line.startswith("//")]
+
+    with codecs.open(os.path.join(bot.data_dir, "kills.json"), encoding="utf-8") as f:
+        kills = json.load(f)
 
 
 def is_self(conn, target):
     """
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     :type target: str
     """
     if re.search("(^..?.?.?self|{})".format(re.escape(conn.nick.lower())), target.lower()):
@@ -31,9 +43,9 @@ def is_self(conn, target):
 @asyncio.coroutine
 @hook.command()
 def lart(text, conn, nick, notice, action):
-    """lart <user> -- LARTs <user>.
+    """<user> - LARTs <user>
     :type text: str
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.irc.BotConnection
     :type nick: str
     """
     target = text.strip()
@@ -55,7 +67,7 @@ def lart(text, conn, nick, notice, action):
 @asyncio.coroutine
 @hook.command()
 def insult(text, conn, nick, notice, message):
-    """insult <user> -- Makes the bot insult <user>.
+    """<user> - insults <user>
     :type text: str
     :type conn: core.irc.BotConnection
     :type nick: str
@@ -76,7 +88,7 @@ def insult(text, conn, nick, notice, message):
 @asyncio.coroutine
 @hook.command()
 def flirt(text, conn, nick, notice, message):
-    """flirt <user> -- Makes the bot flirt with <user>.
+    """<user> - flirts with <user>
     :type text: str
     :type conn: core.irc.BotConnection
     :type nick: str
@@ -92,3 +104,27 @@ def flirt(text, conn, nick, notice, message):
         target = nick
 
     message('{}, {}'.format(target, random.choice(flirts)))
+
+
+@asyncio.coroutine
+@hook.command()
+def kill(text, conn, nick, notice, action):
+    """<user> - kills <user>
+    :type text: str
+    :type conn: core.irc.BotConnection
+    :type nick: str
+    """
+    target = text.strip()
+
+    if " " in target:
+        notice("Invalid username!")
+        return
+
+    # if the user is trying to make the bot kill itself, kill them
+    if is_self(conn, target):
+        target = nick
+
+    generator = textgen.TextGenerator(kills["templates"], kills["parts"], variables={"user": target})
+
+    # act out the message
+    action(generator.generate_string())
diff --git a/plugins/brainfuck.py b/plugins/brainfuck.py
index c73cecf..cc9932a 100644
--- a/plugins/brainfuck.py
+++ b/plugins/brainfuck.py
@@ -14,7 +14,7 @@ MAX_STEPS = 1000000
 @asyncio.coroutine
 @hook.command(["brainfuck", "bf"])
 def bf(text):
-    """bf <prog> -- Executes <prog> as Brainfuck code.
+    """<prog> - executes <prog> as Brainfuck code
     :type text: str
     """
 
diff --git a/plugins/choose.py b/plugins/choose.py
index c3ce1ba..2127345 100644
--- a/plugins/choose.py
+++ b/plugins/choose.py
@@ -8,7 +8,7 @@ from cloudbot import hook
 @asyncio.coroutine
 @hook.command()
 def choose(text, notice):
-    """choose <choice1>, [choice2], [choice3], etc. -- Randomly picks one of the given choices.
+    """<choice1>, [choice2], [choice3], etc. - randomly picks one of the given choices
     :type text: str
     """
     choices = re.findall(r'([^,\s]+)', text)
diff --git a/plugins/coin.py b/plugins/coin.py
index 398402a..2c8e75d 100644
--- a/plugins/coin.py
+++ b/plugins/coin.py
@@ -7,7 +7,7 @@ from cloudbot import hook
 @asyncio.coroutine
 @hook.command(autohelp=False)
 def coin(text, notice, action):
-    """coin [amount] -- Flips [amount] of coins.
+    """[amount] - flips [amount] coins
     :type text: str
     """
 
diff --git a/plugins/core_sieve.py b/plugins/core_sieve.py
index 404ffde..2329f79 100644
--- a/plugins/core_sieve.py
+++ b/plugins/core_sieve.py
@@ -8,18 +8,19 @@ buckets = {}
 
 
 @hook.sieve
-def sieve_suite(bot, event, hook):
+def sieve_suite(bot, event, _hook):
     """
     :type bot: cloudbot.core.bot.CloudBot
     :type event: cloudbot.core.events.BaseEvent
+    :type _hook: cloudbot.core.pluginmanager.Hook
     """
     conn = event.conn
     # check ignore bots
-    if event.irc_command == 'PRIVMSG' and event.nick.endswith('bot') and hook.ignore_bots:
+    if event.irc_command == 'PRIVMSG' and event.nick.endswith('bot') and _hook.ignore_bots:
         return None
 
     # check acls
-    acl = conn.config.get('acls', {}).get(hook.function_name)
+    acl = conn.config.get('acls', {}).get(_hook.function_name)
     if acl:
         if 'deny-except' in acl:
             allowed_channels = list(map(str.lower, acl['deny-except']))
@@ -31,13 +32,13 @@ def sieve_suite(bot, event, hook):
                 return None
 
     # check disabled_commands
-    if hook.type == "command":
+    if _hook.type == "command":
         disabled_commands = conn.config.get('disabled_commands', [])
         if event.triggered_command in disabled_commands:
             return None
 
     # check permissions
-    allowed_permissions = hook.permissions
+    allowed_permissions = _hook.permissions
     if allowed_permissions:
         allowed = False
         for perm in allowed_permissions:
@@ -50,7 +51,7 @@ def sieve_suite(bot, event, hook):
             return None
 
     # check command spam tokens
-    if hook.type == "command":
+    if _hook.type == "command":
         uid = event.chan
 
         if not uid in buckets:
diff --git a/plugins/cryptocoins.py b/plugins/cryptocoins.py
index c087ebc..4f9b4c6 100644
--- a/plugins/cryptocoins.py
+++ b/plugins/cryptocoins.py
@@ -31,8 +31,7 @@ exchanges = {
 
 @hook.command(["btc", "bitcoin"], autohelp=False)
 def bitcoin(text, notice):
-    """bitcoin <exchange> -- Gets current exchange rate for bitcoins from several exchanges, default is Blockchain.
-    Supports MtGox, Bitpay, Coinbase and BitStamp.
+    """[mtgox|bitpay|coinbase|bitstamp] - gets bitcoin exchange rate using <exchange>, defaulting to blockchain
     :type text: str
     """
     text = text.lower()
@@ -55,7 +54,7 @@ def bitcoin(text, notice):
 
 @hook.command(["ltc", "litecoin"], autohelp=False)
 def litecoin(message):
-    """litecoin -- gets current exchange rate for litecoins from BTC-E"""
+    """- gets litecoin exchange rate from BTC-E"""
     data = http.get_json("https://btc-e.com/api/2/ltc_usd/ticker")
     ticker = data['ticker']
     message("Current: \x0307${:,.2f}\x0f - High: \x0307${:,.2f}\x0f"
diff --git a/plugins/cypher.py b/plugins/cypher.py
index 4052c27..adc8736 100644
--- a/plugins/cypher.py
+++ b/plugins/cypher.py
@@ -39,7 +39,7 @@ def decode(password, encoded, notice):
 @asyncio.coroutine
 @hook.command()
 def cypher(text, notice):
-    """cypher <pass> <string> -- Cyphers <string> with <password>.
+    """<pass> <string> - cyphers <string> with <password>
     :type text: str
     """
     split = text.split(None, 1)
@@ -54,7 +54,7 @@ def cypher(text, notice):
 @asyncio.coroutine
 @hook.command()
 def decypher(text, notice):
-    """decypher <pass> <string> -- Decyphers <string> with <password>.
+    """<pass> <string> - decyphers <string> with <password>
     :type text: str
     """
     split = text.split(None, 1)
diff --git a/plugins/debug.py b/plugins/debug.py
index 55d5dd4..e296aae 100644
--- a/plugins/debug.py
+++ b/plugins/debug.py
@@ -4,8 +4,7 @@ import psutil
 
 from cloudbot import hook
 
-
-@hook.command
+@hook.command(autohelp=False)
 def debug():
     # get a Process object for the bot using psutil
     process = psutil.Process(os.getpid())
diff --git a/plugins/dice.py b/plugins/dice.py
index a380992..30b281f 100644
--- a/plugins/dice.py
+++ b/plugins/dice.py
@@ -38,7 +38,7 @@ def n_rolls(count, n):
 @asyncio.coroutine
 @hook.command(["roll", "dice"])
 def dice(text, notice):
-    """dice <dice roll> -- Simulates dice rolls. Example: 'dice 2d20-d5+4 roll 2': D20s, subtract 1D5, add 4
+    """<dice roll> - simulates dice rolls. Example: 'dice 2d20-d5+4 roll 2': D20s, subtract 1D5, add 4
     :type text: str
     """
 
diff --git a/plugins/dictionary.py b/plugins/dictionary.py
index 23691d2..fa1f9e1 100644
--- a/plugins/dictionary.py
+++ b/plugins/dictionary.py
@@ -44,7 +44,7 @@ def format_output(h, definition, show_examples):
 
 @hook.command(["dictionary", "define"])
 def define(text):
-    """define <word> -- Fetches definition of <word>.
+    """<word> - fetches definition of <word>
     :type text: str
     """
 
@@ -72,7 +72,7 @@ def define(text):
 
 @hook.command(["e", "etymology"])
 def etymology(text):
-    """etymology <word> -- Retrieves the etymology of <word>.
+    """<word> - retrieves the etymology of <word>
     :type text: str
     """
 
diff --git a/plugins/domainr.py b/plugins/domainr.py
index f24109e..37b62ee 100644
--- a/plugins/domainr.py
+++ b/plugins/domainr.py
@@ -20,7 +20,7 @@ def format_domain(domain):
 
 @hook.command(["domain", "domainr"])
 def domainr(text):
-    """domainr <domain> - Use domain.nr's API to search for a domain, and similar domains.
+    """<domain> - uses domain.nr's API to search for a domain, and similar domains
     :type text: str
     """
     try:
diff --git a/plugins/drama.py b/plugins/drama.py
index ac7940e..35a80d4 100644
--- a/plugins/drama.py
+++ b/plugins/drama.py
@@ -9,8 +9,7 @@ ed_url = "http://encyclopediadramatica.se/"
 
 @hook.command()
 def drama(text):
-    """drama <phrase> -- Gets the first paragraph of
-    the Encyclopedia Dramatica article on <phrase>."""
+    """<phrase> - gets the first paragraph of the Encyclopedia Dramatica article on <phrase>"""
 
     data = http.get_json(api_url, search=text)
 
diff --git a/plugins/eightball.py b/plugins/eightball.py
index 2dc8073..a6fa25f 100644
--- a/plugins/eightball.py
+++ b/plugins/eightball.py
@@ -23,7 +23,7 @@ def load_responses(bot):
 @asyncio.coroutine
 @hook.command(["8ball", "8", "eightball"])
 def eightball(action):
-    """8ball <question> -- The all knowing magic eight ball, in electronic form. Ask and it shall be answered!"""
+    """<question> - asks the all knowing magic electronic eight ball <question>"""
 
     magic = formatting.multiword_replace(random.choice(responses), color_codes)
     action("shakes the magic 8 ball... {}".format(magic))
diff --git a/plugins/encrypt.py b/plugins/encrypt.py
index c6f1d5e..f78e10f 100644
--- a/plugins/encrypt.py
+++ b/plugins/encrypt.py
@@ -64,7 +64,7 @@ def get_salt(bot):
 @asyncio.coroutine
 @hook.command()
 def encrypt(text, bot, db, notice):
-    """encrypt <pass> <string> -- Encrypts <string> with <pass>. (<string> can only be decrypted using this bot)
+    """<pass> <string> - encrypts <string> with <pass>. (<string> can only be decrypted using this bot)
     :type text: str
     :type bot: core.bot.CloudBot
     :type db: sqlalchemy.orm.session.Session
@@ -103,8 +103,8 @@ def encrypt(text, bot, db, notice):
 @asyncio.coroutine
 @hook.command()
 def decrypt(text, bot, db, notice):
-    """decrypt <pass> <string> -- Decrypts <string> with <pass>. (can only decrypt strings encrypted on this bot)
-    :type bot: core.bot.CloudBot
+    """<pass> <string> - decrypts <string> with <pass>. (can only decrypt strings encrypted on this bot)
+    :type bot: cloudbot.core.bot.CloudBot
     :type db: sqlalchemy.orm.session.Session
     """
 
diff --git a/plugins/fact.py b/plugins/fact.py
index 95c0c0e..4b2386f 100644
--- a/plugins/fact.py
+++ b/plugins/fact.py
@@ -3,7 +3,7 @@ from cloudbot import hook, http, web
 
 @hook.command(autohelp=False)
 def fact():
-    """fact -- Gets a random fact from OMGFACTS."""
+    """- gets a random fact from OMGFACTS"""
 
     attempts = 0
 
diff --git a/plugins/factoids.py b/plugins/factoids.py
index 1de18d5..2afe29b 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -73,7 +73,7 @@ def del_factoid(db, word):
 @asyncio.coroutine
 @hook.command(["r", "remember"], permissions=["addfactoid"])
 def remember(text, nick, db, notice):
-    """remember <word> [+]<data> -- Remembers <data> with <word>. Add + to <data> to append."""
+    """<word> [+]<data> - remembers <data> with <word> - add + to <data> to append"""
 
     append = False
 
@@ -108,7 +108,7 @@ def remember(text, nick, db, notice):
 
 @hook.command(["f", "forget"], permissions=["delfactoid"])
 def forget(text, db, notice):
-    """forget <word> -- Forgets a remembered <word>."""
+    """<word> - forgets previously remembered <word>"""
 
     data = factoid_cache.get(text)
 
@@ -124,7 +124,7 @@ def forget(text, db, notice):
 @asyncio.coroutine
 @hook.command()
 def info(text, notice):
-    """info <factoid> -- Shows the source of a factoid."""
+    """<factoid> - shows the source of a factoid"""
 
     text = text.strip()
 
@@ -137,7 +137,7 @@ def info(text, notice):
 @asyncio.coroutine
 @hook.regex(r'^\? ?(.+)')
 def factoid(inp, input, db, message, action):
-    """?<word> -- Shows what data is associated with <word>."""
+    """<word> - shows what data is associated with <word>"""
 
     # split up the input
     split = inp.group(1).strip().split(" ")
@@ -154,8 +154,8 @@ def factoid(inp, input, db, message, action):
         if data.startswith("<py>"):
             code = data[4:].strip()
             variables = 'input="""{}"""; nick="{}"; chan="{}"; bot_nick="{}";'.format(arguments.replace('"', '\\"'),
-                                                                                      input.nick, input.chan,
-                                                                                      input.conn.nick)
+                                                                                      event.nick, event.chan,
+                                                                                      event.conn.nick)
             result = pyexec.eval_py(variables + code)
         else:
             result = data
@@ -178,16 +178,17 @@ def factoid(inp, input, db, message, action):
 
 @asyncio.coroutine
 @hook.command(autohelp=False, permissions=["listfactoids"])
-def listfactoids(reply):
+def listfactoids(notice):
+    """- lists all available factoids"""
     reply_text = []
     reply_text_length = 0
     for word in factoid_cache.keys():
         added_length = len(word) + 2
         if reply_text_length + added_length > 400:
-            reply(", ".join(reply_text))
+            notice(", ".join(reply_text))
             reply_text = []
             reply_text_length = 0
         else:
             reply_text.append(word)
             reply_text_length += added_length
-    return ", ".join(reply_text)
+    notice(", ".join(reply_text))
diff --git a/plugins/fishbans.py b/plugins/fishbans.py
index 4b96ee6..a16d748 100644
--- a/plugins/fishbans.py
+++ b/plugins/fishbans.py
@@ -13,7 +13,7 @@ api_url = "http://api.fishbans.com/stats/{}/"
 @hook.command(["bans", "fishbans"], async=True)
 @asyncio.coroutine
 def fishbans(text, loop):
-    """fishbans <user> -- Gets information on <user>s minecraft bans from fishbans"""
+    """<user> - gets information on <user>'s minecraft bans from fishbans"""
     user = text.strip()
 
     try:
@@ -44,7 +44,7 @@ def fishbans(text, loop):
 @hook.command(async=True)
 @asyncio.coroutine
 def bancount(text, loop):
-    """bancount <user> -- Gets a count of <user>s minecraft bans from fishbans"""
+    """<user> - gets a count of <user>'s minecraft bans from fishbans"""
     user = text.strip()
 
     try:
diff --git a/plugins/fmylife.py b/plugins/fmylife.py
index 86a63c7..fe24b88 100644
--- a/plugins/fmylife.py
+++ b/plugins/fmylife.py
@@ -7,10 +7,10 @@ from cloudbot import hook
 
 fml_cache = []
 
-
-def refresh_cache():
+@asyncio.coroutine
+def refresh_cache(loop):
     """ gets a page of random FMLs and puts them into a dictionary """
-    response = requests.get('http://www.fmylife.com/random/')
+    response = yield from loop.run_in_executor(None, requests.get, 'http://www.fmylife.com/random/')
     soup = BeautifulSoup(response.text)
 
     for e in soup.find_all('div', {'class': 'post article'}):
@@ -19,16 +19,17 @@ def refresh_cache():
         fml_cache.append((fml_id, text))
 
 
+@asyncio.coroutine
 @hook.onload()
-def initial_refresh():
+def initial_refresh(loop):
     # do an initial refresh of the cache
-    refresh_cache()
+    yield from refresh_cache(loop)
 
 
 @asyncio.coroutine
 @hook.command(autohelp=False)
 def fml(reply, loop):
-    """fml -- Gets a random quote from fmyfife.com."""
+    """- gets a random quote from fmyfife.com"""
 
     # grab the last item in the fml cache and remove it
     fml_id, text = fml_cache.pop()
@@ -36,4 +37,4 @@ def fml(reply, loop):
     reply('(#{}) {}'.format(fml_id, text))
     # refresh fml cache if its getting empty
     if len(fml_cache) < 3:
-        yield from loop.run_in_executor(None, refresh_cache)
+        yield from refresh_cache(loop)
diff --git a/plugins/foods.py b/plugins/foods.py
index 73a8035..7891688 100644
--- a/plugins/foods.py
+++ b/plugins/foods.py
@@ -1,4 +1,5 @@
 import re
+import asyncio
 import random
 
 from cloudbot import hook
@@ -46,9 +47,10 @@ potatoes = ['AC Belmont', 'AC Blue Pride', 'AC Brador', 'AC Chaleur', 'AC Domino
             'Yukon Gold']
 
 
-@hook.command
-def potato(text, action=None):
-    """potato <user> - Makes <user> a tasty little potato."""
+@asyncio.coroutine
+@hook.command()
+def potato(text, action):
+    """<user> - makes <user> a tasty little potato"""
     text = text.strip()
 
     if not re.match("^[A-Za-z0-9_|.-\]\[]*$", text.lower()):
@@ -63,9 +65,10 @@ def potato(text, action=None):
     action("{} a {} {} {} potato for {} and serves it with a small {}!".format(method, flavor, size, potato_type, text,
                                                                                side_dish))
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def cake(text, action):
-    """cake <user> - Gives <user> an awesome cake."""
+    """<user> - gives <user> an awesome cake"""
     text = text.strip()
 
     if not re.match("^[A-Za-z0-9_|.-\]\[]*$", text.lower()):
@@ -82,9 +85,10 @@ def cake(text, action):
                                                                          side_dish))
 
 
-@hook.command
+@asyncio.coroutine
+@hook.command()
 def cookie(text, action):
-    """cookie <user> - Gives <user> a cookie"""
+    """<user> - gives <user> a cookie"""
     text = text.strip()
 
     if not re.match("^[A-Za-z0-9_|.-\]\[]*$", text.lower()):
diff --git a/plugins/fortune.py b/plugins/fortune.py
index df84421..b85ff7c 100644
--- a/plugins/fortune.py
+++ b/plugins/fortune.py
@@ -10,12 +10,11 @@ def load_fortunes(bot):
     path = os.path.join(bot.data_dir, "fortunes.txt")
     global fortunes
     with open(path) as f:
-        fortunes = [line.strip() for line in f.readlines()
-                    if not line.startswith("//")]
+        fortunes = [line.strip() for line in f.readlines() if not line.startswith("//")]
 
 
 @asyncio.coroutine
 @hook.command(autohelp=False)
 def fortune():
-    """fortune -- Fortune cookies on demand."""
+    """- hands out a fortune cookie"""
     return random.choice(fortunes)
diff --git a/plugins/geoip.py b/plugins/geoip.py
index 921397a..cb8c7da 100644
--- a/plugins/geoip.py
+++ b/plugins/geoip.py
@@ -34,7 +34,7 @@ def load_regions(bot):
 
 @hook.command()
 def geoip(text):
-    """geoip <host/ip> -- Gets the location of <host/ip>"""
+    """<host/ip> - gets the location of <host/ip>"""
 
     try:
         record = geo.record_by_name(text)
@@ -48,7 +48,7 @@ def geoip(text):
         # it's a lazy patch, but it should do the job
         try:
             data["region"] = ", " + regions[record["country_code"]][record["region_name"]]
-        except:
+        except Exception:
             data["region"] = ""
     else:
         data["region"] = ""
@@ -56,4 +56,5 @@ def geoip(text):
     data["cc"] = record["country_code"] or "N/A"
     data["country"] = record["country_name"] or "Unknown"
     data["city"] = record["city"] or "Unknown"
+
     return "\x02Country:\x02 {country} ({cc}), \x02City:\x02 {city}{region}".format(**data)
diff --git a/plugins/github.py b/plugins/github.py
index 69e61f0..ea60828 100644
--- a/plugins/github.py
+++ b/plugins/github.py
@@ -7,9 +7,9 @@ shortcuts = {
 }
 
 
-@hook.command
+@hook.command()
 def issues(text):
-    """issues <repo> [issue] - Get issues for <repo>. If [issue] is specified, summarize that issue."""
+    """<username|repo> [number] - gets issue [number]'s summary, or the open issue count if no issue is specified"""
     args = text.split()
     repo = args[0] if args[0] not in shortcuts else shortcuts[args[0]]
     issue = args[1] if len(args) > 1 else None
diff --git a/plugins/google.py b/plugins/google.py
index d7a8db3..e3f1172 100644
--- a/plugins/google.py
+++ b/plugins/google.py
@@ -12,7 +12,7 @@ def api_get(kind, query):
 
 @hook.command(["googleimage", "gis", "image"])
 def googleimage(text):
-    """gis <query> -- Returns first Google Image result for <query>."""
+    """<query> - returns the first google image result for <query>"""
 
     parsed = api_get('images', text)
     if not 200 <= parsed['responseStatus'] < 300:
@@ -24,7 +24,7 @@ def googleimage(text):
 
 @hook.command(["google", "g", "search"])
 def google(text):
-    """google <query> -- Returns first google search result for <query>."""
+    """<query> - returns the first google search result for <query>"""
 
     parsed = api_get('web', text)
     if not 200 <= parsed['responseStatus'] < 300:
diff --git a/plugins/google_translate.py b/plugins/google_translate.py
index 6c10ab2..cdd2ad2 100644
--- a/plugins/google_translate.py
+++ b/plugins/google_translate.py
@@ -75,9 +75,7 @@ def match_language(fragment):
 
 @hook.command()
 def translate(text, bot):
-    """translate [source language [target language]] <sentence> -- translates
-    <sentence> from source language (default autodetect) to target
-    language (default English) using Google Translate"""
+    """[source language [target language]] <sentence> - translates <sentence> from source language (default autodetect) to target language (default English) using Google Translate"""
 
     api_key = bot.config.get("api_keys", {}).get("googletranslate", None)
     if not api_key:
diff --git a/plugins/googleurlparse.py b/plugins/googleurlparse.py
index 7271bfe..b25694b 100644
--- a/plugins/googleurlparse.py
+++ b/plugins/googleurlparse.py
@@ -5,9 +5,7 @@ from cloudbot import hook
 
 @hook.command(autohelp=False)
 def googleurl(text, db, nick):
-    """googleurl [nickname] - Converts Google urls (google.com/url) to normal urls
-       where possible, in the specified nickname's last message. If nickname isn't provided,
-       action will be performed on user's last message"""
+    """[nickname] - converts Google urls (google.com/url) to normal urls where possible, [nickname]'s last message, defaulting to the caller's last message if no nickname is specified"""
     if not text:
         text = nick
     last_message = db.execute("select name, quote from seen_user where name"
diff --git a/plugins/help.py b/plugins/help.py
index cc8a6a8..30029f6 100644
--- a/plugins/help.py
+++ b/plugins/help.py
@@ -8,10 +8,10 @@ from cloudbot import hook
 @asyncio.coroutine
 @hook.command("help", autohelp=False)
 def help_command(text, conn, bot, notice, has_permission):
-    """help  -- Gives a list of commands/help for a command.
+    """[command] - gives help for [command], or lists all available commands if no command is specified
     :type text: str
-    :type conn: core.irc.BotConnection
-    :type bot: core.bot.CloudBot
+    :type conn: cloudbot.core.connection.BotConnection
+    :type bot: cloudbot.core.bot.CloudBot
     """
     if text:
         searching_for = text.lower().strip()
diff --git a/plugins/history.py b/plugins/history.py
index add40d0..6a5b48b 100644
--- a/plugins/history.py
+++ b/plugins/history.py
@@ -53,7 +53,7 @@ def chat_tracker(event, db, conn):
 @asyncio.coroutine
 @hook.command(autohelp=False)
 def resethistory(event, conn):
-    """resethistory - Resets chat history for the current channel"""
+    """- resets chat history for the current channel"""
     try:
         conn.history[event.chan].clear()
         return "Reset chat history for current channel."
@@ -62,9 +62,9 @@ def resethistory(event, conn):
         return "There is no history for this channel."
 
 
-@hook.command
+@hook.command()
 def seen(text, nick, chan, db, event, conn):
-    """seen <nick> <channel> -- Tell when a nickname was last in active in one of this bot's channels."""
+    """<nick> <channel> - tells when a nickname was last in active in one of my channels"""
 
     if event.conn.nick.lower() == text.lower():
         return "You need to get your eyes checked."
@@ -77,8 +77,8 @@ def seen(text, nick, chan, db, event, conn):
 
     db_init(db, conn.name)
 
-    last_seen = db.execute("select name, time, quote from seen_user where name"
-                           " like :name and chan = :chan", {'name': text, 'chan': chan}).fetchone()
+    last_seen = db.execute("select name, time, quote from seen_user where name like :name and chan = :chan",
+                           {'name': text, 'chan': chan}).fetchone()
 
     if last_seen:
         reltime = timesince.timesince(last_seen[1])
diff --git a/plugins/horoscope.py b/plugins/horoscope.py
index 394cb49..1eff2b9 100644
--- a/plugins/horoscope.py
+++ b/plugins/horoscope.py
@@ -11,7 +11,7 @@ def init(db):
 
 @hook.command(autohelp=False)
 def horoscope(text, db, notice, nick):
-    """horoscope <sign> -- Get your horoscope."""
+    """<sign> - get your horoscope"""
 
     # check if the user asked us not to save his details
     dontsave = text.endswith(" dontsave")
diff --git a/plugins/hulu.py b/plugins/hulu.py
index c12eaf4..3bf7973 100644
--- a/plugins/hulu.py
+++ b/plugins/hulu.py
@@ -16,7 +16,7 @@ def hulu_url(match):
 
 @hook.command('hulu')
 def hulu_search(text):
-    """hulu <search> - Search Hulu"""
+    """<query> - searches Hulu for <query>"""
     result = http.get_soup(
         "http://m.hulu.com/search?dp_identifier=hulu&{}&items_per_page=1&page=1".format(urlencode({'query': text})))
     data = result.find('results').find('videos').find('video')
diff --git a/plugins/ignore.py b/plugins/ignore.py
index c324d33..fed4128 100644
--- a/plugins/ignore.py
+++ b/plugins/ignore.py
@@ -55,7 +55,7 @@ def ignore_sieve(bot, event, _hook):
 @asyncio.coroutine
 @hook.command(autohelp=False)
 def ignored(notice, conn):
-    """ignored -- Lists ignored channels/users."""
+    """- lists all channels and users I'm ignoring"""
     ignorelist = conn.config["plugins"]["ignore"]["ignored"]
     if ignorelist:
         notice("Ignored channels/users are: {}".format(", ".join(ignorelist)))
@@ -66,7 +66,7 @@ def ignored(notice, conn):
 
 @hook.command(permissions=["ignore"])
 def ignore(text, bot, conn, notice):
-    """ignore <channel|nick|host> -- Makes the bot ignore <channel|user>."""
+    """<channel|nick|usermask> - adds <channel|nick> to my ignore list"""
     target = text.lower()
     if "!" not in target or "@" not in target:
         target = "{}!*@*".format(target)
@@ -83,7 +83,7 @@ def ignore(text, bot, conn, notice):
 
 @hook.command(permissions=["ignore"])
 def unignore(text, bot, conn, notice):
-    """unignore <channel|user> -- Makes the bot listen to <channel|user>."""
+    """<channel|nick|usermask> - removes <channel|nick|usermask> from my ignore list"""
     target = text.lower()
     if "!" not in target or "@" not in target:
         target = "{}!*@*".format(target)
diff --git a/plugins/imdb.py b/plugins/imdb.py
index 20a05a7..87dd83f 100644
--- a/plugins/imdb.py
+++ b/plugins/imdb.py
@@ -10,7 +10,7 @@ imdb_re = (r'(.*:)//(imdb.com|www.imdb.com)(:[0-9]+)?(.*)', re.I)
 
 @hook.command()
 def imdb(text):
-    """imdb <movie> -- Gets information about <movie> from IMDb."""
+    """imdb <movie> - gets information about <movie> from IMDb"""
 
     strip = text.strip()
 
diff --git a/plugins/imgur.py b/plugins/imgur.py
index 61eede9..f951e47 100644
--- a/plugins/imgur.py
+++ b/plugins/imgur.py
@@ -18,8 +18,7 @@ def is_valid(data):
 
 @hook.command(autohelp=False)
 def imgur(text):
-    """imgur [subreddit] -- Gets the first page of imgur images from [subreddit] and returns a link to them.
-     If [subreddit] is undefined, return any imgur images"""
+    """[subreddit] - returns a link to the first page of imgur images from [subreddit], or the first page of all imgur images if no subreddit is provided"""
     if text:
         # see if the input ends with "nsfw"
         show_nsfw = text.endswith(" nsfw")
diff --git a/plugins/kernel.py b/plugins/kernel.py
index d9c0043..ae8434d 100644
--- a/plugins/kernel.py
+++ b/plugins/kernel.py
@@ -5,6 +5,7 @@ from cloudbot import hook, http
 
 @hook.command(autohelp=False)
 def kernel(reply):
+    """- gets a list of linux kernel versions"""
     contents = http.get("https://www.kernel.org/finger_banner")
     contents = re.sub(r'The latest(\s*)', '', contents)
     contents = re.sub(r'version of the Linux kernel is:(\s*)', '- ', contents)
diff --git a/plugins/kill.py b/plugins/kill.py
deleted file mode 100644
index 9c491f9..0000000
--- a/plugins/kill.py
+++ /dev/null
@@ -1,35 +0,0 @@
-import asyncio
-import json
-
-from cloudbot import hook, textgen
-
-
-def get_generator(_json, variables):
-    data = json.loads(_json)
-    return textgen.TextGenerator(data["templates"],
-                                 data["parts"], variables=variables)
-
-
-@asyncio.coroutine
-@hook.command()
-def kill(text, action=None, nick=None, conn=None, notice=None):
-    """kill <user> -- Makes the bot kill <user>."""
-    target = text.strip()
-
-    if " " in target:
-        notice("Invalid username!")
-        return
-
-    # if the user is trying to make the bot kill itself, kill them
-    if target.lower() == conn.nick.lower() or target.lower() == "itself":
-        target = nick
-
-    variables = {
-        "user": target
-    }
-
-    with open("./data/kills.json") as f:
-        generator = get_generator(f.read(), variables)
-
-    # act out the message
-    action(generator.generate_string())
diff --git a/plugins/lastfm.py b/plugins/lastfm.py
index 27e16ea..4b89fbd 100644
--- a/plugins/lastfm.py
+++ b/plugins/lastfm.py
@@ -7,7 +7,7 @@ api_url = "http://ws.audioscrobbler.com/2.0/?format=json"
 
 @hook.command(["lastfm", "l"], autohelp=False)
 def lastfm(text, nick, db, bot, notice):
-    """lastfm [user] [dontsave] -- Displays the now playing (or last played) track of LastFM user [user]."""
+    """[user] [dontsave] - displays the now playing (or last played) track of LastFM user [user]"""
     api_key = bot.config.get("api_keys", {}).get("lastfm")
     if not api_key:
         return "error: no api key set"
diff --git a/plugins/lmgtfy.py b/plugins/lmgtfy.py
index 0458be3..9703eda 100644
--- a/plugins/lmgtfy.py
+++ b/plugins/lmgtfy.py
@@ -3,7 +3,7 @@ from cloudbot import hook, web, http
 
 @hook.command(["lmgtfy", "gfy"])
 def lmgtfy(text):
-    """lmgtfy [phrase] - Posts a google link for the specified phrase"""
+    """[phrase] - gets a lmgtfy.com link for the specified phrase"""
 
     link = "http://lmgtfy.com/?q={}".format(http.quote_plus(text))
 
diff --git a/plugins/lyrics.py b/plugins/lyrics.py
index 50682a5..c2951d6 100644
--- a/plugins/lyrics.py
+++ b/plugins/lyrics.py
@@ -3,9 +3,9 @@ from cloudbot import hook, http, web
 url = "http://search.azlyrics.com/search.php?q="
 
 
-@hook.command
+@hook.command()
 def lyrics(text):
-    """lyrics <search> - Search AZLyrics.com for song lyrics"""
+    """<search> - search AZLyrics.com for song lyrics"""
     if "pastelyrics" in text:
         dopaste = True
         text = text.replace("pastelyrics", "").strip()
diff --git a/plugins/metacritic.py b/plugins/metacritic.py
index 71dbf35..c56e792 100644
--- a/plugins/metacritic.py
+++ b/plugins/metacritic.py
@@ -8,8 +8,7 @@ from cloudbot import hook, http
 
 @hook.command(["metacritic", "mc"])
 def metacritic(inp):
-    """mc [all|movie|tv|album|x360|ps3|pc|gba|ds|3ds|wii|vita|wiiu|xone|ps4] <title>
-    Gets rating for <title> from metacritic on the specified medium."""
+    """[all|movie|tv|album|x360|ps3|pc|gba|ds|3ds|wii|vita|wiiu|xone|ps4] <title> - gets rating for <title> from metacritic on the specified medium"""
 
     args = inp.strip()
 
diff --git a/plugins/minecraft_bukget.py b/plugins/minecraft_bukget.py
index 998d049..5bdef9e 100644
--- a/plugins/minecraft_bukget.py
+++ b/plugins/minecraft_bukget.py
@@ -116,7 +116,7 @@ def format_output(data):
 
 @hook.command(["bukget", "plugin"])
 def bukget(text, reply, message):
-    """bukget <slug/name> - Look up a plugin on dev.bukkit.org"""
+    """<slug/name> - gets details on a plugin from dev.bukkit.org"""
     # get the plugin slug using search
     try:
         slug = plugin_search(text)
@@ -138,7 +138,7 @@ def bukget(text, reply, message):
 
 @hook.command(autohelp=None)
 def randomplugin(reply, message):
-    """randomplugin - Gets a random plugin from dev.bukkit.org"""
+    """- gets details on a random plugin from dev.bukkit.org"""
     # get a random plugin slug
     try:
         slug = plugin_random()
diff --git a/plugins/minecraft_items.py b/plugins/minecraft_items.py
index e20b075..759a335 100644
--- a/plugins/minecraft_items.py
+++ b/plugins/minecraft_items.py
@@ -52,7 +52,7 @@ with open("./data/itemids.txt") as f:
 @asyncio.coroutine
 @hook.command(["mcitem", "mcid"])
 def mcitem(text, reply):
-    """mcitem <item/id> -- gets the id from an item or vice versa"""
+    """<item/id> - gets the id for <item> or the item name for <id>"""
     text = text.lower().strip()
 
     if text == "":
@@ -83,7 +83,7 @@ def mcitem(text, reply):
 @asyncio.coroutine
 @hook.command(["mccraft", "mcrecipe"])
 def mcrecipe(text, reply):
-    """mcrecipe <item> -- gets the crafting recipe for an item"""
+    """<item> -- gets the crafting recipe for <item>"""
     text = text.lower().strip()
 
     results = [recipe.line for recipe in recipelist
diff --git a/plugins/minecraft_ping.py b/plugins/minecraft_ping.py
index 5b91ceb..95b7918 100644
--- a/plugins/minecraft_ping.py
+++ b/plugins/minecraft_ping.py
@@ -220,7 +220,7 @@ def format_output(data):
 
 @hook.command(["mcping", "mcp"])
 def mcping(text):
-    """mcping <server>[:port] - Ping a Minecraft server to check status."""
+    """<server[:port]> - gets the MOTD of the Minecraft server at <server[:port]>"""
     try:
         host, port = parse_input(text)
     except ParseError as e:
diff --git a/plugins/minecraft_status.py b/plugins/minecraft_status.py
index 02f193e..f73452e 100644
--- a/plugins/minecraft_status.py
+++ b/plugins/minecraft_status.py
@@ -5,7 +5,7 @@ from cloudbot import hook, http
 
 @hook.command(autohelp=False)
 def mcstatus():
-    """mcstatus -- Checks the status of various Mojang (the creators of Minecraft) servers."""
+    """- gets the status of various Mojang (Minecraft) servers"""
 
     try:
         request = http.get("http://status.mojang.com/check")
diff --git a/plugins/minecraft_user.py b/plugins/minecraft_user.py
index 4b7eec0..ca98223 100644
--- a/plugins/minecraft_user.py
+++ b/plugins/minecraft_user.py
@@ -67,7 +67,7 @@ def get_profile(name):
 
 @hook.command(["mcuser", "mcpaid", "haspaid"])
 def mcuser(text):
-    """mcpaid <username> -- Gets information about the Minecraft user <account>."""
+    """<username> - gets information about the Minecraft user <account>"""
     user = text.strip()
 
     try:
diff --git a/plugins/minecraft_wiki.py b/plugins/minecraft_wiki.py
index 1d724f9..570723d 100644
--- a/plugins/minecraft_wiki.py
+++ b/plugins/minecraft_wiki.py
@@ -8,8 +8,7 @@ mc_url = "http://minecraft.gamepedia.com/"
 
 @hook.command()
 def mcwiki(text):
-    """mcwiki <phrase> -- Gets the first paragraph of
-    the Minecraft Wiki article on <phrase>."""
+    """mcwiki <phrase> - gets the first paragraph of the Minecraft Wiki article on <phrase>"""
 
     try:
         j = http.get_json(api_url, search=text)
diff --git a/plugins/mlia.py b/plugins/mlia.py
index 337eb03..065e960 100644
--- a/plugins/mlia.py
+++ b/plugins/mlia.py
@@ -23,7 +23,7 @@ refresh_cache()
 
 @hook.command(autohelp=False)
 def mlia(reply):
-    """mlia -- Gets a random quote from MyLifeIsAverage.com."""
+    """- gets a random quote from MyLifeIsAverage.com"""
     # grab the last item in the mlia cache and remove it
     mlia_id, text = mlia_cache.pop()
     # reply with the mlia we grabbed
diff --git a/plugins/namegen.py b/plugins/namegen.py
index 07375af..c9b909c 100644
--- a/plugins/namegen.py
+++ b/plugins/namegen.py
@@ -12,9 +12,9 @@ def get_generator(_json):
 
 @hook.command(autohelp=False)
 def namegen(text, bot, notice):
-    """namegen [generator] -- Generates some names using the chosen generator.
+    """[generator|list] - generates some names using the chosen generator, or lists all generators if 'list' is specified
     :type bot: core.bot.CloudBot
-    'namegen list' will display a list of all generators."""
+    """
 
     # clean up the input
     inp = text.strip().lower()
diff --git a/plugins/newegg.py b/plugins/newegg.py
index cdf90c2..7f38a15 100644
--- a/plugins/newegg.py
+++ b/plugins/newegg.py
@@ -71,9 +71,9 @@ def newegg_url(match):
     return format_item(item, show_url=False)
 
 
-@hook.command
+@hook.command()
 def newegg(text):
-    """newegg <item name> -- Searches newegg.com for <item name>"""
+    """newegg <item name> - searches newegg.com for <item name>"""
 
     # form the search request
     request = {
diff --git a/plugins/notes.py b/plugins/notes.py
index 7ff65e2..8b42920 100644
--- a/plugins/notes.py
+++ b/plugins/notes.py
@@ -91,7 +91,7 @@ def db_search(db, nick, query):
 
 @hook.command(["note", "notes"])
 def note(text, nick, db, notice):
-    """note(s) <add|del|list|search> args -- Manipulates your list of notes."""
+    """<add|del|list|search> args - manipulates your list of notes"""
 
     db_init(db)
 
diff --git a/plugins/op.py b/plugins/op.py
index 2c6a37b..e1d148a 100644
--- a/plugins/op.py
+++ b/plugins/op.py
@@ -31,71 +31,55 @@ def mode_cmd_no_target(mode, text, text_inp, chan, conn, notice):
 
 @hook.command(permissions=["op_ban", "op"])
 def ban(text, conn, chan, notice):
-    """ban [channel] <user> -- Makes the bot ban <user> in [channel].
-    If [channel] is blank the bot will ban <user> in
-    the channel the command was used in."""
+    """[channel] <user> - bans <user> in [channel], or in the caller's channel if no channel is specified"""
     mode_cmd("+b", "ban", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_ban", "op"])
 def unban(text, conn, chan, notice):
-    """unban [channel] <user> -- Makes the bot unban <user> in [channel].
-    If [channel] is blank the bot will unban <user> in
-    the channel the command was used in."""
+    """[channel] <user> - unbans <user> in [channel], or in the caller's channel if no channel is specified"""
     mode_cmd("-b", "unban", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_quiet", "op"])
 def quiet(text, conn, chan, notice):
-    """quiet [channel] <user> -- Makes the bot quiet <user> in [channel].
-    If [channel] is blank the bot will quiet <user> in
-    the channel the command was used in."""
+    """[channel] <user> - quiets <user> in [channel], or in the caller's channel if no channel is specified"""
     mode_cmd("+q", "quiet", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_quiet", "op"])
 def unquiet(text, conn, chan, notice):
-    """unquiet [channel] <user> -- Makes the bot unquiet <user> in [channel].
-    If [channel] is blank the bot will unquiet <user> in
-    the channel the command was used in."""
+    """[channel] <user> - unquiets <user> in [channel], or in the caller's channel if no channel is specified"""
     mode_cmd("-q", "unquiet", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_voice", "op"])
 def voice(text, conn, chan, notice):
-    """voice [channel] <user> -- Makes the bot voice <user> in [channel].
-    If [channel] is blank the bot will voice <user> in
-    the channel the command was used in."""
+    """[channel] <user> - voices <user> in [channel], or in the caller's channel if no channel is specified"""
     mode_cmd("+v", "voice", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_voice", "op"])
 def devoice(text, conn, chan, notice):
-    """devoice [channel] <user> -- Makes the bot devoice <user> in [channel].
-    If [channel] is blank the bot will devoice <user> in
-    the channel the command was used in."""
+    """[channel] <user> - devoices <user> in [channel], or in the caller's channel if no channel is specified"""
     mode_cmd("-v", "devoice", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_op", "op"])
 def op(text, conn, chan, notice):
-    """op [channel] <user> -- Makes the bot op <user> in [channel].
-    If [channel] is blank the bot will op <user> in
-    the channel the command was used in."""
+    """[channel] <user> - ops <user> in [channel], or in the caller's channel if no channel is specified"""
     mode_cmd("+o", "op", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_op", "op"])
 def deop(text, conn, chan, notice):
-    """deop [channel] <user> -- Makes the bot deop <user> in [channel].
-    If [channel] is blank the bot will deop <user> in
-    the channel the command was used in."""
+    """[channel] <user> - deops <user> in [channel], or in the caller's channel if no channel is specified"""
     mode_cmd("-o", "deop", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_topic", "op"])
 def topic(text, conn, chan):
-    """topic [channel] <topic> -- Change the topic of a channel."""
+    """[channel] <topic> - changes the topic to <topic> in [channel], or in the caller's channel if no channel is specified"""
     split = text.split(" ")
     if split[0].startswith("#"):
         message = " ".join(split[1:])
@@ -107,9 +91,7 @@ def topic(text, conn, chan):
 
 @hook.command(permissions=["op_kick", "op"])
 def kick(text, chan, conn, notice):
-    """kick [channel] <user> [reason] -- Makes the bot kick <user> in [channel]
-    If [channel] is blank the bot will kick the <user> in
-    the channel the command was used in."""
+    """[channel] <user> - kicks <user> from [channel], or from the caller's channel if no channel is specified"""
     split = text.split(" ")
 
     if split[0].startswith("#"):
@@ -135,7 +117,7 @@ def kick(text, chan, conn, notice):
 
 @hook.command(permissions=["op_rem", "op"])
 def remove(text, chan, conn):
-    """remove [channel] [user] -- Force a user to part from a channel."""
+    """[channel] <user> - force removes <user> from [channel], or in the caller's channel if no channel is specified"""
     split = text.split(" ")
     if split[0].startswith("#"):
         message = " ".join(split[1:])
@@ -149,31 +131,23 @@ def remove(text, chan, conn):
 
 @hook.command(permissions=["op_mute", "op"], autohelp=False)
 def mute(text, conn, chan, notice):
-    """mute [channel] -- Makes the bot mute a channel..
-    If [channel] is blank the bot will mute
-    the channel the command was used in."""
+    """[channel] - mutes [channel], or in the caller's channel if no channel is specified"""
     mode_cmd_no_target("+m", "mute", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_mute", "op"], autohelp=False)
 def unmute(text, conn, chan, notice):
-    """mute [channel] -- Makes the bot mute a channel..
-    If [channel] is blank the bot will mute
-    the channel the command was used in."""
+    """[channel] - unmutes [channel], or in the caller's channel if no channel is specified"""
     mode_cmd_no_target("-m", "unmute", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_lock", "op"], autohelp=False)
 def lock(text, conn, chan, notice):
-    """lock [channel] -- Makes the bot lock a channel.
-    If [channel] is blank the bot will mute
-    the channel the command was used in."""
+    """[channel] - locks [channel], or in the caller's channel if no channel is specified"""
     mode_cmd_no_target("+i", "lock", text, chan, conn, notice)
 
 
 @hook.command(permissions=["op_lock", "op"], autohelp=False)
 def unlock(text, conn, chan, notice):
-    """unlock [channel] -- Makes the bot unlock a channel..
-    If [channel] is blank the bot will mute
-    the channel the command was used in."""
+    """[channel] - unlocks [channel], or in the caller's channel if no channel is specified"""
     mode_cmd_no_target("-i", "unlock", text, chan, conn, notice)
diff --git a/plugins/osrc.py b/plugins/osrc.py
index 977e052..01a3a71 100644
--- a/plugins/osrc.py
+++ b/plugins/osrc.py
@@ -4,9 +4,9 @@ from cloudbot import hook, http, web
 user_url = "http://osrc.dfm.io/{}"
 
 
-@hook.command
+@hook.command()
 def osrc(text):
-    """osrc <github user> -- Gets an Open Source Report Card for <github user>"""
+    """<github user> - gets an Open Source Report Card for <github user> from osrc.dfm.io"""
 
     user_nick = text.strip()
     url = user_url.format(user_nick)
diff --git a/plugins/pagecheck.py b/plugins/pagecheck.py
index 33fc807..2c2fb26 100644
--- a/plugins/pagecheck.py
+++ b/plugins/pagecheck.py
@@ -3,9 +3,9 @@ import urllib.parse
 from cloudbot import hook, http, urlnorm
 
 
-@hook.command
+@hook.command(["down", "offline", "up"])
 def down(text):
-    """down <url> -- Checks if the site at <url> is up or down.
+    """<url> - checks if <url> is online or offline
     :type text: str
     """
 
@@ -21,9 +21,9 @@ def down(text):
         return '{} seems to be down'.format(text)
 
 
-@hook.command
+@hook.command()
 def isup(text):
-    """isup -- uses isup.me to see if a site is up or not
+    """<url> - uses isup.me to check if <url> is online or offline
     :type text: str
     """
 
diff --git a/plugins/password.py b/plugins/password.py
index 55d4456..e3b90ed 100644
--- a/plugins/password.py
+++ b/plugins/password.py
@@ -14,8 +14,7 @@ with open("data/password_words.txt") as f:
 
 @hook.command(autohelp=False)
 def password(text, notice):
-    """password <length> [types] -- Generates a password of <length> (default 10).
-    [types] can include 'alpha', 'no caps', 'numeric', 'symbols' or any combination of the inp, eg. 'numbers symbols'"""
+    """[length [types]] - generates a password of <length> (default 10). [types] can include 'alpha', 'no caps', 'numeric', 'symbols' or any combination: eg. 'numbers symbols'"""
     okay = []
 
     # find the length needed for the password
@@ -58,7 +57,7 @@ def password(text, notice):
 
 @hook.command(["rpass", "rpassword", "readablepassword"], autohelp=False)
 def readable_password(text, notice):
-    """rpass [length]  -- Generates an easy to remember password with [length] (default 4) commonly used words."""
+    """[length] - generates an easy to remember password with [length] (default 4) commonly used words"""
     if text:
         try:
             length = int(text)
diff --git a/plugins/ping.py b/plugins/ping.py
index 1ac9431..a21ae5c 100644
--- a/plugins/ping.py
+++ b/plugins/ping.py
@@ -8,9 +8,9 @@ from cloudbot import hook
 ping_regex = re.compile(r"(\d+.\d+)/(\d+.\d+)/(\d+.\d+)/(\d+.\d+)")
 
 
-@hook.command
+@hook.command()
 def ping(text, reply):
-    """ping <host> [count] -- Pings <host> [count] times."""
+    """<host> [count] - pings <host> [count] times"""
 
     if os.name == "nt":
         return "Sorry, this command is not supported on Windows systems."
diff --git a/plugins/plpaste.py b/plugins/plpaste.py
index 874459f..c13600d 100644
--- a/plugins/plpaste.py
+++ b/plugins/plpaste.py
@@ -5,7 +5,7 @@ from cloudbot import hook, web
 
 @hook.command(permissions=["adminonly"])
 def plpaste(text, bot):
-    """
+    """<command> - pastes the plugin file that contains <command>
     :type text: str
     :type bot: core.bot.CloudBot
     """
diff --git a/plugins/python.py b/plugins/python.py
index d7ab3a0..624bae3 100644
--- a/plugins/python.py
+++ b/plugins/python.py
@@ -2,8 +2,8 @@ from cloudbot import hook
 from cloudbot.util.pyexec import eval_py
 
 
-@hook.command
+@hook.command(["python", "py"])
 def python(text):
-    """python <prog> -- Executes <prog> as Python code."""
+    """<python code> - executes <python code> using eval.appspot.com"""
 
     return eval_py(text)
diff --git a/plugins/qrcode.py b/plugins/qrcode.py
index 8831e5d..4f47cb1 100644
--- a/plugins/qrcode.py
+++ b/plugins/qrcode.py
@@ -3,9 +3,9 @@ from cloudbot import hook, web, http
 
 
 @hook.command('qr')
-@hook.command
+@hook.command()
 def qrcode(text):
-    """qrcode [link] returns a link for a QR code."""
+    """[link] - returns a link to a QR code image for [link]"""
 
     args = {
         "cht": "qr",  # chart type (QR)
diff --git a/plugins/quote.py b/plugins/quote.py
index add0dcd..3fab3e5 100644
--- a/plugins/quote.py
+++ b/plugins/quote.py
@@ -121,10 +121,9 @@ def get_quote_by_chan(db, chan, num=False):
 
 
 @hook.command('q')
-@hook.command
+@hook.command()
 def quote(inp, nick='', chan='', db=None, notice=None):
-    """quote [#chan] [nick] [#n]/.quote add <nick> <msg>
-    Gets random or [#n]th quote by <nick> or from <#chan>/adds quote."""
+    """[#chan] [nick] [#n] OR add <nick> <message> - gets the [#n]th quote by <nick> (defaulting to random) OR adds <message> as a quote for <nick> in the caller's channel"""
     create_table_if_not_exists(db)
 
     add = re.match(r"add[^\w@]+(\S+?)>?\s+(.*)", inp, re.I)
diff --git a/plugins/rdio.py b/plugins/rdio.py
index 5295041..c09f479 100644
--- a/plugins/rdio.py
+++ b/plugins/rdio.py
@@ -1,4 +1,5 @@
 import urllib
+import urllib.parse
 import json
 import re
 
@@ -16,14 +17,14 @@ def getdata(inp, types, api_key, api_secret):
     return data
 
 
-@hook.command
-def rdio(inp, bot=None):
-    """ rdio <search term> - alternatives: .rdiot (track), .rdioar (artist), .rdioal (album) """
+@hook.command()
+def rdio(text, bot):
+    """<query> - searches rdio for <query>"""
     api_key = bot.config.get("api_keys", {}).get("rdio_key")
     api_secret = bot.config.get("api_keys", {}).get("rdio_secret")
     if not api_key:
         return "error: no api key set"
-    data = getdata(inp, "Track,Album,Artist", api_key, api_secret)
+    data = getdata(text, "Track,Album,Artist", api_key, api_secret)
     try:
         info = data['result']['results'][0]
     except IndexError:
@@ -46,14 +47,14 @@ def rdio(inp, bot=None):
             return "\x02{}\x02 - {}".format(name, url)
 
 
-@hook.command
-def rdiot(inp, bot=None):
-    """ rdiot <search term> - Search for tracks on rdio """
+@hook.command()
+def rdiot(text, bot):
+    """<query> - searches rdio for tracks matching <query>"""
     api_key = bot.config.get("api_keys", {}).get("rdio_key")
     api_secret = bot.config.get("api_keys", {}).get("rdio_secret")
     if not api_key:
         return "error: no api key set"
-    data = getdata(inp, "Track", api_key, api_secret)
+    data = getdata(text, "Track", api_key, api_secret)
     try:
         info = data['result']['results'][0]
     except IndexError:
@@ -65,14 +66,14 @@ def rdiot(inp, bot=None):
     return "\x02{}\x02 by \x02{}\x02 - {} - {}".format(name, artist, album, url)
 
 
-@hook.command
-def rdioar(inp, bot=None):
-    """ rdioar <search term> - Search for artists on rdio """
+@hook.command()
+def rdioar(text, bot):
+    """<query> - searches rdio for artists matching <query>"""
     api_key = bot.config.get("api_keys", {}).get("rdio_key")
     api_secret = bot.config.get("api_keys", {}).get("rdio_secret")
     if not api_key:
         return "error: no api key set"
-    data = getdata(inp, "Artist", api_key, api_secret)
+    data = getdata(text, "Artist", api_key, api_secret)
     try:
         info = data['result']['results'][0]
     except IndexError:
@@ -82,14 +83,14 @@ def rdioar(inp, bot=None):
     return "\x02{}\x02 - {}".format(name, url)
 
 
-@hook.command
-def rdioal(inp, bot=None):
-    """ rdioal <search term> - Search for albums on rdio """
+@hook.command()
+def rdioal(text, bot):
+    """<query> - searches rdio for albums matching <query>"""
     api_key = bot.config.get("api_keys", {}).get("rdio_key")
     api_secret = bot.config.get("api_keys", {}).get("rdio_secret")
     if not api_key:
         return "error: no api key set"
-    data = getdata(inp, "Album", api_key, api_secret)
+    data = getdata(text, "Album", api_key, api_secret)
     try:
         info = data['result']['results'][0]
     except IndexError:
@@ -104,7 +105,7 @@ rdio_re = (r'(.*:)//(rd.io|www.rdio.com|rdio.com)(:[0-9]+)?(.*)', re.I)
 
 
 @hook.regex(*rdio_re)
-def rdio_url(match, bot=None):
+def rdio_url(match, bot):
     api_key = bot.config.get("api_keys", {}).get("rdio_key")
     api_secret = bot.config.get("api_keys", {}).get("rdio_secret")
     if not api_key:
diff --git a/plugins/recipe.py b/plugins/recipe.py
index 2ba55b4..9effff4 100644
--- a/plugins/recipe.py
+++ b/plugins/recipe.py
@@ -43,7 +43,7 @@ def get_data(url):
 
 @hook.command(autohelp=False)
 def recipe(text):
-    """recipe [term] - Gets a recipe for [term], or ets a random recipe if [term] is not provided"""
+    """[term] - gets a recipe for [term], or gets a random recipe if no term is specified"""
     if text:
         # get the recipe URL by searching
         try:
@@ -85,7 +85,7 @@ def recipe(text):
 
 @hook.command(autohelp=False)
 def dinner():
-    """dinner - WTF IS FOR DINNER"""
+    """- gets a random dinner from whatthefuckshouldimakefordinner.com"""
     try:
         page = http.open(random_url)
     except (http.HTTPError, http.URLError) as e:
diff --git a/plugins/reddit.py b/plugins/reddit.py
index 74e8837..5a3ca2d 100644
--- a/plugins/reddit.py
+++ b/plugins/reddit.py
@@ -26,13 +26,13 @@ def reddit_url(match):
 
 
 @hook.command(autohelp=False)
-def reddit(inp):
-    """reddit <subreddit> [n] -- Gets a random post from <subreddit>, or gets the [n]th post in the subreddit."""
+def reddit(text):
+    """<subreddit> [n] - gets a random post from <subreddit>, or gets the [n]th post in the subreddit"""
     id_num = None
 
-    if inp:
+    if text:
         # clean and split the input
-        parts = inp.lower().strip().split()
+        parts = text.lower().strip().split()
 
         # find the requested post number (if any)
         if len(parts) > 1:
diff --git a/plugins/rss.py b/plugins/rss.py
index 997e01f..0b851d3 100644
--- a/plugins/rss.py
+++ b/plugins/rss.py
@@ -2,7 +2,7 @@ from cloudbot import hook, formatting, web
 
 
 @hook.command("feed")
-@hook.command
+@hook.command()
 def rss(text, message):
     """rss <feed> -- Gets the first three items from the RSS feed <feed>."""
     limit = 3
diff --git a/plugins/scene.py b/plugins/scene.py
index fb34b46..97bad91 100644
--- a/plugins/scene.py
+++ b/plugins/scene.py
@@ -4,7 +4,7 @@ from cloudbot import hook, http, timesince
 
 
 @hook.command("scene")
-@hook.command
+@hook.command()
 def pre(inp):
     """pre <query> -- searches scene releases using orlydb.com"""
 
diff --git a/plugins/shorten.py b/plugins/shorten.py
index a6aca70..1ef455e 100644
--- a/plugins/shorten.py
+++ b/plugins/shorten.py
@@ -1,9 +1,9 @@
 from cloudbot import hook, web
 
 
-@hook.command
+@hook.command()
 def shorten(text):
-    """shorten <url> [custom] - Shorten a url. [custom] is an optional custom shortlink."""
+    """<url> [custom] - shortens a url with [custom] as an optional custom shortlink"""
     args = text.split()
     url = args[0]
     custom = args[1] if len(args) > 1 else None
@@ -14,9 +14,9 @@ def shorten(text):
         return e.message
 
 
-@hook.command
+@hook.command()
 def gitio(text):
-    """gitio <url> [custom] -- Shorten a Github url with git.io. [custom] is an optional custom shortlink."""
+    """<url> [custom] - shortens a github URL <url> using git.io with [custom] as an optional custom shortlink"""
     args = text.split()
     url = args[0]
     custom = args[1] if len(args) > 1 else None
@@ -27,9 +27,9 @@ def gitio(text):
         return e.message
 
 
-@hook.command
+@hook.command()
 def isgd(text):
-    """shorten <url> [custom] - Shorten a url with is.gd. [custom] is an optional custom shortlink."""
+    """<url> [custom] - shortens a url using is.gd with [custom] as an optional custom shortlink"""
     args = text.split()
     url = args[0]
     custom = args[1] if len(args) > 1 else None
diff --git a/plugins/slap.py b/plugins/slap.py
index a777b78..64042b7 100644
--- a/plugins/slap.py
+++ b/plugins/slap.py
@@ -9,7 +9,7 @@ def get_generator(_json, variables):
                                  data["parts"], variables=variables)
 
 
-@hook.command
+@hook.command()
 def slap(text, action, nick, conn, notice):
     """slap <user> -- Makes the bot slap <user>."""
     target = text.strip()
diff --git a/plugins/slogan.py b/plugins/slogan.py
index 6b88e97..73b3b38 100644
--- a/plugins/slogan.py
+++ b/plugins/slogan.py
@@ -7,7 +7,7 @@ with open("./data/slogans.txt") as f:
                if not line.startswith("//")]
 
 
-@hook.command
+@hook.command()
 def slogan(text):
     """slogan <word> -- Makes a slogan for <word>."""
     out = random.choice(slogans)
diff --git a/plugins/snopes.py b/plugins/snopes.py
index 746cc41..6affa17 100644
--- a/plugins/snopes.py
+++ b/plugins/snopes.py
@@ -5,7 +5,7 @@ from cloudbot import hook, http
 search_url = "http://search.atomz.com/search/?sp_a=00062d45-sp00000000"
 
 
-@hook.command
+@hook.command()
 def snopes(text):
     """snopes <topic> -- Searches snopes for an urban legend about <topic>."""
 
diff --git a/plugins/spellcheck.py b/plugins/spellcheck.py
index 9cf3983..70f128e 100644
--- a/plugins/spellcheck.py
+++ b/plugins/spellcheck.py
@@ -6,7 +6,7 @@ from cloudbot import hook
 locale = "en_US"
 
 
-@hook.command
+@hook.command()
 def spell(text):
     """spell <word/sentence> -- Check spelling of a word or sentence."""
 
diff --git a/plugins/spotify.py b/plugins/spotify.py
index cfc5f0c..c51f6e7 100644
--- a/plugins/spotify.py
+++ b/plugins/spotify.py
@@ -32,7 +32,7 @@ def sptfy(inp, sptfy=False):
 
 
 @hook.command('sptrack')
-@hook.command
+@hook.command()
 def spotify(inp):
     """spotify <song> -- Search Spotify for <song>"""
     try:
@@ -50,7 +50,7 @@ def spotify(inp):
                                                   data["tracks"][0]["artists"][0]["name"], url)
 
 
-@hook.command
+@hook.command()
 def spalbum(inp):
     """spalbum <album> -- Search Spotify for <album>"""
     try:
@@ -68,7 +68,7 @@ def spalbum(inp):
                                                   data["albums"][0]["artists"][0]["name"], url)
 
 
-@hook.command
+@hook.command()
 def spartist(inp):
     """spartist <artist> -- Search Spotify for <artist>"""
     try:
diff --git a/plugins/steam.py b/plugins/steam.py
index 7594134..1fc9498 100644
--- a/plugins/steam.py
+++ b/plugins/steam.py
@@ -63,7 +63,7 @@ def steam_url(match):
     return get_steam_info("http://store.steampowered.com" + match.group(4))
 
 
-@hook.command
+@hook.command()
 def steam(inp):
     """steam [search] - Search for specified game/trailer/DLC"""
     page = http.get("http://store.steampowered.com/search/?term=" + inp)
diff --git a/plugins/stock.py b/plugins/stock.py
index 5bf6a97..efcf4ff 100644
--- a/plugins/stock.py
+++ b/plugins/stock.py
@@ -1,7 +1,7 @@
 from cloudbot import hook, web
 
 
-@hook.command
+@hook.command()
 def stock(inp):
     """stock <symbol> -- gets stock information"""
     sym = inp.strip().lower()
diff --git a/plugins/suggest.py b/plugins/suggest.py
index b1a234a..de8c213 100644
--- a/plugins/suggest.py
+++ b/plugins/suggest.py
@@ -5,7 +5,7 @@ from bs4 import BeautifulSoup
 from cloudbot import hook, http, formatting
 
 
-@hook.command
+@hook.command()
 def suggest(text):
     """suggest <phrase> -- Gets suggested phrases for a google search"""
 
diff --git a/plugins/title.py b/plugins/title.py
index 58fac57..ccd5fb2 100644
--- a/plugins/title.py
+++ b/plugins/title.py
@@ -3,7 +3,7 @@ from bs4 import BeautifulSoup
 from cloudbot import hook, http, urlnorm
 
 
-@hook.command
+@hook.command()
 def title(text):
     """title <url> -- gets the title of a web page"""
     url = urlnorm.normalize(text, assume_scheme="http")
diff --git a/plugins/tvdb.py b/plugins/tvdb.py
index 18e7003..c407df6 100644
--- a/plugins/tvdb.py
+++ b/plugins/tvdb.py
@@ -62,7 +62,7 @@ def get_episode_info(episode, api_key):
     return first_aired, air_date, episode_desc
 
 
-@hook.command
+@hook.command()
 @hook.command('tv')
 def tv_next(inp, bot=None):
     """tv <series> -- Get the next episode of <series>."""
@@ -112,7 +112,7 @@ def tv_next(inp, bot=None):
         return "The next episodes of {}: {}".format(series_name, next_eps)
 
 
-@hook.command
+@hook.command()
 @hook.command('tv_prev')
 def tv_last(inp, bot=None):
     """tv_last <series> -- Gets the most recently aired episode of <series>."""
diff --git a/plugins/twitch.py b/plugins/twitch.py
index 3f62670..383516a 100644
--- a/plugins/twitch.py
+++ b/plugins/twitch.py
@@ -55,7 +55,7 @@ def twitch_url(match):
 
 
 @hook.command('twitchviewers')
-@hook.command
+@hook.command()
 def twviewers(inp):
     inp = inp.split("/")[-1]
     if test(inp):
diff --git a/plugins/twitter.py b/plugins/twitter.py
index e9b5b4d..e17d8c3 100644
--- a/plugins/twitter.py
+++ b/plugins/twitter.py
@@ -54,7 +54,7 @@ def twitter_url(match, bot=None):
 
 @hook.command("tw")
 @hook.command("twatter")
-@hook.command
+@hook.command()
 def twitter(inp, bot=None):
     """twitter <user> [n] -- Gets last/[n]th tweet from <user>"""
 
@@ -139,7 +139,7 @@ def twitter(inp, bot=None):
 
 
 @hook.command("twinfo")
-@hook.command
+@hook.command()
 def twuser(inp, bot=None):
     """twuser <user> -- Get info on the Twitter user <user>"""
 
diff --git a/plugins/update.py b/plugins/update.py
index d022172..89032d0 100644
--- a/plugins/update.py
+++ b/plugins/update.py
@@ -3,7 +3,7 @@ from git import Repo
 from cloudbot import hook, web
 
 
-@hook.command
+@hook.command()
 def update():
     repo = Repo()
     git = repo.git
@@ -17,7 +17,7 @@ def update():
         return pull
 
 
-@hook.command
+@hook.command()
 def version():
     repo = Repo()
 
diff --git a/plugins/utility.py b/plugins/utility.py
index 6183346..0d65d65 100644
--- a/plugins/utility.py
+++ b/plugins/utility.py
@@ -35,7 +35,7 @@ def strip(string):
 
 ## TODO: make this capitalize sentences correctly
 @hook.command("capitalise")
-@hook.command
+@hook.command()
 def capitalize(text):
     """capitalize <string> -- Capitalizes <string>.
     :type text: str
@@ -43,25 +43,25 @@ def capitalize(text):
     return ". ".join([sentence.capitalize() for sentence in text.split(". ")])
 
 
-@hook.command
+@hook.command()
 def upper(text):
     """upper <string> -- Convert string to uppercase."""
     return text.upper()
 
 
-@hook.command
+@hook.command()
 def lower(text):
     """lower <string> -- Convert string to lowercase."""
     return text.lower()
 
 
-@hook.command
+@hook.command()
 def titlecase(text):
     """title <string> -- Convert string to title case."""
     return text.title()
 
 
-@hook.command
+@hook.command()
 def swapcase(text):
     """swapcase <string> -- Swaps the capitalization of <string>."""
     return text.swapcase()
@@ -103,14 +103,14 @@ def base64_check(text):
         return "'{}' is a valid base64 encoded string".format(text)
 
 
-@hook.command
+@hook.command()
 def unescape(text):
     """unescape <string> -- Unicode unescapes <string>."""
     decoder = codecs.getdecoder("unicode_escape")
     return decoder(text)[0]
 
 
-@hook.command
+@hook.command()
 def escape(text):
     """escape <string> -- Unicode escapes <string>."""
     encoder = codecs.getencoder("unicode_escape")
@@ -119,7 +119,7 @@ def escape(text):
 # length
 
 
-@hook.command
+@hook.command()
 def length(text):
     """length <string> -- Gets the length of <string>"""
     return "The length of that string is {} characters.".format(len(text))
@@ -128,7 +128,7 @@ def length(text):
 # reverse
 
 
-@hook.command
+@hook.command()
 def reverse(text):
     """reverse <string> -- Reverses <string>."""
     return text[::-1]
@@ -147,7 +147,7 @@ def hash_command(text):
 # novelty
 
 
-@hook.command
+@hook.command()
 def munge(text):
     """munge <text> -- Munges up <text>."""
     return formatting.munge(text)
@@ -156,7 +156,7 @@ def munge(text):
 # colors - based on code by Reece Selwood - <https://github.com/hitzler/homero>
 
 
-@hook.command
+@hook.command()
 def rainbow(text):
     text = str(text)
     text = strip(text)
@@ -171,7 +171,7 @@ def rainbow(text):
     return out
 
 
-@hook.command
+@hook.command()
 def wrainbow(text):
     text = str(text)
     col = list(colors.items())
@@ -183,7 +183,7 @@ def wrainbow(text):
     return ' '.join(out)
 
 
-@hook.command
+@hook.command()
 def usa(text):
     text = strip(text)
     c = [colors['red'], '\x0300', colors['blue']]
diff --git a/plugins/valvesounds.py b/plugins/valvesounds.py
index 55a27de..14fe031 100644
--- a/plugins/valvesounds.py
+++ b/plugins/valvesounds.py
@@ -32,13 +32,13 @@ def get_sound_info(game, search):
         return "{} (and {} others: {})".format(items[0], len(items) - 1, web.paste("\n".join(items)))
 
 
-@hook.command
+@hook.command()
 def portal2(text):
     """portal2 <quote> - Look up Portal 2 quote. Example: .portal2 demand to see life's manager"""
     return get_sound_info("portal2", text)
 
 
-@hook.command
+@hook.command()
 def portal2dlc(text):
     """portal2dlc <quote> - Look up Portal 2 DLC quote. Example: .portal2dlc1 these exhibits are interactive"""
     return get_sound_info("portal2dlc1", text)
@@ -50,7 +50,7 @@ def portal2dlc2(text):
     return get_sound_info("portal2dlc2", text)
 
 
-@hook.command
+@hook.command()
 def portal2music(text):
     """portal2music <title> - Look up Portal 2 music. Example: .portal2music turret opera"""
     return get_sound_info("portal2music", text)
@@ -74,7 +74,7 @@ def tf2(text):
     return get_sound_info("tf2", text)
 
 
-@hook.command
+@hook.command()
 def tf2music(text):
     """tf2music title - Look up TF2 music lyrics. Example: .tf2music rocket jump waltz"""
     return get_sound_info("tf2music", text)
diff --git a/plugins/xkcd.py b/plugins/xkcd.py
index efaacba..e02d9a7 100644
--- a/plugins/xkcd.py
+++ b/plugins/xkcd.py
@@ -36,7 +36,7 @@ def xkcd_url(match):
     return xkcd_info(xkcd_id)
 
 
-@hook.command
+@hook.command()
 def xkcd(text):
     """xkcd <search term> - Search for xkcd comic matching <search term>"""
     return xkcd_search(text)
diff --git a/plugins/yahooanswers.py b/plugins/yahooanswers.py
index 423f919..315baf5 100644
--- a/plugins/yahooanswers.py
+++ b/plugins/yahooanswers.py
@@ -1,7 +1,7 @@
 from cloudbot import hook, web, formatting
 
 
-@hook.command
+@hook.command()
 def answer(text):
     """answer <query> -- find the answer to a question on Yahoo! Answers"""
 
-- 
2.0.0

