From 80625781fb1d052bcb69e88d24eee59f9994d4b2 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:13:03 -0800
Subject: [PATCH] Change ctcp reply to return 'Eliralin'


diff --git a/plugins/core_ctcp.py b/plugins/core_ctcp.py
index 1393228..8453d57 100644
--- a/plugins/core_ctcp.py
+++ b/plugins/core_ctcp.py
@@ -6,7 +6,7 @@ from cloudbot import hook
 # CTCP responses
 @hook.regex(r'^\x01VERSION\x01$')
 def ctcp_version(notice):
-    notice('\x01VERSION: CloudBot - http://git.io/cloudbotirc')
+    notice('\x01VERSION Eliralin #refresh - http://git.io/elirefresh')
 
 
 @hook.regex(r'^\x01PING\x01$')
-- 
1.9.3

