From c3e0fb757e3d6283ac12e6f4450ad72784dcab10 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:13:03 -0800
Subject: [PATCH] Change ctcp reply to return 'Eliralin'


diff --git a/modules/core_ctcp.py b/modules/core_ctcp.py
index 2c18f9b..d4a7b7c 100644
--- a/modules/core_ctcp.py
+++ b/modules/core_ctcp.py
@@ -6,7 +6,7 @@ from util import hook
 # CTCP responses
 @hook.regex(r'^\x01VERSION\x01$')
 def ctcp_version(notice):
-    notice('\x01VERSION: CloudBot - http://git.io/cloudbotirc')
+    notice('\x01VERSION Eliralin #refresh - http://git.io/elirefresh')
 
 
 @hook.regex(r'^\x01PING\x01$')
-- 
1.9.2

