From a1de381cdd940764795f4f58022702d06bbafd7d Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 26 Apr 2014 19:43:21 -0700
Subject: [PATCH] Replace symbol for factoids.

Changes ?fact to ^fact

diff --git a/plugins/factoids.py b/plugins/factoids.py
index c2f8e6e..1906644 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -11,7 +11,7 @@ from cloudbot import botvars, hook, formatting, web
 
 re_lineends = re.compile(r'[\r\n]*')
 
-FACTOID_CHAR = "?"  # TODO: config
+FACTOID_CHAR = "^"  # TODO: config
 
 # some simple "shortcodes" for formatting purposes
 shortcodes = {
-- 
2.0.0

