From 8943b9165b111f9e62cc52d6162cea9c00e6f83d Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 26 Apr 2014 18:39:33 -0700
Subject: [PATCH] Change reply format from ({}) to {},


diff --git a/cloudbot/core/events.py b/cloudbot/core/events.py
index 24beac0..8e3c693 100644
--- a/cloudbot/core/events.py
+++ b/cloudbot/core/events.py
@@ -126,7 +126,7 @@ class BaseEvent:
         if target == self.nick:
             self.conn.msg(target, message)
         else:
-            self.conn.msg(target, "({}) {}".format(self.nick, message))
+            self.conn.msg(target, "{}, {}".format(self.nick, message))
 
     def action(self, message, target=None):
         """sends an action to the current channel/user or a specific channel/user
-- 
2.0.0

