From 1037a2758d71d19301432cc95bb8889a4bdaf23a Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:13:03 -0800
Subject: [PATCH] Change ctcp reply to return 'Eliralin'


diff --git a/plugins/core_ctcp.py b/plugins/core_ctcp.py
index 229ae7f..6ef1828 100644
--- a/plugins/core_ctcp.py
+++ b/plugins/core_ctcp.py
@@ -8,7 +8,7 @@ from cloudbot import hook
 @asyncio.coroutine
 @hook.regex(r'^\x01VERSION\x01$')
 def ctcp_version(notice):
-    notice('\x01VERSION: CloudBot - http://git.io/cloudbotirc')
+    notice('\x01VERSION Eliralin #refresh - http://git.io/elirefresh')
 
 
 @asyncio.coroutine
-- 
2.0.0

