From 5f7a46dc3701db8ba396e6bba0f2d7051fab3d59 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 26 Apr 2014 19:43:21 -0700
Subject: [PATCH] Replace symbol for factoids.

Changes ?fact to ^fact

diff --git a/modules/factoids.py b/modules/factoids.py
index bba8f92..d82b9d4 100644
--- a/modules/factoids.py
+++ b/modules/factoids.py
@@ -130,7 +130,7 @@ def info(text, notice):
         notice("Unknown Factoid.")
 
 
-@hook.regex(r'^\? ?(.+)')
+@hook.regex(r'^\^ ?(.+)')
 def factoid(inp, input, db, message, action):
     """?<word> -- Shows what data is associated with <word>."""
 
-- 
1.9.2

