From 35c744ecb0bf43667f80003c2d87c5399ac29b9e Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 26 Apr 2014 19:49:34 -0700
Subject: [PATCH] Add more dinner phrases


diff --git a/modules/recipe.py b/modules/recipe.py
index d4ef996..5f7b742 100644
--- a/modules/recipe.py
+++ b/modules/recipe.py
@@ -17,6 +17,11 @@ phrases = [
     "WHY DON'T YOU EAT SOME FUCKING \x02{}?\x02",
     "MAKE SOME FUCKING \x02{}\x02",
     "INDUCE FOOD COMA WITH SOME FUCKING \x02{}\x02"
+    "CLASSILY PARTAKE IN SOME FUCKING \x02{}\x02",
+    "COOK UP SOME FUCKING \x02{}\x02",
+    "CURE YOUR MOUTH'S POST TRAUMATIC STRESS DISORDER WITH SOME FUCKING \x02{}\x02",
+    "PROCURE SOME CHILD LABOR TO COOK UP SOME FUCKING \x02{}\x02",
+    "YOUR INDECISION IS FAR LESS APPETIZING THAN SOME FUCKING \x02{}\x02"
 ]
 
 clean_key = lambda i: i.split("#")[1]
-- 
1.9.2

