From cf4ae7bc4d9815b7a71462320a9162dd5df457ef Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 26 Apr 2014 19:49:34 -0700
Subject: [PATCH] Add more dinner phrases


diff --git a/plugins/recipe.py b/plugins/recipe.py
index 2ba55b4..07008f5 100644
--- a/plugins/recipe.py
+++ b/plugins/recipe.py
@@ -17,6 +17,11 @@ phrases = [
     "WHY DON'T YOU EAT SOME FUCKING \x02{}?\x02",
     "MAKE SOME FUCKING \x02{}\x02",
     "INDUCE FOOD COMA WITH SOME FUCKING \x02{}\x02"
+    "CLASSILY PARTAKE IN SOME FUCKING \x02{}\x02",
+    "COOK UP SOME FUCKING \x02{}\x02",
+    "CURE YOUR MOUTH'S POST TRAUMATIC STRESS DISORDER WITH SOME FUCKING \x02{}\x02",
+    "PROCURE SOME CHILD LABOR TO COOK UP SOME FUCKING \x02{}\x02",
+    "YOUR INDECISION IS FAR LESS APPETIZING THAN SOME FUCKING \x02{}\x02"
 ]
 
 clean_key = lambda i: i.split("#")[1]
-- 
2.0.0

