From 189419c1317d1f1295f77bc84644a8dc144acb7d Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 26 Apr 2014 19:43:21 -0700
Subject: [PATCH] Replace symbol for factoids.

Changes ?fact to ^fact

diff --git a/plugins/factoids.py b/plugins/factoids.py
index 2afe29b..8a5806d 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -135,7 +135,7 @@ def info(text, notice):
 
 
 @asyncio.coroutine
-@hook.regex(r'^\? ?(.+)')
+@hook.regex(r'^\^ ?(.+)')
 def factoid(inp, input, db, message, action):
     """<word> - shows what data is associated with <word>"""
 
-- 
2.0.0

