From 914d9b7c530ea57cdd5d18925caace49b89c4938 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 29 Apr 2014 12:31:44 -0700
Subject: [PATCH] Change regex_chans default from True to False


diff --git a/plugins/regex_chans.py b/plugins/regex_chans.py
index 7be6be0..61a7903 100644
--- a/plugins/regex_chans.py
+++ b/plugins/regex_chans.py
@@ -14,7 +14,7 @@ table = Table(
 # Default value.
 # If True, all channels without a setting will have regex enabled
 # If False, all channels without a setting will have regex disabled
-default_enabled = True
+default_enabled = False
 
 
 @hook.onload()
-- 
1.9.3

