From 806704a85cd5ee10b928a44911d3205132b3bf91 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 11 Jun 2014 02:15:07 -0700
Subject: [PATCH] Apply migrateit - todo: merge


diff --git a/plugins/eliralin_teamcity.py b/plugins/eliralin_teamcity.py
index c206042..f62c8f6 100644
--- a/plugins/eliralin_teamcity.py
+++ b/plugins/eliralin_teamcity.py
@@ -111,7 +111,7 @@ def find_download_url(build_type_element):
 database = ProjectDatabase()
 
 
-@hook.command(["teamcity", "ci"])
+@hook.command("teamcity", "ci")
 def teamcity(text, bot, reply, message):
     """teamcity [project] - Searches for project on teamcity, and displays project URL and download"""
     database.ensure_loaded(bot=bot)
@@ -141,7 +141,7 @@ def teamcity(text, bot, reply, message):
     database.after_check()
 
 
-@hook.command(["reloadci", "reload_teamcity"], permissions=["botcontrol"], autohelp=False)
+@hook.command("reloadci", "reload_teamcity", permissions=["botcontrol"], autohelp=False)
 def reload_teamcity(bot, reply):
     """reloadci - Reloads teamcity database manually"""
     database.load_key(bot)
@@ -150,7 +150,7 @@ def reload_teamcity(bot, reply):
     database.after_check()
 
 
-@hook.command(["listci", "list_teamcity"], permissions=["botcontrol"], autohelp=False)
+@hook.command("listci", "list_teamcity", permissions=["botcontrol"], autohelp=False)
 def list_teamcity(bot, reply):
     """listci - Lists all projects loaded into database from teamcity"""
     database.ensure_loaded(bot=bot)
diff --git a/plugins/eliralin_utility.py b/plugins/eliralin_utility.py
index 908fc8b..77f54b5 100644
--- a/plugins/eliralin_utility.py
+++ b/plugins/eliralin_utility.py
@@ -5,7 +5,7 @@ import socket
 from cloudbot import hook
 
 @asyncio.coroutine
-@hook.command(["josephus", "jose"])
+@hook.command("josephus", "jose")
 def josephus(text):
     """[size] [every x] [starting person] - calculates who dies last """
     split = text.split()
@@ -35,7 +35,7 @@ def pet(action, nick):
 
 
 @asyncio.coroutine
-@hook.command(["hug", "huggle"], autohelp=False)
+@hook.command("hug", "huggle", autohelp=False)
 def huggle(text, action, nick):
     """[user] - huggles [user]"""
     if text:
diff --git a/plugins/minecraft_status.py b/plugins/minecraft_status.py
index 855eccc..820d623 100644
--- a/plugins/minecraft_status.py
+++ b/plugins/minecraft_status.py
@@ -11,7 +11,7 @@ red_prefix = "\x02\x0f"
 red_suffix = ": \x034\x02\u2716"
 
 
-@hook.command(["mcs", "mcstatus", "mojang"], autohelp=False)
+@hook.command("mcs", "mcstatus", "mojang", autohelp=False)
 def mcstatus():
     """- gets the status of various Mojang (Minecraft) servers"""
 
diff --git a/plugins/steal.py b/plugins/steal.py
index 098bf99..8513d5f 100644
--- a/plugins/steal.py
+++ b/plugins/steal.py
@@ -44,7 +44,7 @@ def add_word(db, stolen):
         pass  # for lack of a better thing to do
 
 
-@hook.command(["steal", "stealit"], autohelp=False)
+@hook.command("steal", "stealit", autohelp=False)
 def stealit(inp, db, nick, action):
     """steal [username [object]] - Steals an object from a user, or a random previously stolen object."""
     args = inp.strip().split()
-- 
2.0.0

