From 8c3ca523e5df6bcea7591c91a148409d35016a91 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Thu, 29 May 2014 22:01:16 -0700
Subject: [PATCH] Catch the case where bukget returns no authors for a given
 plugin


diff --git a/plugins/minecraft_bukget.py b/plugins/minecraft_bukget.py
index 5bdef9e..fff0559 100644
--- a/plugins/minecraft_bukget.py
+++ b/plugins/minecraft_bukget.py
@@ -1,3 +1,4 @@
+import json
 import time
 import random
 
@@ -88,8 +89,12 @@ def format_output(data):
     name = data["plugin_name"]
     description = formatting.truncate_str(data['description'], 30)
     url = data['website']
-    authors = data['authors'][0]
-    authors = authors[0] + "\u200b" + authors[1:]
+    if data['authors']:
+        authors = data['authors'][0]
+        authors = authors[0] + "\u200b" + authors[1:]
+    else:
+        authors = "Unknown"
+
     stage = data['stage']
 
     current_version = data['versions'][0]
-- 
2.0.0

