From 0a81c759148938073507d0c7b55b1044806a360a Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 23 Apr 2014 16:39:07 -0700
Subject: [PATCH] Disable the command limiting system when using qpt


diff --git a/plugins/core_sieve.py b/plugins/core_sieve.py
index 78e0d48..eb85b10 100644
--- a/plugins/core_sieve.py
+++ b/plugins/core_sieve.py
@@ -1,5 +1,5 @@
 import asyncio
-from cloudbot import bucket, hook
+from cloudbot import bucket, hook, qpt
 
 TOKENS = 10
 RESTORE_RATE = 2
@@ -54,6 +54,8 @@ def sieve_suite(bot, event, _hook):
 
     # check command spam tokens
     if _hook.type == "command":
+        if qpt.is_enabled():
+            return event  # Don't run tokens when testing plugins - need to do it quickly
         uid = event.chan
 
         if not uid in buckets:
-- 
2.0.0

