From 1a738e0f3e05d3c907ecdc7254447fd9142b5785 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Thu, 1 May 2014 03:05:13 -0700
Subject: [PATCH] Add a plugin to trigger objgraph, to look for memory leaks


diff --git a/modules/graph.py b/modules/graph.py
new file mode 100644
index 0000000..d051e15
--- /dev/null
+++ b/modules/graph.py
@@ -0,0 +1,17 @@
+import objgraph
+import gc
+
+from util import hook
+
+
+@hook.command(autohelp=False)
+def show_types():
+    print(gc.collect())
+    objgraph.show_most_common_types(limit=20)
+
+
+@hook.command(autohelp=False)
+def show_growth():
+    print(gc.collect())
+    objgraph.show_growth(limit=10)
+
diff --git a/requirements.txt b/requirements.txt
index ffa21e9..af1b1f1 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -9,3 +9,4 @@ pycrypto
 pbkdf2
 tweepy
 yql
+objgraph
-- 
1.9.2

