From 1adc2d41b289ba7ebbda5c49336a23b317b89326 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 12 May 2014 15:09:15 -0700
Subject: [PATCH] Make core_misc's keepalive work with asyncio.sleep to avoid
 having an unrequired thread.


diff --git a/modules/core_misc.py b/modules/core_misc.py
index eb08262..69967f0 100644
--- a/modules/core_misc.py
+++ b/modules/core_misc.py
@@ -1,3 +1,4 @@
+import asyncio
 import socket
 import time
 
@@ -58,7 +59,7 @@ def onjoin(conn, bot):
     bot.logger.info("ONJOIN hook completed. Bot ready.")
 
 
-@hook.event('004', threaded=True, singlethread=True)
+@hook.event('004', threaded=False)
 def keep_alive(conn):
     """
     :type conn: core.irc.BotConnection
@@ -67,4 +68,4 @@ def keep_alive(conn):
     if keepalive:
         while True:
             conn.cmd('PING', [conn.nick])
-            time.sleep(60)
+            yield from asyncio.sleep(60)
-- 
1.9.3

