From c4ead2167188bcb06234bddcde8f96ca0ac6030e Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Thu, 29 May 2014 14:22:15 -0700
Subject: [PATCH] Hotfix for regex_chans


diff --git a/plugins/regex_chans.py b/plugins/regex_chans.py
index 61a7903..fcfac80 100644
--- a/plugins/regex_chans.py
+++ b/plugins/regex_chans.py
@@ -54,16 +54,15 @@ def delete_status(db, conn, chan):
 
 
 @hook.sieve()
-def sieve_regex(bot, input, plugin):
-    if plugin.type == "regex" and input.chan.startswith("#") and plugin.module.title != "factoids":
-        status = status_cache.get((input.conn.name, input.chan))
+def sieve_regex(bot, event, _hook):
+    if _hook.type == "regex" and event.chan.startswith("#") and _hook.plugin.title != "factoids":
+        status = status_cache.get((event.conn.name, event.chan))
         if status != "ENABLED" and (status == "DISABLED" or not default_enabled):
-            bot.logger.info("[{}] Denying {} from {}"
-                            .format(input.conn.readable_name, plugin.function_name, input.chan))
+            bot.logger.info("[{}] Denying {} from {}".format(event.conn.readable_name, _hook.function_name, event.chan))
             return None
-        bot.logger.info("[{}] Allowing {} to {}".format(input.conn.readable_name, plugin.function_name, input.chan))
+        bot.logger.info("[{}] Allowing {} to {}".format(event.conn.readable_name, _hook.function_name, event.chan))
 
-    return input
+    return event
 
 
 @hook.command(autohelp=False, permissions=["botcontrol"])
-- 
1.9.3

