From ea51c370bb3715f8d070e0ad4e7625ce92fe1a7f Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 23 Apr 2014 16:39:07 -0700
Subject: [PATCH] Disable the command limiting system when using qpt


diff --git a/plugins/core_sieve.py b/plugins/core_sieve.py
index 2329f79..760485d 100644
--- a/plugins/core_sieve.py
+++ b/plugins/core_sieve.py
@@ -1,4 +1,4 @@
-from cloudbot import bucket, hook
+from cloudbot import bucket, hook, qpt
 
 TOKENS = 10
 RESTORE_RATE = 2
@@ -52,6 +52,8 @@ def sieve_suite(bot, event, _hook):
 
     # check command spam tokens
     if _hook.type == "command":
+        if qpt.is_enabled():
+            return event  # Don't run tokens when testing plugins - need to do it quickly
         uid = event.chan
 
         if not uid in buckets:
-- 
2.0.0

