From d119ff577749713edce0676a21d59060fb123652 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 27 May 2014 01:29:58 -0700
Subject: [PATCH] Update the main 400 event to use asyncio


diff --git a/plugins/core_misc.py b/plugins/core_misc.py
index 2a0c8a2..a296c88 100644
--- a/plugins/core_misc.py
+++ b/plugins/core_misc.py
@@ -1,6 +1,5 @@
 import asyncio
 import socket
-import time
 
 from cloudbot import hook
 
@@ -12,7 +11,7 @@ socket.setdefaulttimeout(10)
 def invite(paramlist, conn):
     """
     :type paramlist: list[str]
-    :type conn: core.irc.BotConnection
+    :type conn: cloudbot.core.connection.BotConnection
     """
     invite_join = conn.config.get('invite_join', True)
     if invite_join:
@@ -20,11 +19,11 @@ def invite(paramlist, conn):
 
 
 # Identify to NickServ (or other service)
-@hook.event('004')
+@hook.event('004', threaded=False)
 def onjoin(conn, bot):
     """
-    :type conn: core.irc.BotConnection
-    :type bot: core.bot.CloudBot
+    :type conn: cloudbot.core.connection.BotConnection
+    :type bot: cloudbot.core.bot.CloudBot
     """
     bot.logger.info("ONJOIN hook triggered.")
     nickserv = conn.config.get('nickserv')
@@ -42,7 +41,7 @@ def onjoin(conn, bot):
                 conn.msg(nickserv_name, "{} {}".format(nickserv_command, nickserv_password))
             if "censored_strings" in bot.config:
                 bot.config['censored_strings'].append(nickserv_password)
-            time.sleep(1)
+            yield from asyncio.sleep(1)
 
     # Set bot modes
     mode = conn.config.get('mode')
@@ -54,7 +53,7 @@ def onjoin(conn, bot):
     bot.logger.info('Joining channels.')
     for channel in conn.channels:
         conn.join(channel)
-        time.sleep(1)
+        yield from asyncio.sleep(1)
 
     bot.logger.info("ONJOIN hook completed. Bot ready.")
 
-- 
1.9.3

