From a209d3c6f8123823106865d0ee5d4080cb04c3cf Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 12 May 2014 15:18:07 -0700
Subject: [PATCH] Use call_soon_threadsafe to stop bot in exit_gracefully


diff --git a/cloudbot.py b/cloudbot.py
index 179ad47..de25ceb 100755
--- a/cloudbot.py
+++ b/cloudbot.py
@@ -37,7 +37,7 @@ def main():
             # we are currently in the process of restarting
             stopped_while_restarting = True
         else:
-            cloudbot.stop()
+            cloudbot.loop.call_soon_threadsafe(cloudbot.stop, "Killed")
 
         # restore the original handler so if they do it again it triggers
         signal.signal(signal.SIGINT, original_sigint)
-- 
1.9.3

