From 42e676d6287c30c8635608ed4e27e5ff2874b098 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 1 Jun 2014 00:42:43 -0700
Subject: [PATCH] Formatting imports


diff --git a/plugins/debug.py b/plugins/debug.py
index 9896f08..55d5dd4 100644
--- a/plugins/debug.py
+++ b/plugins/debug.py
@@ -1,8 +1,10 @@
 import os
+
 import psutil
 
 from cloudbot import hook
 
+
 @hook.command
 def debug():
     # get a Process object for the bot using psutil
diff --git a/plugins/fishbans.py b/plugins/fishbans.py
index 24d626c..4b96ee6 100644
--- a/plugins/fishbans.py
+++ b/plugins/fishbans.py
@@ -1,8 +1,9 @@
 from urllib.parse import quote_plus
 
-import requests
 import asyncio
 
+import requests
+
 from cloudbot import hook, formatting
 
 
diff --git a/plugins/fmylife.py b/plugins/fmylife.py
index 593d8d1..86a63c7 100644
--- a/plugins/fmylife.py
+++ b/plugins/fmylife.py
@@ -1,10 +1,10 @@
-from bs4 import BeautifulSoup
 import asyncio
+
+from bs4 import BeautifulSoup
 import requests
 
 from cloudbot import hook
 
-
 fml_cache = []
 
 
diff --git a/plugins/log.py b/plugins/log.py
index df44215..cbc5153 100644
--- a/plugins/log.py
+++ b/plugins/log.py
@@ -4,7 +4,6 @@ log.py: written by Scaevolus 2009
 edited 2014
 """
 import asyncio
-
 import os
 import codecs
 import time
diff --git a/plugins/minecraft_bukget.py b/plugins/minecraft_bukget.py
index 1dfe9d8..7bdd01b 100644
--- a/plugins/minecraft_bukget.py
+++ b/plugins/minecraft_bukget.py
@@ -1,4 +1,3 @@
-import json
 import time
 import random
 
diff --git a/plugins/newegg.py b/plugins/newegg.py
index 3ca14f2..cdf90c2 100644
--- a/plugins/newegg.py
+++ b/plugins/newegg.py
@@ -4,6 +4,7 @@ import re
 from cloudbot import hook, http, formatting, web
 
 
+
 ## CONSTANTS
 
 ITEM_URL = "http://www.newegg.com/Product/Product.aspx?Item={}"
diff --git a/plugins/profiling.py b/plugins/profiling.py
index 2ed38e2..ca001e5 100644
--- a/plugins/profiling.py
+++ b/plugins/profiling.py
@@ -1,6 +1,6 @@
 import gc
-
 import asyncio
+
 import objgraph
 from pympler import muppy, summary, tracker
 
diff --git a/plugins/rss.py b/plugins/rss.py
index 2c0930e..997e01f 100644
--- a/plugins/rss.py
+++ b/plugins/rss.py
@@ -1,4 +1,4 @@
-from cloudbot import hook, http, formatting, web
+from cloudbot import hook, formatting, web
 
 
 @hook.command("feed")
diff --git a/plugins/wow.py b/plugins/wow.py
index 784ae13..9600da6 100644
--- a/plugins/wow.py
+++ b/plugins/wow.py
@@ -8,11 +8,13 @@ armoury, armory: Request data from the armoury API and format it into something
 """
 
 import re
+
 import requests
-from util import hook, http, web
 
-@hook.command('armory')
-@hook.command
+from cloudbot import hook, web
+
+
+@hook.command(["armory", "armoury"])
 def armoury(inp):
     """armoury  [realm] [character name] [region = EU] - Look up character and returns API data."""
 
@@ -52,6 +54,7 @@ def armoury(inp):
 
     return wow_armoury_data(link)
 
+
 def wow_armoury_data(link):
     """Sends the API request, and returns the data accordingly (in json if raw, nicely formatted if not)."""
     try:
@@ -61,6 +64,7 @@ def wow_armoury_data(link):
 
     return wow_armoury_format(data, link)
 
+
 def wow_armoury_format(data, link):
     """Format armoury data into a human readable string"""
 
@@ -85,13 +89,15 @@ def wow_armoury_format(data, link):
 
         try:
             return '{0} is a level \x0307{1}\x0F {2} {3} on {4} with \x0307{5}\x0F achievement points and \x0307{6}\x0F honourable kills. Armoury Profile: {7}' \
-                .format(data['name'], data['level'], wow_get_gender(data['gender']), wow_get_class(data['class'], True), data['realm'],
-                    data['achievementPoints'], data['totalHonorableKills'], web.isgd(niceurl))
+                .format(data['name'], data['level'], wow_get_gender(data['gender']), wow_get_class(data['class'], True),
+                        data['realm'],
+                        data['achievementPoints'], data['totalHonorableKills'], web.isgd(niceurl))
         except Exception as e:
             return 'Unable to fetch information for {}. Does the realm or character exist? ({})'.format(niceurl, str(e))
 
     return 'An unexpected error occured.'
 
+
 def wow_get_gender(genderid):
     """Formats a gender ID to a readable gender name"""
     gender = 'unknown'
@@ -103,7 +109,8 @@ def wow_get_gender(genderid):
 
     return gender
 
-def wow_get_class(classid, colours = False):
+
+def wow_get_class(classid, colours=False):
     """Formats a class ID to a readable name, data from http://eu.battle.net/api/wow/data/character/classes"""
     if colours:
         # Format their colours according to class colours.
@@ -119,12 +126,12 @@ def wow_get_class(classid, colours = False):
             11: "Druid"
         }
 
-
     if classid in classids:
         return classids[classid]
     else:
         return 'unknown'
 
+
 def wow_get_race(raceid):
     """Formats a race ID to a readable race name, data from http://eu.battle.net/api/wow/data/character/races"""
     raceids = {
@@ -138,6 +145,7 @@ def wow_get_race(raceid):
     else:
         return 'unknown'
 
+
 def wow_region_shortname(region):
     """Returns a short region name, which functions as battle.net their subdomain (i.e. eu.battle.net)"""
     validregions = {
-- 
2.0.0

