From da362d3d032a6c9dc0a45bfbaed109704abe9c55 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Fri, 30 May 2014 18:40:18 -0700
Subject: [PATCH] More debugging stuff


diff --git a/cloudbot/core/events.py b/cloudbot/core/events.py
index 4b48ea6..8e3c693 100644
--- a/cloudbot/core/events.py
+++ b/cloudbot/core/events.py
@@ -212,10 +212,12 @@ class CommandEvent(BaseEvent):
         """sends a notice containing this command's docstring to the current channel/user or a specific channel/user
         :type target: str
         """
+        if self.triggered_command is None:
+            raise ValueError("Triggered command not set on this event")
         if self.hook.doc is None:
-            message = self.conn.config["command_prefix"] + self.hook.name + " requires additional arguments."
+            message = self.conn.config["command_prefix"] + self.triggered_command + " requires additional arguments."
         else:
-            message = self.conn.config["command_prefix"] + self.hook.name + self.hook.doc
+            message = self.conn.config["command_prefix"] + self.triggered_command + self.hook.doc
         self.notice(message, target=target)
 
 
diff --git a/cloudbot/core/pluginmanager.py b/cloudbot/core/pluginmanager.py
index 2482786..ceed6cc 100644
--- a/cloudbot/core/pluginmanager.py
+++ b/cloudbot/core/pluginmanager.py
@@ -398,10 +398,15 @@ class PluginManager:
         :type hook: cloudbot.core.pluginmanager.Hook | cloudbot.core.pluginmanager.CommandHook
         :rtype: bool
         """
+        self.bot.logger.info("Launching {}:{}".format(hook.plugin.title, hook.function_name))
         if hook.type != "onload":  # we don't need sieves on onload hooks.
             for sieve in self.bot.plugin_manager.sieves:
                 event = yield from self._sieve(sieve, event, hook)
                 if event is None:
+                    self.bot.logger.info("Canceled launching {}:{} because of {}:{}".format(hook.plugin.title,
+                                                                                            hook.function_name,
+                                                                                            sieve.plugin.title,
+                                                                                            sieve.function_name))
                     return False
 
         if hook.type == "command" and hook.auto_help and not event.text and hook.doc is not None:
-- 
2.0.0

