From e77d6f3d35d838d2860d08d2eb861cd38d083d9c Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 29 Apr 2014 12:31:44 -0700
Subject: [PATCH] Change regex_chans default from True to False


diff --git a/plugins/regex_chans.py b/plugins/regex_chans.py
index 498f8a1..2cb835c 100644
--- a/plugins/regex_chans.py
+++ b/plugins/regex_chans.py
@@ -15,7 +15,7 @@ table = Table(
 # Default value.
 # If True, all channels without a setting will have regex enabled
 # If False, all channels without a setting will have regex disabled
-default_enabled = True
+default_enabled = False
 
 
 @hook.onload()
-- 
2.0.0

