From f2d8b0858b2851f6c79117a22062272b78117795 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 13 Jan 2014 23:13:03 -0800
Subject: [PATCH] Change ctcp reply to return 'Eliralin'


diff --git a/plugins/core_ctcp.py b/plugins/core_ctcp.py
index 557c24a..8a1b601 100644
--- a/plugins/core_ctcp.py
+++ b/plugins/core_ctcp.py
@@ -9,7 +9,7 @@ import cloudbot
 @asyncio.coroutine
 @hook.regex(r'^\x01VERSION\x01$')
 def ctcp_version(notice):
-    notice("\x01VERSION: CloudBotRefresh v{} - http://cloudbot.pw/".format(cloudbot.__version__))
+    notice("\x01VERSION: Eliralin v{} - http://git.io/elirefresh".format(cloudbot.__version__))
 
 
 @asyncio.coroutine
-- 
2.0.0

