From a1f6edec7e81e8e772b25adfa5332bab474d981c Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sat, 26 Apr 2014 19:43:21 -0700
Subject: [PATCH] Replace symbol for factoids.

Changes ?fact to ^fact

diff --git a/plugins/factoids.py b/plugins/factoids.py
index 7ece5cf..dd23d97 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -11,7 +11,7 @@ from cloudbot.util import botvars, formatting, web
 
 re_lineends = re.compile(r'[\r\n]*')
 
-FACTOID_CHAR = "?"  # TODO: config
+FACTOID_CHAR = "^"  # TODO: config
 
 # some simple "shortcodes" for formatting purposes
 shortcodes = {
-- 
2.0.0

