From 7da216fcf237fe4167964e74c4fdc0073171ccd8 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 18 Jun 2014 18:13:56 -0500
Subject: [PATCH] Fix weather.isgd to weather.shorten


diff --git a/plugins/weather.py b/plugins/weather.py
index 0a78a1c..79d6ee0 100644
--- a/plugins/weather.py
+++ b/plugins/weather.py
@@ -86,7 +86,7 @@ def weather(text, reply, db, nick, bot, notice):
         "tomorrow_high_c": forecast_tomorrow['high']['celsius'],
         "tomorrow_low_f": forecast_tomorrow['low']['fahrenheit'],
         "tomorrow_low_c": forecast_tomorrow['low']['celsius'],
-        "url": web.isgd(response["current_observation"]['forecast_url'] + "?apiref=e535207ff4757b18")
+        "url": web.shorten(response["current_observation"]['forecast_url'] + "?apiref=e535207ff4757b18")
     }
 
     reply("{place} - \x02Current:\x02 {conditions}, {temp_f}F/{temp_c}C, {humidity}, "
-- 
2.0.0

