From 46dc327eae1e2e510bd02dbbeb3c4c0612f7ce61 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 17 Jun 2014 23:59:52 -0500
Subject: [PATCH] Use CloudBotRefresh - http://cloudbot.pw instead of CloudBot
 - http://git.io/refresh for the default realname


diff --git a/config.default b/config.default
index 78c7123..238d3b1 100644
--- a/config.default
+++ b/config.default
@@ -11,7 +11,7 @@
             },
             "nick": "MyCloudBot",
             "user": "cloudbot",
-            "real_name": "CloudBot - http://git.io/refresh",
+            "real_name": "CloudBotRefresh - http://cloudbot.pw",
             "channels": ["#cloudbot", "#cloudbot2"],
             "disabled_commands": [],
             "acls": {},
-- 
2.0.0

