From 06b0cb2b98ae22d0412a95c3e64dcaa50bd4e640 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Wed, 18 Jun 2014 20:12:16 -0500
Subject: [PATCH] Not at all a tracking system


diff --git a/cloudbot/connection.py b/cloudbot/connection.py
index 4d48010..16845c8 100644
--- a/cloudbot/connection.py
+++ b/cloudbot/connection.py
@@ -121,3 +121,39 @@ class Connection:
     @property
     def connected(self):
         raise NotImplementedError
+
+
+# TODO: Tracking of user 'mode' in channels
+class User:
+    """
+    :param name: The nickname of this User
+    :param ident: The IRC ident of this User, if applicable
+    :param host: The hostname of this User, if applicable
+    :param mask: The IRC mask (nick!ident@host), if applicable
+    :type name: str
+    :type ident: str
+    :type host: str
+    :type mask: str
+    """
+
+    def __init__(self, name, ident, host, mask):
+        self.name = name
+        self.ident = ident
+        self.host = host
+        self.mask = mask
+
+
+class Channel:
+    """
+    name: the name of this channel
+    users: A dict from nickname to User in this channel
+    user_modes: A dict from User to an str containing all of the user's modes in this channel
+    history: A list of (User, message content)
+    :type name: str
+    :type users: dict[str, User]
+    :type user_modes: dict[User, str]
+    """
+    def __init__(self, name):
+        self.name = name
+        self.users = {}
+        self.user_modes = {}
-- 
2.0.0

