From 278905f02fe994a36c0d5d2d5f759351533bfb64 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 17 Jun 2014 23:30:33 -0500
Subject: [PATCH] Change default realname to CloudBotRefresh, using cloudbot.pw


diff --git a/config.default b/config.default
index 9248a53..29a3b86 100644
--- a/config.default
+++ b/config.default
@@ -11,7 +11,7 @@
             },
             "nick": "MyCloudBot",
             "user": "cloudbot",
-            "real_name": "CloudBot - http://git.io/refresh",
+            "real_name": "CloudBotRefresh - http://cloudbot.pw",
             "channels": ["#cloudbot", "#cloudbot2"],
             "disabled_commands": [],
             "acls": {},
-- 
2.0.0

