From 15c1f7c83617ff7e5975bebb0ed9006d357ec090 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Tue, 17 Jun 2014 23:31:57 -0500
Subject: [PATCH] Remove the event.server property


diff --git a/cloudbot/core/events.py b/cloudbot/core/events.py
index a1895ff..62830c5 100644
--- a/cloudbot/core/events.py
+++ b/cloudbot/core/events.py
@@ -207,16 +207,6 @@ class BaseEvent:
             self.db = None
 
     @property
-    def server(self):
-        """
-        :rtype: str
-        """
-        if self.conn is not None:
-            return self.conn.server
-        else:
-            return None
-
-    @property
     def event(self):
         """
         :rtype; cloudbot.core.events.BaseEvent
-- 
2.0.0

