From 3a90894d02e0702275518979d69460bff29cd55f Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 16 Jun 2014 22:51:19 -0700
Subject: [PATCH] Add support for plugins using the old `commandname <args> --
 description` format, detecting it using `doc.split(None, 1)[0].isalpha()`


diff --git a/cloudbot/core/events.py b/cloudbot/core/events.py
index c0de108..7ba49f9 100644
--- a/cloudbot/core/events.py
+++ b/cloudbot/core/events.py
@@ -324,7 +324,13 @@ class CommandEvent(BaseEvent):
             message = "{}{} requires additional arguments.".format(self.conn.config["command_prefix"],
                                                                    self.triggered_command)
         else:
-            message = "{}{} {}".format(self.conn.config["command_prefix"], self.triggered_command, self.hook.doc)
+            if self.hook.doc.split()[0].isalpha():
+                # this is using the old format of `name <args> - doc`
+                message = "{}{}".format(self.conn.config["command_prefix"], self.hook.doc)
+            else:
+                # this is using the new format of `<args> - doc`
+                message = "{}{} {}".format(self.conn.config["command_prefix"], self.triggered_command, self.hook.doc)
+
         self.notice(message, target=target)
 
 
diff --git a/plugins/help.py b/plugins/help.py
index 958c9d0..31677f0 100644
--- a/plugins/help.py
+++ b/plugins/help.py
@@ -25,7 +25,13 @@ def help_command(text, conn, bot, notice, has_permission):
         if searching_for in bot.plugin_manager.commands:
             doc = bot.plugin_manager.commands[searching_for].doc
             if doc:
-                notice(conn.config["command_prefix"] + doc)
+                if doc.split()[0].isalpha():
+                    # this is using the old format of `name <args> - doc`
+                    message = "{}{}".format(conn.config["command_prefix"], doc)
+                else:
+                    # this is using the new format of `<args> - doc`
+                    message = "{}{} {}".format(conn.config["command_prefix"], searching_for, doc)
+                notice(message)
             else:
                 notice("Command {} has no additional documentation.".format(searching_for))
         else:
-- 
2.0.0

